<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2017-03-26" />

<title>Calculate transcript abundances</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Calculate transcript abundances</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2017-03-26</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-03-28</p>
<p><strong>Code version:</strong> 0e2e80a</p>
<p>We can use several different methods to calculate transcript abundances. Here I will use Stringtie and HTSeq to calculate TPM values.</p>
<div id="running-stringtie" class="section level2">
<h2>Running stringtie</h2>
<p>First, we make sure we have the appropriate annotation file:</p>
<pre><code>cat data/annotations/exons_nuclear_3D7_v24.gff | \
  gffread -F -E -o- &gt; output/transcript_abundance/stringtie_annotation.gff</code></pre>
<p>Now let’s run Stringtie:</p>
<pre><code>for strain in 3d7 hb3 it; do
  for i in $(seq 1 7); do
    for s in rf fr; do
      stringtie -G output/transcript_abundance/annotation.gff \
        -e \
        --&quot;${s}&quot; \
        -p 8 \
        -A output/transcript_abundance/stringtie/&quot;${strain}_tp${i}_${s}.abund&quot; \
        &quot;data/bam/mapped_to_3d7/${strain}.3d7_v3_chr.tp${i}.bam&quot; &gt; \
        &quot;output/transcript_abundance/stringtie/${strain}_tp${i}_${s}.gtf&quot;
    done
  done
done</code></pre>
</div>
<div id="running-htseq" class="section level2">
<h2>Running HTSeq</h2>
<p>We’ll also quantify read counts using HTSeq. First generate another annotation since HTSeq likes using GTF files:</p>
<pre><code>cat data/annotations/exons_nuclear_3D7_v24.gff | \
  gffread -F -E -T -o- &gt; output/transcript_abundance/htseq_annotation.gtf</code></pre>
<p>And then run HTSeq using the following command:</p>
<pre><code>for strain in 3d7 hb3 it; do
  for i in $(seq 1 7); do
    for s in yes reverse; do
      htseq-count --format=bam --order=pos --stranded=${s} --minaqual=30 --idattr=transcript_id --mode=union \
        &quot;data/bam/mapped_to_3d7/${strain}.3d7_v3_chr.tp${i}.bam&quot; \
        &quot;output/transcript_abundance/htseq_annotation.gtf&quot; &gt; \
        &quot;output/transcript_abundance/htseq/${strain}_tp${i}_${s}.txt&quot;
    done
  done
done</code></pre>
</div>
<div id="clean-and-join-the-results" class="section level2">
<h2>Clean and join the results</h2>
<p>Now we’ll just import the results and reformat them for future use.</p>
<div id="stringtie" class="section level3">
<h3>Stringtie</h3>
<p>Here we’ll import the Stringtie data:</p>
<pre class="r"><code># For each strain
for (strain in c(&quot;3d7&quot;,&quot;hb3&quot;,&quot;it&quot;)) {
  # create a tibble with the right columns
  assign(x     = paste0(&quot;x&quot;,strain,&quot;_tpm&quot;),
         value = tibble::tibble(transcript_id=character(),
                      cov=double(),
                      FPKM=double(),
                      TPM=double(),
                      tp=integer(),
                      strain=character()))
  # for each timepoint
  for (tp in seq(1,7)) {
    # generate file path
    f &lt;- paste0(&quot;../output/transcript_abundance/stringtie/&quot;,strain,&quot;_tp&quot;,tp,&quot;_rf.gtf&quot;)
    # import the file and convert to the right format
    df &lt;- tibble::as_tibble(import.gff(f)) %&gt;% 
            dplyr::filter(type==&quot;transcript&quot;) %&gt;% 
            dplyr::select(transcript_id,cov,FPKM,TPM) %&gt;% 
            dplyr::mutate(cov=as.double(cov),FPKM=as.double(FPKM),TPM=as.double(TPM),tp=tp,strain=strain) %&gt;%
            dplyr::mutate(transcript_id=stringr::str_replace(transcript_id,&quot;rna_&quot;,&quot;&quot;)) %&gt;%
            dplyr::mutate(transcript_id=stringr::str_replace(transcript_id,&quot;-1$&quot;,&quot;&quot;))
    # bind this data to the tibble we created earlier for each strain
    assign(x     = paste0(&quot;x&quot;,strain,&quot;_tpm&quot;),
           value = dplyr::bind_rows(eval(parse(text=paste0(&quot;x&quot;,strain,&quot;_tpm&quot;))),df))
  }
}
# clean up
rm(df)</code></pre>
<p>And let’s save these to output:</p>
<pre class="r"><code># Sense and antisense counts in long format
saveRDS(object=x3d7_tpm,file=&quot;../output/transcript_abundance/stringtie_3d7_abund.rds&quot;)
saveRDS(object=xhb3_tpm,file=&quot;../output/transcript_abundance/stringtie_hb3_abund.rds&quot;)
saveRDS(object=xit_tpm,file=&quot;../output/transcript_abundance/stringtie_it_abund.rds&quot;)</code></pre>
</div>
<div id="htseq" class="section level3">
<h3>HTSeq</h3>
<p>In order to properly calculate TPMs, we need to calculate the effective transcript lengths. To do this we need the mean or median fragment length distribution for each sample. We can generate this data using picard:</p>
<pre><code>for file in $(find data/bam/mapped_to_3d7/ -name *.bam); do 
  picard CollectInsertSizeMetrics INPUT=$file HISTOGRAM_FILE=${file%.*}_insert_size.hist OUTPUT=${file%.*}_insert_size.out; 
done</code></pre>
<p>Now we can read each file in and record the mean fragment length:</p>
<pre class="r"><code># create a tibble with the right columns
inserts &lt;- tibble(sample=character(),mean_insert_size=double(),median_insert_size=double(),tp=integer())
# for each strain
for (strain in c(&quot;3d7&quot;,&quot;hb3&quot;,&quot;it&quot;)) {
  for (tp in seq(1,7)) {
    # generate the file path
    f &lt;- paste0(&quot;../data/bam/mapped_to_3d7/&quot;,strain,&quot;.3d7_v3_chr.tp&quot;,tp,&quot;_insert_size.out&quot;)
    # extract mean and median insert sizes
    mean_is &lt;- read_tsv(f,skip=6,n_max = 1)$MEAN_INSERT_SIZE
    median_is &lt;- read_tsv(f,skip=6,n_max = 1)$MEDIAN_INSERT_SIZE
    # bind this row to the tibble we created earlier
    inserts &lt;- dplyr::bind_rows(inserts,tibble(sample=paste0(&quot;x&quot;,strain,&quot;_tp&quot;,tp),mean_insert_size=mean_is,median_insert_size=median_is,tp=tp))
  }
}</code></pre>
<p>It turns out, however, that using effective transcript lengths in <em>Plasmodium</em> is problematic…Some transcripts end up with a negative length if we use the formula provided for us <a href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">here</a>. So we’ll simply use full transcript lengths instead of “correcting” for that bias.</p>
<p>Import the HTSeq data:</p>
<pre class="r"><code># import our calculate transcript lengths
transcript_lengths &lt;- read_tsv(&quot;../output/transcript_abundance/transcript_lengths.tsv&quot;)

# for each strain
for (strain in c(&quot;3d7&quot;,&quot;hb3&quot;,&quot;it&quot;)) {
  # create a tibble with the right colums
  assign(x     = paste0(&quot;x&quot;,strain,&quot;_counts&quot;),
         value = tibble(transcript_id=character(),
                        sense_counts=double(),
                        antisense_counts=double()))
  # for each time point
  for (tp in seq(1,7)) {
    # generate the file names
    s &lt;- paste0(&quot;../output/transcript_abundance/htseq/&quot;,strain,&quot;_tp&quot;,tp,&quot;_reverse.txt&quot;)
    a &lt;- paste0(&quot;../output/transcript_abundance/htseq/&quot;,strain,&quot;_tp&quot;,tp,&quot;_yes.txt&quot;)
    # and import sense and antisense counts and process their raw values
    st &lt;- read_tsv(s,comment=&quot;__&quot;,col_names=c(&quot;transcript_id&quot;,&quot;sense_counts&quot;)) %&gt;%
      dplyr::mutate(transcript_id=stringr::str_replace(transcript_id,&quot;rna_&quot;,&quot;&quot;)) %&gt;%
      dplyr::mutate(transcript_id=stringr::str_replace(transcript_id,&quot;-1$&quot;,&quot;&quot;))
    at &lt;- read_tsv(a,comment=&quot;__&quot;,col_names=c(&quot;transcript_id&quot;,&quot;antisense_counts&quot;)) %&gt;%       
      dplyr::mutate(transcript_id=stringr::str_replace(transcript_id,&quot;rna_&quot;,&quot;&quot;)) %&gt;%
      dplyr::mutate(transcript_id=stringr::str_replace(transcript_id,&quot;-1$&quot;,&quot;&quot;))
    # join the sense and antisense counts and give it an extra column
    df &lt;- inner_join(st,at) %&gt;% mutate(tp=tp)
    # bind the rows to the strain specific tibble generated above
    assign(x     = paste0(&quot;x&quot;,strain,&quot;_counts&quot;),
           value = dplyr::bind_rows(eval(parse(text=paste0(&quot;x&quot;,strain,&quot;_counts&quot;))),df))
  }
  # join this tibble with our transcript lengths for convenient TPM calculations
   assign(x     = paste0(&quot;x&quot;,strain,&quot;_counts&quot;),
           value = dplyr::inner_join(eval(parse(text=paste0(&quot;x&quot;,strain,&quot;_counts&quot;))),transcript_lengths))
}
# clean up
rm(s,a,st,at,df)</code></pre>
<p>Calculate TPM values:</p>
<pre class="r"><code># Function for calculating TPMs from the count tibbles
# generated above
# 
calc_tpms &lt;- function(counts) {
  # caluclate the normalizing factor for each sample
  # normFactor = sum of all length normalized transcript counts
  normFactors &lt;- counts %&gt;% 
  mutate(normCount=sense_counts/length) %&gt;% 
  group_by(tp) %&gt;% 
  summarise(normFactor=1/sum(normCount))
  
  # calculate the TPMs by joining it together with normFactors
  # and multiplying the length corrected counts by these factors
  # and by one million
  abund &lt;- inner_join(normFactors,counts) %&gt;%
  mutate(sense_tpm=(sense_counts/length)*normFactor*1e6,antisense_tpm=(antisense_counts/length)*normFactor*1e6) %&gt;%
  dplyr::select(transcript_id,sense_counts,antisense_counts,sense_tpm,antisense_tpm,tp)
  
  return(abund)
}

# calculate the TPMs for each strain
x3d7_abund &lt;- calc_tpms(x3d7_counts)
xhb3_abund &lt;- calc_tpms(xhb3_counts)
xit_abund  &lt;- calc_tpms(xit_counts)</code></pre>
<p>Let’s just make sure they all add up to a million as they should:</p>
<pre class="r"><code># sum up TPM values for each time point
x3d7_abund %&gt;% group_by(tp) %&gt;% summarise(total=sum(sense_tpm))</code></pre>
<pre><code># A tibble: 7 × 2
     tp total
  &lt;int&gt; &lt;dbl&gt;
1     1 1e+06
2     2 1e+06
3     3 1e+06
4     4 1e+06
5     5 1e+06
6     6 1e+06
7     7 1e+06</code></pre>
<pre class="r"><code>xhb3_abund %&gt;% group_by(tp) %&gt;% summarise(total=sum(sense_tpm))</code></pre>
<pre><code># A tibble: 7 × 2
     tp total
  &lt;int&gt; &lt;dbl&gt;
1     1 1e+06
2     2 1e+06
3     3 1e+06
4     4 1e+06
5     5 1e+06
6     6 1e+06
7     7 1e+06</code></pre>
<pre class="r"><code>xit_abund %&gt;% group_by(tp) %&gt;% summarise(total=sum(sense_tpm))</code></pre>
<pre><code># A tibble: 7 × 2
     tp total
  &lt;int&gt; &lt;dbl&gt;
1     1 1e+06
2     2 1e+06
3     3 1e+06
4     4 1e+06
5     5 1e+06
6     6 1e+06
7     7 1e+06</code></pre>
<p>Write these abundance values to output:</p>
<pre class="r"><code># Sense TPM tables
select(x3d7_abund,transcript_id,sense_tpm,tp) %&gt;% spread(tp,sense_tpm) %&gt;%
  write_tsv(path=&quot;../output/transcript_abundance/htseq_3d7_sense_tpms.tsv&quot;)
select(xhb3_abund,transcript_id,sense_tpm,tp) %&gt;% spread(tp,sense_tpm) %&gt;%
  write_tsv(path=&quot;../output/transcript_abundance/htseq_hb3_sense_tpms.tsv&quot;)
select(xit_abund,transcript_id,sense_tpm,tp) %&gt;% spread(tp,sense_tpm) %&gt;%
  write_tsv(path=&quot;../output/transcript_abundance/htseq_it_sense_tpms.tsv&quot;)

# Antisense TPM tables
select(x3d7_abund,transcript_id,antisense_tpm,tp) %&gt;% spread(tp,antisense_tpm) %&gt;%
  write_tsv(path=&quot;../output/transcript_abundance/htseq_3d7_antisense_tpms.tsv&quot;)
select(xhb3_abund,transcript_id,antisense_tpm,tp) %&gt;% spread(tp,antisense_tpm) %&gt;%
  write_tsv(path=&quot;../output/transcript_abundance/htseq_hb3_antisense_tpms.tsv&quot;)
select(xit_abund,transcript_id,antisense_tpm,tp) %&gt;% spread(tp,antisense_tpm) %&gt;%
  write_tsv(path=&quot;../output/transcript_abundance/htseq_it_antisense_tpms.tsv&quot;)

# Sense and antisense counts in long format
saveRDS(object=x3d7_abund,file=&quot;../output/transcript_abundance/htseq_3d7_abund.rds&quot;)
saveRDS(object=xhb3_abund,file=&quot;../output/transcript_abundance/htseq_hb3_abund.rds&quot;)
saveRDS(object=xit_abund,file=&quot;../output/transcript_abundance/htseq_it_abund.rds&quot;)</code></pre>
<p>That’s it! Now we can use these files for future analyses.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] rtracklayer_1.34.2   GenomicRanges_1.26.3 GenomeInfoDb_1.10.3 
 [4] IRanges_2.8.1        S4Vectors_0.12.1     BiocGenerics_0.20.0 
 [7] dplyr_0.5.0          purrr_0.2.2          readr_1.0.0         
[10] tidyr_0.6.1          tibble_1.2           ggplot2_2.2.1       
[13] tidyverse_1.1.1     

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.4.0 reshape2_1.4.2            
 [3] haven_1.0.0                lattice_0.20-34           
 [5] colorspace_1.3-2           htmltools_0.3.5           
 [7] yaml_2.1.14                XML_3.98-1.5              
 [9] foreign_0.8-67             DBI_0.5-1                 
[11] BiocParallel_1.8.1         modelr_0.1.0              
[13] readxl_0.1.1               plyr_1.8.4                
[15] stringr_1.2.0              zlibbioc_1.20.0           
[17] Biostrings_2.42.1          workflowr_0.3.0           
[19] munsell_0.4.3              gtable_0.2.0              
[21] rvest_0.3.2                psych_1.6.12              
[23] evaluate_0.10              Biobase_2.34.0            
[25] knitr_1.15.1               forcats_0.2.0             
[27] broom_0.4.2                Rcpp_0.12.9               
[29] scales_0.4.1               backports_1.0.5           
[31] jsonlite_1.3               XVector_0.14.0            
[33] Rsamtools_1.26.1           mnormt_1.5-5              
[35] hms_0.3                    digest_0.6.12             
[37] stringi_1.1.2              grid_3.3.2                
[39] rprojroot_1.2              tools_3.3.2               
[41] bitops_1.0-6               magrittr_1.5              
[43] lazyeval_0.2.0             RCurl_1.95-4.8            
[45] Matrix_1.2-8               xml2_1.1.1                
[47] lubridate_1.6.0            assertthat_0.1            
[49] rmarkdown_1.3              httr_1.2.1                
[51] R6_2.2.0                   git2r_0.18.0              
[53] GenomicAlignments_1.10.0   nlme_3.1-131              </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
