<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2018-09-24" />

<title>Comparing predicted UTRs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Comparing predicted UTRs</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2018-09-24</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-09-25</p>
<p><strong>Code version:</strong> c5562e7</p>
<div id="comparing-different-utr-and-tss-predictions" class="section level2">
<h2>Comparing different UTR and TSS predictions</h2>
<p>Since we were able to predict both UTRs and TSSs using our data, we wanted to know how our predictions compared to previously published predictions. Here, we compare the UTRs predicted in <a href="https://www.ncbi.nlm.nih.gov/pubmed/25493618">Caro et al.</a> and <a href="https://www.ncbi.nlm.nih.gov/pubmed/26947071">Adjalley et al.</a>.</p>
<p>What we see is that although there are some large deviations, for the majority of 5UTR and TSS predictions, the results are not very different with a mean hovering around zero base pairs of difference between the start positions of our predicted 5’ UTRs and TSS and those that were previously published.</p>
<p>First, let’s import our 5UTR data:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(org.Pf.plasmo.db)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">aliases &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(<span class="kw">data.frame</span>(org.Pf.plasmoALIAS2ORF))</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4">utrs &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.gff</span>(<span class="st">&quot;../output/final_utrs/final_5utrs_3d7.gff&quot;</span>))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">utrs<span class="op">$</span>Parent &lt;-<span class="st"> </span><span class="kw">unlist</span>(utrs<span class="op">$</span>Parent)</a></code></pre></div>
<div id="comparing-to-caro-et-al." class="section level3">
<h3>Comparing to Caro et al.</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Comparing our UTR estimates to the Derisi predictions</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">derisi1 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.bed</span>(<span class="st">&quot;../data/compare_utrs/GSM1410291_UTRs_1.bed&quot;</span>))</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">derisi2 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.bed</span>(<span class="st">&quot;../data/compare_utrs/GSM1410292_UTRs_2.bed&quot;</span>))</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">derisi3 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.bed</span>(<span class="st">&quot;../data/compare_utrs/GSM1410293_UTRs_3.bed&quot;</span>))</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">derisi4 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.bed</span>(<span class="st">&quot;../data/compare_utrs/GSM1410294_UTRs_4.bed&quot;</span>))</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">derisi5 &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.bed</span>(<span class="st">&quot;../data/compare_utrs/GSM1410295_UTRs_5.bed&quot;</span>))</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">fix_derisi_utrs &lt;-<span class="st"> </span><span class="cf">function</span>(set) {</a>
<a class="sourceLine" id="cb2-10" data-line-number="10">  set<span class="op">$</span>name &lt;-<span class="st"> </span>stringi<span class="op">::</span><span class="kw">stri_replace_last</span>(set<span class="op">$</span>name,<span class="dt">replacement=</span><span class="st">&quot; &quot;</span>,<span class="dt">regex=</span><span class="st">&quot;_&quot;</span>)</a>
<a class="sourceLine" id="cb2-11" data-line-number="11">  set &lt;-<span class="st"> </span>set <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">separate</span>(name,<span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>,<span class="st">&quot;type&quot;</span>), <span class="dt">sep =</span><span class="st">&quot; &quot;</span>)</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">  set<span class="op">$</span>gene_id &lt;-<span class="st"> </span><span class="kw">toupper</span>(set<span class="op">$</span>gene_id)</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">  out &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(set, aliases, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>=<span class="st">&quot;alias_symbol&quot;</span>))</a>
<a class="sourceLine" id="cb2-14" data-line-number="14">  <span class="kw">return</span>(out)</a>
<a class="sourceLine" id="cb2-15" data-line-number="15">}</a>
<a class="sourceLine" id="cb2-16" data-line-number="16"></a>
<a class="sourceLine" id="cb2-17" data-line-number="17">derisi1 &lt;-<span class="st"> </span><span class="kw">fix_derisi_utrs</span>(derisi1)</a>
<a class="sourceLine" id="cb2-18" data-line-number="18">derisi2 &lt;-<span class="st"> </span><span class="kw">fix_derisi_utrs</span>(derisi2)</a>
<a class="sourceLine" id="cb2-19" data-line-number="19">derisi3 &lt;-<span class="st"> </span><span class="kw">fix_derisi_utrs</span>(derisi3)</a>
<a class="sourceLine" id="cb2-20" data-line-number="20">derisi4 &lt;-<span class="st"> </span><span class="kw">fix_derisi_utrs</span>(derisi4)</a>
<a class="sourceLine" id="cb2-21" data-line-number="21">derisi5 &lt;-<span class="st"> </span><span class="kw">fix_derisi_utrs</span>(derisi5)</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(derisi1, gene_id.y, width)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, width)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">compare_derisi &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id.y&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">rm</span>(tmp1,tmp2)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x,width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">color=</span><span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x<span class="op">-</span>width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">cor.test</span>(compare_derisi<span class="op">$</span>width.x,compare_derisi<span class="op">$</span>width.y)</a></code></pre></div>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  compare_derisi$width.x and compare_derisi$width.y
t = 44.624, df = 7619, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4372115 0.4728149
sample estimates:
      cor 
0.4551951 </code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(derisi2, gene_id.y, width)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, width)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">compare_derisi &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id.y&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="kw">rm</span>(tmp1,tmp2)</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x,width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">color=</span><span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x<span class="op">-</span>width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">cor.test</span>(compare_derisi<span class="op">$</span>width.x,compare_derisi<span class="op">$</span>width.y)</a></code></pre></div>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  compare_derisi$width.x and compare_derisi$width.y
t = 39.839, df = 7926, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3899543 0.4266379
sample estimates:
     cor 
0.408461 </code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(derisi3, gene_id.y, width)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, width)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">compare_derisi &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id.y&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="kw">rm</span>(tmp1,tmp2)</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x,width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">color=</span><span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x<span class="op">-</span>width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">cor.test</span>(compare_derisi<span class="op">$</span>width.x,compare_derisi<span class="op">$</span>width.y)</a></code></pre></div>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  compare_derisi$width.x and compare_derisi$width.y
t = 40.612, df = 7201, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4127160 0.4503005
sample estimates:
      cor 
0.4316956 </code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(derisi3, gene_id.y, width)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, width)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">compare_derisi &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id.y&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw">rm</span>(tmp1,tmp2)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x,width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">color=</span><span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x<span class="op">-</span>width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">cor.test</span>(compare_derisi<span class="op">$</span>width.x,compare_derisi<span class="op">$</span>width.y)</a></code></pre></div>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  compare_derisi$width.x and compare_derisi$width.y
t = 40.612, df = 7201, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.4127160 0.4503005
sample estimates:
      cor 
0.4316956 </code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(derisi4, gene_id.y, width)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, width)</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">compare_derisi &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id.y&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="kw">rm</span>(tmp1,tmp2)</a>
<a class="sourceLine" id="cb19-5" data-line-number="5"></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x,width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>()<span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">color=</span><span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x<span class="op">-</span>width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">cor.test</span>(compare_derisi<span class="op">$</span>width.x,compare_derisi<span class="op">$</span>width.y)</a></code></pre></div>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  compare_derisi$width.x and compare_derisi$width.y
t = 39.132, df = 7197, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3996269 0.4377261
sample estimates:
      cor 
0.4188608 </code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(derisi5, gene_id.y, width)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, width)</a>
<a class="sourceLine" id="cb23-3" data-line-number="3">compare_derisi &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_id.y&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="kw">rm</span>(tmp1,tmp2)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5"></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x,width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_abline</span>(<span class="dt">color=</span><span class="st">&quot;red&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">ggplot</span>(compare_derisi,<span class="kw">aes</span>(width.x<span class="op">-</span>width.y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">cor.test</span>(compare_derisi<span class="op">$</span>width.x,compare_derisi<span class="op">$</span>width.y)</a></code></pre></div>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  compare_derisi$width.x and compare_derisi$width.y
t = 29.152, df = 8625, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.2801573 0.3185779
sample estimates:
     cor 
0.299489 </code></pre>
</div>
<div id="comparing-to-adjalley-et-al." class="section level3">
<h3>Comparing to Adjalley et al.</h3>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="co"># Comparing our UTR estimates to the Adjalley predictions</span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2">adjalley &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.gff</span>(<span class="st">&quot;../data/compare_utrs/sorted_Adjalley_Chabbert_TSSs.gff&quot;</span>))</a>
<a class="sourceLine" id="cb27-3" data-line-number="3"></a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">8</span>) {</a>
<a class="sourceLine" id="cb27-5" data-line-number="5">  adjalley_filtered &lt;-<span class="st"> </span>adjalley <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">position=</span>(start<span class="op">+</span>end)<span class="op">/</span><span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb27-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(FilterSize<span class="op">&gt;</span>i)</a>
<a class="sourceLine" id="cb27-8" data-line-number="8">  tmp1 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(adjalley_filtered, AssignedFeat, position)</a>
<a class="sourceLine" id="cb27-9" data-line-number="9">  tmp2 &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(utrs, Parent, start)</a>
<a class="sourceLine" id="cb27-10" data-line-number="10">  compare_adjalley &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tmp1,tmp2,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;AssignedFeat&quot;</span>=<span class="st">&quot;Parent&quot;</span>))</a>
<a class="sourceLine" id="cb27-11" data-line-number="11">  g &lt;-<span class="st"> </span><span class="kw">ggplot</span>(compare_adjalley,<span class="kw">aes</span>(position<span class="op">-</span>start)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_histogram</span>(<span class="dt">fill=</span><span class="st">&quot;#aec6cf&quot;</span>,<span class="dt">color=</span><span class="st">&quot;grey50&quot;</span>)</a>
<a class="sourceLine" id="cb27-12" data-line-number="12">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb27-13" data-line-number="13">}</a></code></pre></div>
<p><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/comparing_utrs.Rmd/unnamed-chunk-9-8.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">rm</span>(tmp1,tmp2)</a></code></pre></div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Gentoo/Linux

Matrix products: default
BLAS: /usr/local/lib64/R/lib/libRblas.so
LAPACK: /usr/local/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] bindrcpp_0.2.2                       
 [2] BSgenome.Pfalciparum.PlasmoDB.v24_1.0
 [3] BSgenome_1.48.0                      
 [4] rtracklayer_1.40.6                   
 [5] Biostrings_2.48.0                    
 [6] XVector_0.20.0                       
 [7] GenomicRanges_1.32.6                 
 [8] GenomeInfoDb_1.16.0                  
 [9] org.Pf.plasmo.db_3.6.0               
[10] AnnotationDbi_1.42.1                 
[11] IRanges_2.14.10                      
[12] S4Vectors_0.18.3                     
[13] Biobase_2.40.0                       
[14] BiocGenerics_0.26.0                  
[15] scales_1.0.0                         
[16] cowplot_0.9.3                        
[17] magrittr_1.5                         
[18] forcats_0.3.0                        
[19] stringr_1.3.1                        
[20] dplyr_0.7.6                          
[21] purrr_0.2.5                          
[22] readr_1.1.1                          
[23] tidyr_0.8.1                          
[24] tibble_1.4.2                         
[25] ggplot2_3.0.0                        
[26] tidyverse_1.2.1                      

loaded via a namespace (and not attached):
 [1] nlme_3.1-137                bitops_1.0-6               
 [3] matrixStats_0.54.0          lubridate_1.7.4            
 [5] bit64_0.9-7                 httr_1.3.1                 
 [7] rprojroot_1.3-2             tools_3.5.0                
 [9] backports_1.1.2             R6_2.2.2                   
[11] DBI_1.0.0                   lazyeval_0.2.1             
[13] colorspace_1.3-2            withr_2.1.2                
[15] tidyselect_0.2.4            bit_1.1-14                 
[17] compiler_3.5.0              git2r_0.23.0               
[19] cli_1.0.0                   rvest_0.3.2                
[21] xml2_1.2.0                  DelayedArray_0.6.5         
[23] labeling_0.3                digest_0.6.15              
[25] Rsamtools_1.32.3            rmarkdown_1.10             
[27] R.utils_2.6.0               pkgconfig_2.0.2            
[29] htmltools_0.3.6             rlang_0.2.2                
[31] readxl_1.1.0                rstudioapi_0.7             
[33] RSQLite_2.1.1               bindr_0.1.1                
[35] jsonlite_1.5                BiocParallel_1.14.2        
[37] R.oo_1.22.0                 RCurl_1.95-4.11            
[39] GenomeInfoDbData_1.1.0      Matrix_1.2-14              
[41] Rcpp_0.12.18                munsell_0.5.0              
[43] R.methodsS3_1.7.1           stringi_1.2.4              
[45] yaml_2.2.0                  SummarizedExperiment_1.10.1
[47] zlibbioc_1.26.0             plyr_1.8.4                 
[49] grid_3.5.0                  blob_1.1.1                 
[51] crayon_1.3.4                lattice_0.20-35            
[53] haven_1.1.2                 hms_0.4.2                  
[55] knitr_1.20                  pillar_1.3.0               
[57] XML_3.98-1.16               glue_1.3.0                 
[59] evaluate_0.11               modelr_0.1.2               
[61] cellranger_1.1.0            gtable_0.2.0               
[63] assertthat_0.2.0            broom_0.5.0                
[65] GenomicAlignments_1.16.0    memoise_1.1.0              
[67] workflowr_1.1.1            </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
