<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2018-09-25" />

<title>Make final UTRs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Make final UTRs</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2018-09-25</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-10-24</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/llinaslab/sanger_rnaseq/tree/dd9d56a8fbb07e41b641cd5e2fda6fe162519c0a" target="_blank">dd9d56a</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.httr-oauth
    Ignored:    code/.DS_Store
    Ignored:    code/differential_expression/
    Ignored:    code/differential_phase/
    Ignored:    data/
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    docs/figure/neighboring_genes.Rmd/.DS_Store
    Ignored:    output/ctss_clustering/
    Ignored:    output/differential_detection/
    Ignored:    output/differential_expression/
    Ignored:    output/differential_phase/
    Ignored:    output/extensive_transcription/
    Ignored:    output/final_utrs/
    Ignored:    output/gcbias/
    Ignored:    output/homopolymer_analysis/
    Ignored:    output/neighboring_genes/
    Ignored:    output/promoter_architecture/
    Ignored:    output/tfbs_analysis/
    Ignored:    output/transcript_abundance/

Untracked files:
    Untracked:  _workflowr.yml
    Untracked:  figures/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/analyze_neighboring_genes.Rmd
    Modified:   analysis/array_correlations.Rmd
    Modified:   analysis/calculate_transcript_abundance.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/comparing_utrs.Rmd
    Modified:   analysis/ctss_clustering.Rmd
    Modified:   analysis/dynamic_tss.Rmd
    Modified:   analysis/extensive_transcription.Rmd
    Modified:   analysis/final_utrs.Rmd
    Modified:   analysis/gcbias.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Modified:   analysis/process_neighboring_genes.Rmd
    Modified:   analysis/promoter_architecture.Rmd
    Modified:   analysis/strain_differential_detection.Rmd
    Modified:   analysis/strain_differential_expression.Rmd
    Modified:   analysis/strain_differential_phase.Rmd
    Modified:   analysis/tfbs_analysis.Rmd
    Deleted:    docs/Rplots.pdf

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/analysis/final_utrs.Rmd" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
<td style="text-align:left;">
added homopolymer analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/llinaslab/sanger_rnaseq/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/final_utrs.html" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
<td style="text-align:left;">
added homopolymer analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/1e6d9bb5d1c2fb8fe76c5503529cad7b349d237a/analysis/final_utrs.Rmd" target="_blank">1e6d9bb</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-24
</td>
<td style="text-align:left;">
comparing UTRs
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/analysis/final_utrs.Rmd" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
<td style="text-align:left;">
hellooooo
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/llinaslab/sanger_rnaseq/f59e2e352af3a95d97e00cda9084806c9d33bd03/docs/final_utrs.html" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
<td style="text-align:left;">
hellooooo
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<div id="making-finallongest-utrs" class="section level2">
<h2>Making final/longest UTRs</h2>
<p>There were several methods and data sets used to predict 5’ and 3’ UTRs.</p>
<p>The first method was using the RNA-seq data sets and continuous read coverage. Any continuous coverage found surrounding known transcripts that met a read threshold cutoff of 5 reads or more was added as a UTR to the transcript from which it extended depending on whether it was on the 5’ or 3’ end of the transcript. This was done for each strain inidividually.</p>
<p>The second method to predict 5’ UTRs was done by using a variant of nanoCAGE in order to tag the extreme ends of mRNA molecules and sequence them along with a synthetic oligo used for the pull down of these extreme ends. No 3’ UTRs of transcripts was predicted using this approach although both ends were tagged.</p>
<p>Finally, the 5’ mRNA capture data was used to correct 5’ UTR predictions in all strains IF there was coverage support from the RNA-seq data in that loci where the capture data predicted it to be. If there was no prediction based on the 5’ capture data, then coverage from the 3D7 strain was used to “repair” the 5’ UTRs as well.</p>
<p>These predictions were all combined into lists of “longest possible transcripts” for each strain. Priority was given to predictions:</p>
<ol style="list-style-type: decimal">
<li>5’ mRNA capture prediction</li>
<li>5’ mRNA capture prediction used to repair the coverage prediction</li>
<li>3D7 coverage repaired UTRs</li>
<li>Original RNA-seq UTR predictions</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># 3D7</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="ex">Rscript</span> scripts/create_final_utrs.R \</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">  ../data/utrs/repaired_cov_utrs/repaired_with_TSO_5UTRs/repaired_with_tso_5UTRs.3d7.filtered.rescued_5utrs.gff \</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">  ../data/utrs/repaired_cov_utrs/repaired_with_3D7_coverage_5UTRs/repaired_with_3d7_cov_5UTRs.3d7.filtered.rescued_5utrs.gff \</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">  ../data/utrs/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff \</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">  ../data/utrs/original_utrs/tso_thr5.gff \</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">  ../data/utrs/original_utrs/final.3utrs.3d7.3d7_v3_chr.idc.gff \</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">  TRUE \</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">  ../output/final_utrs/final_utrs_3d7.gff</a>
<a class="sourceLine" id="cb1-10" data-line-number="10">  </a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co"># HB3</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="ex">Rscript</span> scripts/create_final_utrs.R \</a>
<a class="sourceLine" id="cb1-13" data-line-number="13">  ../data/utrs/repaired_cov_utrs/repaired_with_TSO_5UTRs/repaired_with_tso_5UTRs.hb3.filtered.rescued_5utrs.gff \</a>
<a class="sourceLine" id="cb1-14" data-line-number="14">  ../data/utrs/repaired_cov_utrs/repaired_with_3D7_coverage_5UTRs/repaired_with_3d7_cov_5UTRs.hb3.filtered.rescued_5utrs.gff \</a>
<a class="sourceLine" id="cb1-15" data-line-number="15">  ../data/utrs/original_utrs/final.5utrs.hb3.3d7_v3_chr.idc.gff \</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">  ../data/utrs/original_utrs/tso_thr5.gff \</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">  ../data/utrs/original_utrs/final.3utrs.hb3.3d7_v3_chr.idc.gff \</a>
<a class="sourceLine" id="cb1-18" data-line-number="18">  TRUE \</a>
<a class="sourceLine" id="cb1-19" data-line-number="19">  ../output/final_utrs/final_utrs_hb3.gff</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#IT</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="ex">Rscript</span> scripts/create_final_utrs.R \</a>
<a class="sourceLine" id="cb1-23" data-line-number="23">  ../data/utrs/repaired_cov_utrs/repaired_with_TSO_5UTRs/repaired_with_tso_5UTRs.it.filtered.rescued_5utrs.gff \</a>
<a class="sourceLine" id="cb1-24" data-line-number="24">  ../data/utrs/repaired_cov_utrs/repaired_with_3D7_coverage_5UTRs/repaired_with_3d7_cov_5UTRs.it.filtered.rescued_5utrs.gff \</a>
<a class="sourceLine" id="cb1-25" data-line-number="25">  ../data/utrs/original_utrs/final.5utrs.it.3d7_v3_chr.idc.gff \</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">  ../data/utrs/original_utrs/tso_thr5.gff \</a>
<a class="sourceLine" id="cb1-27" data-line-number="27">  ../data/utrs/original_utrs/final.3utrs.it.3d7_v3_chr.idc.gff \</a>
<a class="sourceLine" id="cb1-28" data-line-number="28">  TRUE \</a>
<a class="sourceLine" id="cb1-29" data-line-number="29">  ../output/final_utrs/final_utrs_it.gff</a></code></pre></div>
<p>Finally, we should read in the resulting data:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># function to read in the data</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">read_utrs &lt;-<span class="st"> </span><span class="cf">function</span>(file, input_strain) {</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  df &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(file)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(Parent,type,width)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  df<span class="op">$</span>strain &lt;-<span class="st"> </span>input_strain</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  df<span class="op">$</span>Parent &lt;-<span class="st"> </span><span class="kw">as.character</span>(df<span class="op">$</span>Parent)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">  <span class="kw">return</span>(df)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"></a>
<a class="sourceLine" id="cb2-10" data-line-number="10">utrs3d7 &lt;-<span class="st"> </span><span class="kw">read_utrs</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_3d7_plasmodb_compatible.gff&quot;</span>, <span class="st">&quot;3D7&quot;</span>)</a>
<a class="sourceLine" id="cb2-11" data-line-number="11">utrshb3 &lt;-<span class="st"> </span><span class="kw">read_utrs</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_hb3_plasmodb_compatible.gff&quot;</span>, <span class="st">&quot;HB3&quot;</span>)</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">utrsit &lt;-<span class="st"> </span><span class="kw">read_utrs</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_it_plasmodb_compatible.gff&quot;</span>, <span class="st">&quot;IT&quot;</span>)</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">utrs &lt;-<span class="st"> </span><span class="kw">rbind</span>(utrs3d7,utrshb3,utrsit)</a></code></pre></div>
</div>
<div id="utrs" class="section level2">
<h2>5’ UTRs</h2>
<p>What does the distribution of 5’ UTRs look like?</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">utrs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;5UTR&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>width)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey90&quot;</span>,<span class="dt">bins=</span><span class="dv">30</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="st">  </span><span class="kw">facet_grid</span>(.<span class="op">~</span>strain) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">55</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/5utrs-hist-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of 5utrs-hist-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/5utrs-hist-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/docs/figure/final_utrs.Rmd/5utrs-hist-1.png" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">utrs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;5UTR&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>strain,<span class="dt">y=</span>width)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">fill=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/5utrs-boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of 5utrs-boxplots-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/5utrs-boxplots-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/docs/figure/final_utrs.Rmd/5utrs-boxplots-1.png" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
</tr>
</tbody>
</table>
</details>
<p>How do summary statistics vary between strains?</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">utrs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;5UTR&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(strain) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">mean=</span><span class="kw">mean</span>(width),<span class="dt">median=</span><span class="kw">median</span>(width),<span class="dt">sd=</span><span class="kw">sd</span>(width))</a></code></pre></div>
<pre><code># A tibble: 3 x 4
  strain  mean median    sd
  &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 3D7     663.    507  591.
2 HB3     541.    438  458.
3 IT      541.    428  499.</code></pre>
</div>
<div id="utrs-1" class="section level2">
<h2>3’ UTRs</h2>
<p>What does the distribution of 3’ UTRs look like?</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">utrs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;3UTR&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>width)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey90&quot;</span>,<span class="dt">bins=</span><span class="dv">30</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="st">  </span><span class="kw">facet_grid</span>(.<span class="op">~</span>strain) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">55</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/3utrs-hist-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of 3utrs-hist-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/3utrs-hist-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/docs/figure/final_utrs.Rmd/3utrs-hist-1.png" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">utrs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;3UTR&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>strain,<span class="dt">y=</span>width)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">fill=</span><span class="st">&quot;grey70&quot;</span>)</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/3utrs-boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of 3utrs-boxplots-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/3utrs-boxplots-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/docs/figure/final_utrs.Rmd/3utrs-boxplots-1.png" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
</tr>
</tbody>
</table>
</details>
<p>How do summary statistics vary between strains?</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">utrs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type <span class="op">==</span><span class="st"> &quot;3UTR&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(strain) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">mean=</span><span class="kw">mean</span>(width),<span class="dt">median=</span><span class="kw">median</span>(width),<span class="dt">sd=</span><span class="kw">sd</span>(width))</a></code></pre></div>
<pre><code># A tibble: 3 x 4
  strain  mean median    sd
  &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 3D7     454.   400.  325.
2 HB3     411.   356   330.
3 IT      335.   288   284.</code></pre>
</div>
<div id="homopolymer-analysis" class="section level2">
<h2>Homopolymer analysis</h2>
<p>One thing we were also interested in looking at was why the coverage UTRs some times suffered from “coverage drops”. If we assume that the AT content of the genome isn’t as much of an issue anymore now that we are using PCR-free library preparation and better priming of RNA molecules, then one other issue that could be causing these coverage drops are homopolymer tracts. Another reason we are making this assumption is because the CAGE tags support this conclusion. Where the 5’ UTR often falls short, there is a lack of supporting CAGE tags. If we look upstream, however, we can see CAGE tags AND coverage if we look at the DAFT-seq data.</p>
<p>In the current analysis, I’m forgetting to take into account introns when looking for the presence of long homopolymer tracts within predicted UTRs. Important questions include:</p>
<ol style="list-style-type: decimal">
<li>Do long homopolymer tracts exist within prediced coverage UTRs?</li>
<li>Are UTRs containing long homopolymer tracts on average shorter than those that don’t?</li>
</ol>
<p>To address these questions, I extended 5’ UTRs 100 base pairs beyond their predicted coverage-based TSS, counted the length of the longest homopolymer tract for each nucleotide, and looked whether longer homopolymer tracts tend to occur in UTRs that are shorter.</p>
<p>First let’s make 3D7 5UTR sequences for the analysis:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="fu">cat</span> ../data/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff <span class="kw">|</span> </a>
<a class="sourceLine" id="cb11-2" data-line-number="2">  <span class="fu">awk</span> <span class="st">&#39;BEGIN {{OFS=FS=\&quot;\t\&quot;}} {{split($9,X,\&quot;;\&quot;); split(X[1],Y,\&quot;=\&quot;); print $1,$4-1,$5,Y[2],$5-$4+1,$7}}&#39;</span> <span class="kw">|</span> </a>
<a class="sourceLine" id="cb11-3" data-line-number="3">  <span class="ex">bedtools</span> getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed - -fo output/homopolymer_analysis/3d7_5utrs.fasta -s -name</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">  </a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="ex">bedtools</span> flank -i ../data/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff -g ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.lengths -s -l 100 -r 0 <span class="kw">|</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6">  <span class="fu">awk</span> <span class="st">&#39;BEGIN {{OFS=FS=\&quot;\t\&quot;}} {{split($9,X,\&quot;;\&quot;); split(X[1],Y,\&quot;=\&quot;); print $1,$4-1,$5,Y[2],$5-$4+1,$7}}&#39;</span> <span class="kw">|</span> </a>
<a class="sourceLine" id="cb11-7" data-line-number="7">  <span class="ex">bedtools</span> getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed - -fo ../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs.fasta -s -name</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">  </a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="ex">bedtools</span> slop -i ../data/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff -g ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.lengths -s -l 100 -r 0 <span class="kw">|</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10">  <span class="fu">awk</span> <span class="st">&#39;BEGIN {{OFS=FS=\&quot;\t\&quot;}} {{split($9,X,\&quot;;\&quot;); split(X[1],Y,\&quot;=\&quot;); print $1,$4-1,$5,Y[2],$5-$4+1,$7}}&#39;</span> <span class="kw">|</span> </a>
<a class="sourceLine" id="cb11-11" data-line-number="11">  <span class="ex">bedtools</span> getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed - -fo ../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs.fasta -s -name</a></code></pre></div>
<p>Now let’s count the number of homopolymers:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="ex">Rscript</span> ../code/scripts/final_utrs/count_homopolymers.R -f ../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs.fasta -l 50 -o ../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs_counts.tsv</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="ex">Rscript</span> ../code/scripts/final_utrs/count_homopolymers.R -f ../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs.fasta -l 50 -o ../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs_counts.tsv</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="ex">Rscript</span> ../code/scripts/final_utrs/count_homopolymers.R -f ../output/homopolymer_analysis/3d7_5utrs.fasta -l 50 -o ../output/homopolymer_analysis/3d7_5utrs.fasta_counts.tsv</a></code></pre></div>
<p>Now import that data:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">utrplus &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs_counts.tsv&quot;</span>) <span class="co"># UTR plus 100bp upstream</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">plus&lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs_counts.tsv&quot;</span>) <span class="co"># only 100 bp upstream of UTR</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">utrs &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/homopolymer_analysis/3d7_5utrs_counts.tsv&quot;</span>) <span class="co"># only UTR</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"></a>
<a class="sourceLine" id="cb13-5" data-line-number="5">originalutrs &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/utrs/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff&quot;</span>))</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">originalutrs<span class="op">$</span>Parent &lt;-<span class="st"> </span><span class="kw">as.character</span>(originalutrs<span class="op">$</span>Parent)</a>
<a class="sourceLine" id="cb13-7" data-line-number="7"></a>
<a class="sourceLine" id="cb13-8" data-line-number="8">fixedutrs &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/final_utrs_3d7.gff&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(length_added <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb13-10" data-line-number="10">fixedutrs<span class="op">$</span>Parent &lt;-<span class="st"> </span><span class="kw">as.character</span>(fixedutrs<span class="op">$</span>Parent)</a></code></pre></div>
<p>And now plot the data:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">tmp &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(originalutrs, utrplus, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;Parent&quot;</span> =<span class="st"> &quot;gene_id&quot;</span>))</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"></a>
<a class="sourceLine" id="cb14-3" data-line-number="3">g &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y=</span>width<span class="op">/</span>longest,<span class="dt">x=</span>longest)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.25</span>,<span class="dt">color=</span><span class="st">&quot;black&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="st">  </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>,<span class="dt">color=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Normalized Length&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Longest Homopolymer&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;All UTRs Plus 100 bps&quot;</span>)</a>
<a class="sourceLine" id="cb14-12" data-line-number="12"></a>
<a class="sourceLine" id="cb14-13" data-line-number="13">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/utrplus_homopolymers.png&quot;</span>,g)</a>
<a class="sourceLine" id="cb14-14" data-line-number="14">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/utrplus_homopolymers.svg&quot;</span>,g)</a>
<a class="sourceLine" id="cb14-15" data-line-number="15"></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-6-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/unnamed-chunk-6-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">cor.test</span>(.<span class="op">$</span>width<span class="op">/</span>.<span class="op">$</span>longest,.<span class="op">$</span>longest)))</a></code></pre></div>
<pre><code># A tibble: 1 x 8
  estimate statistic   p.value parameter conf.low conf.high method
     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;int&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt; 
1   -0.335     -23.5 2.33e-115      4386   -0.361    -0.308 Pears…
# ... with 1 more variable: alternative &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">tmp &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(originalutrs, plus, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;Parent&quot;</span> =<span class="st"> &quot;gene_id&quot;</span>))</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"></a>
<a class="sourceLine" id="cb17-3" data-line-number="3">g &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(width <span class="op">&lt;</span><span class="st"> </span><span class="dv">8000</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y=</span>width,<span class="dt">x=</span>longest)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>)</a>
<a class="sourceLine" id="cb17-8" data-line-number="8"></a>
<a class="sourceLine" id="cb17-9" data-line-number="9">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/plus_homopolymers.png&quot;</span>,g)</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/plus_homopolymers.svg&quot;</span>,g)</a>
<a class="sourceLine" id="cb17-11" data-line-number="11"></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/unnamed-chunk-7-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="st">  </span><span class="co">#dplyr::filter(width &lt; 8000) %&gt;%</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">cor.test</span>(.<span class="op">$</span>width,.<span class="op">$</span>longest)))</a></code></pre></div>
<pre><code># A tibble: 1 x 8
  estimate statistic  p.value parameter conf.low conf.high method
     &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;     &lt;int&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt; 
1   -0.174     -11.7 4.58e-31      4386   -0.202    -0.145 Pears…
# ... with 1 more variable: alternative &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">tmp &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(originalutrs, utrs, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;Parent&quot;</span> =<span class="st"> &quot;gene_id&quot;</span>))</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"></a>
<a class="sourceLine" id="cb20-3" data-line-number="3">g &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(width <span class="op">&lt;</span><span class="st"> </span><span class="dv">8000</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y=</span>width<span class="op">/</span>longest,<span class="dt">x=</span>longest)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>)</a>
<a class="sourceLine" id="cb20-8" data-line-number="8"></a>
<a class="sourceLine" id="cb20-9" data-line-number="9">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/utr_homopolymers.png&quot;</span>,g)</a>
<a class="sourceLine" id="cb20-10" data-line-number="10">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/utr_homopolymers.svg&quot;</span>,g)</a>
<a class="sourceLine" id="cb20-11" data-line-number="11"></a>
<a class="sourceLine" id="cb20-12" data-line-number="12"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-8-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/fa4fca88e2750a4c895ff6d85657879ceeb1b649/docs/figure/final_utrs.Rmd/unnamed-chunk-8-1.png" target="_blank">fa4fca8</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-25
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="st">  </span><span class="co">#dplyr::filter(width &lt; 8000) %&gt;%</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">cor.test</span>(.<span class="op">$</span>width<span class="op">/</span>.<span class="op">$</span>longest,.<span class="op">$</span>longest)))</a></code></pre></div>
<pre><code># A tibble: 1 x 8
  estimate statistic p.value parameter conf.low conf.high method
     &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;     &lt;int&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt; 
1   0.0662      4.39 1.15e-5      4383   0.0367    0.0956 Pears…
# ... with 1 more variable: alternative &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">tmp &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(fixedutrs, utrplus, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;Parent&quot;</span> =<span class="st"> &quot;gene_id&quot;</span>))</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"></a>
<a class="sourceLine" id="cb23-3" data-line-number="3">g &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y=</span>width<span class="op">/</span>longest,<span class="dt">x=</span>longest)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha=</span><span class="fl">0.25</span>,<span class="dt">color=</span><span class="st">&quot;black&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="st">  </span><span class="kw">stat_smooth</span>(<span class="dt">method=</span><span class="st">&quot;lm&quot;</span>,<span class="dt">color=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Normalized Length&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Longest Homopolymer&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-11" data-line-number="11"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Fixed UTRs&quot;</span>)</a>
<a class="sourceLine" id="cb23-12" data-line-number="12"></a>
<a class="sourceLine" id="cb23-13" data-line-number="13">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/fixedutr_homopolymers.png&quot;</span>,g)</a>
<a class="sourceLine" id="cb23-14" data-line-number="14">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="st">&quot;../output/final_utrs/fixedutr_homopolymers.svg&quot;</span>,g)</a>
<a class="sourceLine" id="cb23-15" data-line-number="15"></a>
<a class="sourceLine" id="cb23-16" data-line-number="16"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/final_utrs.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">tmp <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(Parent) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">width=</span><span class="kw">unique</span>(width),<span class="dt">longest=</span><span class="kw">max</span>(longest)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">cor.test</span>(.<span class="op">$</span>width<span class="op">/</span>.<span class="op">$</span>longest,.<span class="op">$</span>longest)))</a></code></pre></div>
<pre><code># A tibble: 1 x 8
  estimate statistic  p.value parameter conf.low conf.high method
     &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;     &lt;int&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt; 
1   -0.320     -16.2 4.10e-56      2299   -0.357    -0.283 Pears…
# ... with 1 more variable: alternative &lt;chr&gt;</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Gentoo/Linux

Matrix products: default
BLAS: /usr/local/lib64/R/lib/libRblas.so
LAPACK: /usr/local/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gdtools_0.1.7                        
 [2] bindrcpp_0.2.2                       
 [3] BSgenome.Pfalciparum.PlasmoDB.v24_1.0
 [4] BSgenome_1.48.0                      
 [5] rtracklayer_1.40.6                   
 [6] Biostrings_2.48.0                    
 [7] XVector_0.20.0                       
 [8] GenomicRanges_1.32.7                 
 [9] GenomeInfoDb_1.16.0                  
[10] org.Pf.plasmo.db_3.6.0               
[11] AnnotationDbi_1.42.1                 
[12] IRanges_2.14.12                      
[13] S4Vectors_0.18.3                     
[14] Biobase_2.40.0                       
[15] BiocGenerics_0.26.0                  
[16] scales_1.0.0                         
[17] cowplot_0.9.3                        
[18] magrittr_1.5                         
[19] forcats_0.3.0                        
[20] stringr_1.3.1                        
[21] dplyr_0.7.6                          
[22] purrr_0.2.5                          
[23] readr_1.1.1                          
[24] tidyr_0.8.1                          
[25] tibble_1.4.2                         
[26] ggplot2_3.0.0                        
[27] tidyverse_1.2.1                      

loaded via a namespace (and not attached):
 [1] nlme_3.1-137                bitops_1.0-6               
 [3] matrixStats_0.54.0          lubridate_1.7.4            
 [5] bit64_0.9-7                 httr_1.3.1                 
 [7] rprojroot_1.3-2             tools_3.5.0                
 [9] backports_1.1.2             utf8_1.1.4                 
[11] R6_2.3.0                    DBI_1.0.0                  
[13] lazyeval_0.2.1              colorspace_1.3-2           
[15] withr_2.1.2                 tidyselect_0.2.4           
[17] bit_1.1-14                  compiler_3.5.0             
[19] git2r_0.23.0                cli_1.0.1                  
[21] rvest_0.3.2                 xml2_1.2.0                 
[23] DelayedArray_0.6.6          labeling_0.3               
[25] digest_0.6.17               Rsamtools_1.32.3           
[27] svglite_1.2.1               rmarkdown_1.10             
[29] R.utils_2.7.0               pkgconfig_2.0.2            
[31] htmltools_0.3.6             rlang_0.2.2                
[33] readxl_1.1.0                rstudioapi_0.8             
[35] RSQLite_2.1.1               bindr_0.1.1                
[37] jsonlite_1.5                BiocParallel_1.14.2        
[39] R.oo_1.22.0                 RCurl_1.95-4.11            
[41] GenomeInfoDbData_1.1.0      Matrix_1.2-14              
[43] fansi_0.3.0                 Rcpp_0.12.19               
[45] munsell_0.5.0               R.methodsS3_1.7.1          
[47] stringi_1.2.4               whisker_0.3-2              
[49] yaml_2.2.0                  SummarizedExperiment_1.10.1
[51] zlibbioc_1.26.0             plyr_1.8.4                 
[53] grid_3.5.0                  blob_1.1.1                 
[55] crayon_1.3.4                lattice_0.20-35            
[57] haven_1.1.2                 hms_0.4.2                  
[59] knitr_1.20                  pillar_1.3.0               
[61] reshape2_1.4.3              XML_3.98-1.16              
[63] glue_1.3.0                  evaluate_0.11              
[65] modelr_0.1.2                cellranger_1.1.0           
[67] gtable_0.2.0                assertthat_0.2.0           
[69] broom_0.5.0                 GenomicAlignments_1.16.0   
[71] memoise_1.1.0               workflowr_1.1.1            </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
