<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2017-03-08" />

<title>Make final UTRs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Make final UTRs</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2017-03-08</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-03-30</p>
<p><strong>Code version:</strong> 0e2e80a</p>
<div id="making-finallongest-utrs" class="section level2">
<h2>Making final/longest UTRs</h2>
<p>There were several methods and data sets used to predict 5’ and 3’ UTRs.</p>
<p>The first method was using the RNA-seq data sets and continuous read coverage. Any continuous coverage found surrounding known transcripts that met a read threshold cutoff of 5 reads or more was added as a UTR to the transcript from which it extended depending on whether it was on the 5’ or 3’ end of the transcript. This was done for each strain inidividually.</p>
<p>The second method to predict 5’ UTRs was done by using a variant of nanoCAGE in order to tag the extreme ends of mRNA molecules and sequence them along with a synthetic oligo used for the pull down of these extreme ends. No 3’ UTRs of transcripts was predicted using this approach although both ends were tagged.</p>
<p>Finally, the 5’ mRNA capture data was used to correct 5’ UTR predictions in all strains IF there was coverage support from the RNA-seq data in that loci where the capture data predicted it to be. If there was no prediction based on the 5’ capture data, then coverage from the 3D7 strain was used to “repair” the 5’ UTRs as well.</p>
<p>These predictions were all combined into lists of “longest possible transcripts” for each strain. Priority was given to predictions:</p>
<ol style="list-style-type: decimal">
<li>5’ mRNA capture prediction</li>
<li>5’ mRNA capture prediction used to repair the coverage prediction</li>
<li>3D7 coverage repaired UTRs</li>
<li>Original RNA-seq UTR predictions</li>
</ol>
<p>First we should read in the resulting data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># function to read in the data</span>
read_utrs &lt;-<span class="st"> </span>function(file, input_strain) {
  df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(rtracklayer::<span class="kw">import.gff3</span>(file)) %&gt;%
<span class="st">    </span>dplyr::<span class="kw">select</span>(Parent,type,width)
  df$strain &lt;-<span class="st"> </span>input_strain
  df$Parent &lt;-<span class="st"> </span><span class="kw">as.character</span>(df$Parent)
  <span class="kw">return</span>(df)
}

utrs3d7 &lt;-<span class="st"> </span><span class="kw">read_utrs</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_3d7_plasmodb_compatible.gff&quot;</span>, <span class="st">&quot;3D7&quot;</span>)
utrshb3 &lt;-<span class="st"> </span><span class="kw">read_utrs</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_hb3_plasmodb_compatible.gff&quot;</span>, <span class="st">&quot;HB3&quot;</span>)
utrsit &lt;-<span class="st"> </span><span class="kw">read_utrs</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_it_plasmodb_compatible.gff&quot;</span>, <span class="st">&quot;IT&quot;</span>)
utrs &lt;-<span class="st"> </span><span class="kw">rbind</span>(utrs3d7,utrshb3,utrsit)</code></pre></div>
</div>
<div id="utrs" class="section level2">
<h2>5’ UTRs</h2>
<p>What does the distribution of 5’ UTRs look like?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">utrs %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">filter</span>(type ==<span class="st"> &quot;5UTR&quot;</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>width)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey90&quot;</span>,<span class="dt">bins=</span><span class="dv">30</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">facet_grid</span>(.~strain) +
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">55</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</code></pre></div>
<p><img src="figure/final_utrs.Rmd/5utrs-hist-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">utrs %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">filter</span>(type ==<span class="st"> &quot;5UTR&quot;</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>strain,<span class="dt">y=</span>width)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">fill=</span><span class="st">&quot;grey70&quot;</span>)</code></pre></div>
<p><img src="figure/final_utrs.Rmd/5utrs-boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>How do summary statistics vary between strains?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">utrs %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">filter</span>(type ==<span class="st"> &quot;5UTR&quot;</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">group_by</span>(strain) %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">summarise</span>(<span class="dt">mean=</span><span class="kw">mean</span>(width),<span class="dt">median=</span><span class="kw">median</span>(width),<span class="dt">sd=</span><span class="kw">sd</span>(width))</code></pre></div>
<pre><code># A tibble: 3 × 4
  strain     mean median       sd
   &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;
1    3D7 662.6894    507 591.3160
2    HB3 540.8970    438 457.6042
3     IT 541.4937    428 498.9186</code></pre>
</div>
<div id="utrs-1" class="section level2">
<h2>3’ UTRs</h2>
<p>What does the distribution of 3’ UTRs look like?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">utrs %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">filter</span>(type ==<span class="st"> &quot;3UTR&quot;</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>width)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">color=</span><span class="st">&quot;grey90&quot;</span>,<span class="dt">bins=</span><span class="dv">30</span>) +<span class="st"> </span>
<span class="st">  </span><span class="kw">facet_grid</span>(.~strain) +<span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">55</span>, <span class="dt">hjust =</span> <span class="dv">1</span>))</code></pre></div>
<p><img src="figure/final_utrs.Rmd/3utrs-hist-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">utrs %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">filter</span>(type ==<span class="st"> &quot;3UTR&quot;</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>strain,<span class="dt">y=</span>width)) +<span class="st"> </span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">fill=</span><span class="st">&quot;grey70&quot;</span>)</code></pre></div>
<p><img src="figure/final_utrs.Rmd/3utrs-boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>How do summary statistics vary between strains?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">utrs %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">filter</span>(type ==<span class="st"> &quot;3UTR&quot;</span>) %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">group_by</span>(strain) %&gt;%<span class="st"> </span>
<span class="st">  </span>dplyr::<span class="kw">summarise</span>(<span class="dt">mean=</span><span class="kw">mean</span>(width),<span class="dt">median=</span><span class="kw">median</span>(width),<span class="dt">sd=</span><span class="kw">sd</span>(width))</code></pre></div>
<pre><code># A tibble: 3 × 4
  strain     mean median       sd
   &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;
1    3D7 453.9042  399.5 325.3769
2    HB3 411.2785  356.0 329.6666
3     IT 334.9430  288.0 283.9846</code></pre>
</div>
<div id="homopolymer-analysis" class="section level2">
<h2>Homopolymer analysis</h2>
<p>One thing we were also interested in looking at was why the coverage UTRs some times suffered from “coverage drops”. If we assume that the AT content of the genome isn’t as much of an issue anymore now that we are using PCR-free library preparation and better priming of RNA molecules, then one other issue that could be causing these coverage drops are homopolymer tracts. Another reason we are making this assumption is because the CAGE tags support this conclusion. Where the 5’ UTR often falls short, there is a lack of supporting CAGE tags. If we look upstream, however, we can see CAGE tags AND coverage if we look at the DAFT-seq data.</p>
<p>In the current analysis, I’m forgetting to take into account introns when looking for the presence of long homopolymer tracts within predicted UTRs. Important questions include:</p>
<ol style="list-style-type: decimal">
<li>Do long homopolymer tracts exist within prediced coverage UTRs?</li>
<li>Are UTRs containing long homopolymer tracts on average shorter than those that don’t?</li>
</ol>
<p>To address these questions, I extended 5’ and 3’ UTRs 100 base pairs beyond their predicted coverage TSS and TES, removed intron sequences, counted the number and length of homopolymer tracts, and compared whether longer homopolymer tracts tend to occur in UTRs that are shorter and vice versa.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] scales_0.4.1    cowplot_0.7.0   magrittr_1.5    stringr_1.2.0  
 [5] dplyr_0.5.0     purrr_0.2.2     readr_1.0.0     tidyr_0.6.1    
 [9] tibble_1.2      ggplot2_2.2.1   tidyverse_1.1.1

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.4.0 reshape2_1.4.2            
 [3] haven_1.0.0                lattice_0.20-34           
 [5] colorspace_1.3-2           htmltools_0.3.5           
 [7] stats4_3.3.2               rtracklayer_1.34.2        
 [9] yaml_2.1.14                XML_3.98-1.5              
[11] foreign_0.8-67             DBI_0.5-1                 
[13] BiocParallel_1.8.1         BiocGenerics_0.20.0       
[15] modelr_0.1.0               readxl_0.1.1              
[17] plyr_1.8.4                 zlibbioc_1.20.0           
[19] Biostrings_2.42.1          workflowr_0.3.0           
[21] munsell_0.4.3              gtable_0.2.0              
[23] rvest_0.3.2                psych_1.6.12              
[25] evaluate_0.10              labeling_0.3              
[27] Biobase_2.34.0             knitr_1.15.1              
[29] forcats_0.2.0              IRanges_2.8.1             
[31] GenomeInfoDb_1.10.3        parallel_3.3.2            
[33] broom_0.4.2                Rcpp_0.12.9               
[35] backports_1.0.5            S4Vectors_0.12.1          
[37] jsonlite_1.3               XVector_0.14.0            
[39] Rsamtools_1.26.1           mnormt_1.5-5              
[41] hms_0.3                    digest_0.6.12             
[43] stringi_1.1.2              GenomicRanges_1.26.3      
[45] grid_3.3.2                 rprojroot_1.2             
[47] tools_3.3.2                bitops_1.0-6              
[49] RCurl_1.95-4.8             lazyeval_0.2.0            
[51] Matrix_1.2-8               xml2_1.1.1                
[53] lubridate_1.6.0            assertthat_0.1            
[55] rmarkdown_1.3              httr_1.2.1                
[57] R6_2.2.0                   git2r_0.18.0              
[59] GenomicAlignments_1.10.0   nlme_3.1-131              </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
