<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />


<title>Promoter Architecture</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Promoter Architecture</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>09-25-2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-11-20</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/llinaslab/sanger_rnaseq/tree/dd9d56a8fbb07e41b641cd5e2fda6fe162519c0a" target="_blank">dd9d56a</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.httr-oauth
    Ignored:    code/.DS_Store
    Ignored:    code/differential_expression/
    Ignored:    code/differential_phase/
    Ignored:    data/
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    docs/figure/neighboring_genes.Rmd/.DS_Store
    Ignored:    output/compare/
    Ignored:    output/ctss_clustering/
    Ignored:    output/differential_detection/
    Ignored:    output/differential_expression/
    Ignored:    output/differential_phase/
    Ignored:    output/extensive_transcription/
    Ignored:    output/final_utrs/
    Ignored:    output/gcbias/
    Ignored:    output/homopolymer_analysis/
    Ignored:    output/neighboring_genes/
    Ignored:    output/promoter_architecture/
    Ignored:    output/tfbs_analysis/
    Ignored:    output/transcript_abundance/

Untracked files:
    Untracked:  _workflowr.yml
    Untracked:  docs/figure/tfbs_analysis.Rmd/
    Untracked:  figures/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/analyze_neighboring_genes.Rmd
    Modified:   analysis/array_correlations.Rmd
    Modified:   analysis/calculate_transcript_abundance.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/comparing_utrs.Rmd
    Modified:   analysis/ctss_clustering.Rmd
    Modified:   analysis/dynamic_tss.Rmd
    Modified:   analysis/extensive_transcription.Rmd
    Modified:   analysis/final_utrs.Rmd
    Modified:   analysis/gcbias.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Modified:   analysis/process_neighboring_genes.Rmd
    Modified:   analysis/promoter_architecture.Rmd
    Modified:   analysis/strain_differential_detection.Rmd
    Modified:   analysis/strain_differential_expression.Rmd
    Modified:   analysis/strain_differential_phase.Rmd
    Modified:   analysis/tfbs_analysis.Rmd
    Modified:   code/differential_detection/detect_transcripts.R
    Modified:   code/figures.R
    Deleted:    docs/Rplots.pdf

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/analysis/promoter_architecture.Rmd" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
<td style="text-align:left;">
you’re what message
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/llinaslab/sanger_rnaseq/b1b89808d44aab66572a52174ef09c607bec1f18/docs/promoter_architecture.html" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
<td style="text-align:left;">
you’re what message
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/analysis/promoter_architecture.Rmd" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
<td style="text-align:left;">
hellooooo
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/llinaslab/sanger_rnaseq/f59e2e352af3a95d97e00cda9084806c9d33bd03/docs/promoter_architecture.html" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
<td style="text-align:left;">
hellooooo
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<div id="sequence-features-surrounding-transcription-start-sites" class="section level2">
<h2>Sequence features surrounding transcription start sites</h2>
<p>For this analysis we want to plot nucleotide frequencies up and downstream of predicted TSSs. We can do this by looking at most commonly used TSSs, tag clusters, and promoter clusters.</p>
<p>First we’ll import all the data:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># import genome</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(BSgenome.Pfalciparum.PlasmoDB.v24)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co"># import tag clusters</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">tc_intergenic &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/ctss_clustering/modified/tag_clusters_annotated_intergenic.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6">tc_exonic     &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/ctss_clustering/modified/tag_clusters_annotated_exons.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">tc_intronic   &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/ctss_clustering/modified/tag_clusters_annotated_introns.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co"># import promoter clusters</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10">pc_intergenic &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/ctss_clustering/modified/promoter_clusters_annotated_intergenic.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">pc_exonic     &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/ctss_clustering/modified/promoter_clusters_annotated_exons.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-12" data-line-number="12">pc_intronic   &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/ctss_clustering/modified/promoter_clusters_annotated_introns.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co"># import telomere ranges</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">telomeres  &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/annotations/Pf3D7_v3_subtelomeres.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16"></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co"># import genes as well</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18">genes &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/annotations/genes_nuclear_3D7_v24.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-19" data-line-number="19"></a>
<a class="sourceLine" id="cb1-20" data-line-number="20">distance_to_add &lt;-<span class="st"> </span><span class="dv">500</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co"># original TSO-predicted TSSs</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23">x3d7_tso &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/utrs/original_utrs/tso_thr5.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">+</span>distance_to_add,end<span class="op">+</span>distance_to_add),</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>distance_to_add,end<span class="op">-</span>distance_to_add)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-31" data-line-number="31"></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="co"># import most heavily used TSSs</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33">x3d7_tss &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/final_utrs_3d7.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">+</span>distance_to_add,end<span class="op">+</span>distance_to_add),</a>
<a class="sourceLine" id="cb1-36" data-line-number="36">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>distance_to_add,end<span class="op">-</span>distance_to_add)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-41" data-line-number="41"></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="co"># import TTS for 3D7</span></a>
<a class="sourceLine" id="cb1-43" data-line-number="43">x3d7_tts &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/final_utrs_3d7.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-44" data-line-number="44"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,end<span class="op">+</span>distance_to_add,start<span class="op">+</span>distance_to_add),</a>
<a class="sourceLine" id="cb1-46" data-line-number="46">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,end<span class="op">-</span>distance_to_add,start<span class="op">-</span>distance_to_add)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-47" data-line-number="47"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-48" data-line-number="48"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-49" data-line-number="49"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;3UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-50" data-line-number="50"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-51" data-line-number="51"></a>
<a class="sourceLine" id="cb1-52" data-line-number="52"><span class="co"># HB3</span></a>
<a class="sourceLine" id="cb1-53" data-line-number="53">xhb3_tss &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/final_utrs_hb3.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-54" data-line-number="54"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-55" data-line-number="55"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">+</span>distance_to_add,end<span class="op">+</span>distance_to_add),</a>
<a class="sourceLine" id="cb1-56" data-line-number="56">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>distance_to_add,end<span class="op">-</span>distance_to_add)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-57" data-line-number="57"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-58" data-line-number="58"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-60" data-line-number="60"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-61" data-line-number="61"></a>
<a class="sourceLine" id="cb1-62" data-line-number="62"><span class="co"># IT</span></a>
<a class="sourceLine" id="cb1-63" data-line-number="63">xit_tss &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/final_utrs_it.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-64" data-line-number="64"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-65" data-line-number="65"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">+</span>distance_to_add,end<span class="op">+</span>distance_to_add),</a>
<a class="sourceLine" id="cb1-66" data-line-number="66">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>distance_to_add,end<span class="op">-</span>distance_to_add)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-67" data-line-number="67"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-68" data-line-number="68"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-69" data-line-number="69"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-70" data-line-number="70"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a></code></pre></div>
<p>Let’s set the nucleotide colors to be what we want them to be:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># set colors</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">set_colors &lt;-<span class="st"> </span><span class="cf">function</span>(colors, vars, iname) {</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">  myColors &lt;-<span class="st"> </span><span class="kw">c</span>(colors)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">  <span class="kw">names</span>(myColors) &lt;-<span class="st"> </span><span class="kw">levels</span>(vars)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">  output &lt;-<span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">scale_colour_manual</span>(<span class="dt">name =</span> iname, <span class="dt">values =</span> myColors)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">  <span class="kw">return</span>(output)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">}</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># custom nucleotide colors</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">base_colors &lt;-<span class="st"> </span><span class="kw">set_colors</span>(<span class="kw">c</span>(<span class="st">&quot;#E41A1C&quot;</span>, <span class="st">&quot;#377EB8&quot;</span>, <span class="st">&quot;#F0E442&quot;</span>, <span class="st">&quot;#4DAF4A&quot;</span>), </a>
<a class="sourceLine" id="cb2-10" data-line-number="10">                          <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;G&quot;</span>), <span class="st">&quot;base&quot;</span>)</a></code></pre></div>
<p>Now let’s write the functions to generate the frequency diagrams, information content, and sequence logos:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Use this function to generate position weight matrices</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">generate_pwm &lt;-<span class="st"> </span><span class="cf">function</span>(clusters) {</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  <span class="co"># extract sequences from the genome</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">  seqs &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24,clusters)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">  <span class="co"># convert those sequences into a data frame</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">  tmp &lt;-<span class="st"> </span><span class="kw">lapply</span>(seqs,<span class="cf">function</span>(x) stringr<span class="op">::</span><span class="kw">str_split</span>(<span class="kw">as.character</span>(x),<span class="st">&quot;&quot;</span>)[[<span class="dv">1</span>]])</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">  tmp &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(tmp)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">  <span class="kw">colnames</span>(tmp) &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(tmp)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">  tmp<span class="op">$</span>pos &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(tmp)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">  tmp &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">gather</span>(seq, base, <span class="op">-</span>pos)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">  <span class="co"># calculate the proportion of nucleotides at each position</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12">  pwm &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(<span class="kw">as.numeric</span>(pos)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">A =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;A&quot;</span>)<span class="op">/</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb3-15" data-line-number="15">            <span class="dt">C =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;C&quot;</span>)<span class="op">/</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb3-16" data-line-number="16">            <span class="dt">G =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;G&quot;</span>)<span class="op">/</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb3-17" data-line-number="17">            <span class="dt">T =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;T&quot;</span>)<span class="op">/</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">  <span class="kw">colnames</span>(pwm)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;pos&quot;</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20">  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">pwm=</span>pwm,<span class="dt">seqs=</span>seqs))</a>
<a class="sourceLine" id="cb3-21" data-line-number="21">}</a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co"># Plot the nucleotide frequencies at each position</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23">plot_frequencies &lt;-<span class="st"> </span><span class="cf">function</span>(ipwm) {</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">  ipwm <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="st">    </span>tidyr<span class="op">::</span><span class="kw">gather</span>(base, freq, <span class="op">-</span>pos) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> pos, <span class="dt">y =</span> freq, <span class="dt">color =</span> base)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">size =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">1.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Frequency&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;top&quot;</span>, </a>
<a class="sourceLine" id="cb3-33" data-line-number="33">    <span class="dt">legend.direction=</span><span class="st">&quot;horizontal&quot;</span>,</a>
<a class="sourceLine" id="cb3-34" data-line-number="34">    <span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="st">    </span>base_colors</a>
<a class="sourceLine" id="cb3-36" data-line-number="36">}</a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co"># Plot the sequence logo</span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38">plot_logo &lt;-<span class="st"> </span><span class="cf">function</span>(ipwm,limits) {</a>
<a class="sourceLine" id="cb3-39" data-line-number="39">  logo &lt;-<span class="st"> </span>ipwm[,<span class="dv">2</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb3-40" data-line-number="40">  seqLogo<span class="op">::</span><span class="kw">seqLogo</span>(seqLogo<span class="op">::</span><span class="kw">makePWM</span>((<span class="kw">t</span>(logo[limits[<span class="dv">1</span>]<span class="op">:</span>limits[<span class="dv">2</span>],]))),<span class="dt">ic.scale =</span> T)</a>
<a class="sourceLine" id="cb3-41" data-line-number="41">}</a>
<a class="sourceLine" id="cb3-42" data-line-number="42"><span class="co"># Plot the information content at each position</span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43">plot_info &lt;-<span class="st"> </span><span class="cf">function</span>(ipwm) {</a>
<a class="sourceLine" id="cb3-44" data-line-number="44">  ipwm <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-45" data-line-number="45"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">i =</span> (A <span class="op">*</span><span class="st"> </span><span class="kw">log2</span>(A<span class="op">/</span><span class="fl">0.42</span>)) <span class="op">+</span><span class="st"> </span>(T <span class="op">*</span><span class="st"> </span><span class="kw">log2</span>(T<span class="op">/</span><span class="fl">0.45</span>)) <span class="op">+</span><span class="st"> </span>(G <span class="op">*</span><span class="st"> </span><span class="kw">log2</span>(G<span class="op">/</span><span class="fl">0.07</span>)) <span class="op">+</span><span class="st"> </span>(C <span class="op">*</span><span class="st"> </span><span class="kw">log2</span>(C<span class="op">/</span><span class="fl">0.06</span>))) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-46" data-line-number="46"><span class="st">    </span><span class="kw">gather</span>(base, freq, <span class="op">-</span>pos, <span class="op">-</span>i) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> pos, <span class="dt">y =</span> i)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="st">    </span><span class="kw">geom_line</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Information&quot;</span>)</a>
<a class="sourceLine" id="cb3-51" data-line-number="51">}</a></code></pre></div>
<p>And now we can start making some plots.</p>
<div id="most-commonly-used-5-tss" class="section level3">
<h3>Most commonly used 5’ TSS</h3>
<p>First, we can look at the most commonly used TSSs for each strain:</p>
<div id="d7" class="section level4">
<h4>3D7</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">x3d7pwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(x3d7_tss)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">plot_frequencies</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-5-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-5-1.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">plot_info</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">250</span>,<span class="dv">501</span>,<span class="dv">750</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-5-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-5-2.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">plot_logo</span>(x3d7pwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-5-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-5-3.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="hb3" class="section level4">
<h4>HB3</h4>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">xhb3pwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(xhb3_tss)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw">plot_frequencies</span>(xhb3pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">470</span>,<span class="dv">530</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-6-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-6-1.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">plot_info</span>(xhb3pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">251</span>,<span class="dv">501</span>,<span class="dv">751</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-6-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-6-2.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">plot_logo</span>(xhb3pwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-6-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-6-3.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="it" class="section level4">
<h4>IT</h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">xitpwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(xit_tss)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw">plot_frequencies</span>(xitpwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">470</span>,<span class="dv">530</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-7-1.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">plot_info</span>(xitpwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">251</span>,<span class="dv">501</span>,<span class="dv">751</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-7-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-7-2.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">plot_logo</span>(xitpwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-7-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-7-3.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
</div>
</div>
<div id="most-commonly-used-3-tts" class="section level3">
<h3>Most commonly used 3’ TTS</h3>
<p>We can also look at the most commonly used TTS in 3D7:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">x3d7pwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(x3d7_tts)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw">plot_frequencies</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">470</span>,<span class="dv">530</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">plot_info</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">251</span>,<span class="dv">501</span>,<span class="dv">751</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TTS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">plot_logo</span>(x3d7pwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can also compare this to random intergenic sequences:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">33</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">intergenic &lt;-<span class="st"> </span>GenomicRanges<span class="op">::</span><span class="kw">gaps</span>(genes)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">intergenic &lt;-<span class="st"> </span>intergenic[<span class="kw">is.na</span>(GenomicRanges<span class="op">::</span><span class="kw">findOverlaps</span>(intergenic,telomeres,<span class="dt">select=</span><span class="st">&quot;arbitrary&quot;</span>))]</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">intergenic_seqs &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24,intergenic)</a>
<a class="sourceLine" id="cb16-5" data-line-number="5"></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co"># create random seqs to compare to</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7">extract_random_seqs1 &lt;-<span class="st"> </span><span class="cf">function</span>(seqs,widths) {</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">  <span class="co"># start with the first sequence,</span></a>
<a class="sourceLine" id="cb16-9" data-line-number="9">  <span class="co"># filter by width to avoid errors,</span></a>
<a class="sourceLine" id="cb16-10" data-line-number="10">  <span class="co"># sample randomly from filtered sequences,</span></a>
<a class="sourceLine" id="cb16-11" data-line-number="11">  <span class="co"># grab random interval that matches the length of the</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12">  <span class="co"># promoter sequence</span></a>
<a class="sourceLine" id="cb16-13" data-line-number="13">  fseqs       &lt;-<span class="st"> </span>seqs[<span class="kw">width</span>(seqs) <span class="op">&gt;</span><span class="st"> </span>widths[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb16-14" data-line-number="14">  rsample     &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(fseqs),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb16-15" data-line-number="15">  rseq        &lt;-<span class="st"> </span>fseqs[rsample][[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb16-16" data-line-number="16">  rstart      &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(rseq)<span class="op">-</span>widths[<span class="dv">1</span>]),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb16-17" data-line-number="17">  random_seqs &lt;-<span class="st"> </span>rseq[rstart<span class="op">:</span>(rstart<span class="op">+</span>widths[<span class="dv">1</span>]<span class="op">-</span><span class="dv">1</span>)]</a>
<a class="sourceLine" id="cb16-18" data-line-number="18">  <span class="co"># do this for all sequences</span></a>
<a class="sourceLine" id="cb16-19" data-line-number="19">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">length</span>(widths)) {</a>
<a class="sourceLine" id="cb16-20" data-line-number="20">    fseqs       &lt;-<span class="st"> </span>seqs[<span class="kw">width</span>(seqs) <span class="op">&gt;</span><span class="st"> </span>widths[i]]</a>
<a class="sourceLine" id="cb16-21" data-line-number="21">    rsample     &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(fseqs),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb16-22" data-line-number="22">    rseq        &lt;-<span class="st"> </span>fseqs[rsample][[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb16-23" data-line-number="23">    rstart      &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(rseq)<span class="op">-</span>widths[i]<span class="op">-</span><span class="dv">1</span>),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb16-24" data-line-number="24">    random_seqs &lt;-<span class="st"> </span><span class="kw">unlist</span>(Biostrings<span class="op">::</span><span class="kw">DNAStringSetList</span>(</a>
<a class="sourceLine" id="cb16-25" data-line-number="25">      Biostrings<span class="op">::</span><span class="kw">DNAStringSet</span>(random_seqs),</a>
<a class="sourceLine" id="cb16-26" data-line-number="26">      Biostrings<span class="op">::</span><span class="kw">DNAStringSet</span>(rseq[rstart<span class="op">:</span>(rstart<span class="op">+</span>widths[i]<span class="op">-</span><span class="dv">1</span>)])))</a>
<a class="sourceLine" id="cb16-27" data-line-number="27">  }</a>
<a class="sourceLine" id="cb16-28" data-line-number="28">  <span class="kw">return</span>(random_seqs)</a>
<a class="sourceLine" id="cb16-29" data-line-number="29">}</a>
<a class="sourceLine" id="cb16-30" data-line-number="30"></a>
<a class="sourceLine" id="cb16-31" data-line-number="31">random_seqs &lt;-<span class="st"> </span><span class="kw">extract_random_seqs1</span>(intergenic_seqs,<span class="kw">rep</span>(<span class="dv">1000</span>,<span class="dv">1001</span>))</a>
<a class="sourceLine" id="cb16-32" data-line-number="32"></a>
<a class="sourceLine" id="cb16-33" data-line-number="33">generate_random_pwm &lt;-<span class="st"> </span><span class="cf">function</span>(seqs) {</a>
<a class="sourceLine" id="cb16-34" data-line-number="34">  <span class="co"># convert those sequences into a data frame</span></a>
<a class="sourceLine" id="cb16-35" data-line-number="35">  tmp &lt;-<span class="st"> </span><span class="kw">lapply</span>(seqs,<span class="cf">function</span>(x) stringr<span class="op">::</span><span class="kw">str_split</span>(<span class="kw">as.character</span>(x),<span class="st">&quot;&quot;</span>)[[<span class="dv">1</span>]])</a>
<a class="sourceLine" id="cb16-36" data-line-number="36">  tmp &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(tmp)</a>
<a class="sourceLine" id="cb16-37" data-line-number="37">  <span class="kw">colnames</span>(tmp) &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(tmp)</a>
<a class="sourceLine" id="cb16-38" data-line-number="38">  tmp<span class="op">$</span>pos &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(tmp)</a>
<a class="sourceLine" id="cb16-39" data-line-number="39">  tmp &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">gather</span>(seq, base, <span class="op">-</span>pos)</a>
<a class="sourceLine" id="cb16-40" data-line-number="40">  <span class="co"># calculate the proportion of nucleotides at each position</span></a>
<a class="sourceLine" id="cb16-41" data-line-number="41">  pwm &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-42" data-line-number="42"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(<span class="kw">as.numeric</span>(pos)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb16-43" data-line-number="43"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">A =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;A&quot;</span>)<span class="op">/</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb16-44" data-line-number="44">            <span class="dt">C =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;C&quot;</span>)<span class="op">/</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb16-45" data-line-number="45">            <span class="dt">G =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;G&quot;</span>)<span class="op">/</span><span class="kw">n</span>(), </a>
<a class="sourceLine" id="cb16-46" data-line-number="46">            <span class="dt">T =</span> <span class="kw">sum</span>(base <span class="op">==</span><span class="st"> &quot;T&quot;</span>)<span class="op">/</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb16-47" data-line-number="47"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>()</a>
<a class="sourceLine" id="cb16-48" data-line-number="48">  <span class="kw">colnames</span>(pwm)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;pos&quot;</span></a>
<a class="sourceLine" id="cb16-49" data-line-number="49">  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">pwm=</span>pwm,<span class="dt">seqs=</span>seqs))</a>
<a class="sourceLine" id="cb16-50" data-line-number="50">}</a></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">random_seqs &lt;-<span class="st"> </span>Biostrings<span class="op">::</span><span class="kw">readDNAStringSet</span>(<span class="dt">filepath=</span><span class="st">&quot;../output/promoter_architecture/random_intergenic_seqs.fasta&quot;</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">random_pwm &lt;-<span class="st"> </span><span class="kw">generate_random_pwm</span>(random_seqs)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw">plot_frequencies</span>(random_pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">501</span>,<span class="dv">520</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-20&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;20&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;Random&quot;</span>)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">plot_info</span>(random_pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">251</span>,<span class="dv">501</span>,<span class="dv">751</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">plot_logo</span>(random_pwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>And we can see that the signal sort of goes away. However, there also wasn’t a very strong signal to begin with.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">Biostrings<span class="op">::</span><span class="kw">writeXStringSet</span>(<span class="dt">x=</span>random_pwm<span class="op">$</span>seqs,<span class="dt">filepath=</span><span class="st">&quot;../output/promoter_architecture/random_intergenic_seqs.fasta&quot;</span>)</a></code></pre></div>
</div>
<div id="tag-clusters" class="section level3">
<h3>Tag clusters</h3>
<p>The tag clusters represent high resolution TSSs that are enriched for the 5’ end of transcripts. Thus these TSS predictions are likely more accurate and less biased.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">distance_to_add &lt;-<span class="st"> </span><span class="dv">500</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co"># Use this function to filter tag clusters by total TPM</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4">filter_tag_clusters &lt;-<span class="st"> </span><span class="cf">function</span>(tcs,tpm_threshold,width_threshold) {</a>
<a class="sourceLine" id="cb21-5" data-line-number="5">  broad_ftcs &lt;-<span class="st"> </span>tcs <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="st">    </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">as.numeric</span>(tpm.dominant_ctss)<span class="op">&gt;=</span>tpm_threshold <span class="op">&amp;</span><span class="st"> </span>interquantile_width <span class="op">&gt;=</span><span class="st"> </span>width_threshold) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(seqnames,dominant_ctss,strand,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">distinct</span>(seqnames,dominant_ctss,strand,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">start=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,</a>
<a class="sourceLine" id="cb21-11" data-line-number="11">                               <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">-</span>(distance_to_add<span class="dv">-1</span>),</a>
<a class="sourceLine" id="cb21-12" data-line-number="12">                               <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">-</span>(distance_to_add<span class="dv">-1</span>)),</a>
<a class="sourceLine" id="cb21-13" data-line-number="13">                 <span class="dt">end=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,</a>
<a class="sourceLine" id="cb21-14" data-line-number="14">                            <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">+</span>(distance_to_add<span class="op">+</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb21-15" data-line-number="15">                            <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">+</span>(distance_to_add<span class="op">+</span><span class="dv">1</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="st">    </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb21-17" data-line-number="17">  </a>
<a class="sourceLine" id="cb21-18" data-line-number="18">  sharp_ftcs &lt;-<span class="st"> </span>tcs <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-19" data-line-number="19"><span class="st">    </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-20" data-line-number="20"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">as.numeric</span>(tpm.dominant_ctss)<span class="op">&gt;=</span>tpm_threshold <span class="op">&amp;</span><span class="st"> </span>interquantile_width <span class="op">&lt;</span><span class="st"> </span>width_threshold) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-21" data-line-number="21"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(seqnames,dominant_ctss,strand,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-22" data-line-number="22"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">distinct</span>(seqnames,dominant_ctss,strand,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-23" data-line-number="23"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">start=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,</a>
<a class="sourceLine" id="cb21-24" data-line-number="24">                               <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">-</span>(distance_to_add<span class="dv">-1</span>),</a>
<a class="sourceLine" id="cb21-25" data-line-number="25">                               <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">-</span>(distance_to_add<span class="dv">-1</span>)),</a>
<a class="sourceLine" id="cb21-26" data-line-number="26">                 <span class="dt">end=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,</a>
<a class="sourceLine" id="cb21-27" data-line-number="27">                            <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">+</span>(distance_to_add<span class="op">+</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb21-28" data-line-number="28">                            <span class="kw">as.numeric</span>(dominant_ctss)<span class="op">+</span>(distance_to_add<span class="op">+</span><span class="dv">1</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-29" data-line-number="29"><span class="st">    </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb21-30" data-line-number="30">  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">sharp=</span>sharp_ftcs,<span class="dt">broad=</span>broad_ftcs))</a>
<a class="sourceLine" id="cb21-31" data-line-number="31">}</a></code></pre></div>
<div id="intergenic" class="section level4">
<h4>Intergenic</h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="co"># filter out clusters found in telomeres</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2">tc_intergenic &lt;-<span class="st"> </span>tc_intergenic[<span class="kw">is.na</span>(GenomicRanges<span class="op">::</span><span class="kw">findOverlaps</span>(tc_intergenic,telomeres,<span class="dt">select=</span><span class="st">&quot;arbitrary&quot;</span>))]</a>
<a class="sourceLine" id="cb22-3" data-line-number="3"></a>
<a class="sourceLine" id="cb22-4" data-line-number="4">ftcs &lt;-<span class="st"> </span><span class="kw">filter_tag_clusters</span>(tc_intergenic,<span class="dv">5</span>,<span class="dv">15</span>)</a>
<a class="sourceLine" id="cb22-5" data-line-number="5">ftcssharppwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(ftcs<span class="op">$</span>sharp)</a>
<a class="sourceLine" id="cb22-6" data-line-number="6">ftcsbroadpwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(ftcs<span class="op">$</span>broad)</a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="kw">plot_frequencies</span>(ftcssharppwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">501</span>,<span class="dv">520</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-20&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;20&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-13-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-13-1.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw">plot_frequencies</span>(ftcsbroadpwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">501</span>,<span class="dv">520</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-20&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;20&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-13-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-13-2.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">plot_info</span>(ftcssharppwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">250</span>,<span class="dv">501</span>,<span class="dv">750</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-13-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-13-3.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">plot_info</span>(ftcsbroadpwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">250</span>,<span class="dv">501</span>,<span class="dv">750</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-13-4.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw">plot_logo</span>(ftcssharppwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-13-5.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw">plot_logo</span>(ftcsbroadpwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-13-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>This is really interesting! Do we see the same thing if we look at the original TSO-predicted TSSs using a more biased approach? This method looked a certain distance upstream of every translation start site, used a 5 read threshold to discard positions that were not able to distinguish signal from noise, and designated the position with the greatest coverage upstream of an annotated protein-coding gene as the TSS.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">x3d7pwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(x3d7_tso)</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="kw">plot_frequencies</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">501</span>,<span class="dv">520</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-20&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;20&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-14-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-14-1.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw">plot_frequencies</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">400</span>,<span class="dv">600</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">400</span>,<span class="dv">501</span>,<span class="dv">600</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-100&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;100&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-14-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-14-2.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw">plot_info</span>(x3d7pwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">250</span>,<span class="dv">501</span>,<span class="dv">750</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-14-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-14-3.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw">plot_logo</span>(x3d7pwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-14-4.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-14-4.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="exonic" class="section level4">
<h4>Exonic</h4>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">fetcs &lt;-<span class="st"> </span><span class="kw">filter_tag_clusters</span>(tc_exonic,<span class="dv">5</span>,<span class="dv">15</span>)</a>
<a class="sourceLine" id="cb32-2" data-line-number="2">fetcspwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(<span class="kw">unlist</span>(GenomicRanges<span class="op">::</span><span class="kw">GRangesList</span>(fetcs<span class="op">$</span>sharp,fetcs<span class="op">$</span>broad)))</a>
<a class="sourceLine" id="cb32-3" data-line-number="3"><span class="kw">plot_frequencies</span>(fetcspwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">501</span>,<span class="dv">520</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-20&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;20&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-15-1.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-15-1.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw">plot_info</span>(fetcspwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">250</span>,<span class="dv">501</span>,<span class="dv">750</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-15-2.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-15-2.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">plot_logo</span>(fetcspwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<details>
<summary><em>Expand here to see past versions of unnamed-chunk-15-3.png:</em></summary>
<table style = "border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/docs/figure/promoter_architecture.Rmd/unnamed-chunk-15-3.png" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
</tr>
</tbody>
</table>
</details>
</div>
<div id="intronic" class="section level4">
<h4>Intronic</h4>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">fitcs &lt;-<span class="st"> </span><span class="kw">filter_tag_clusters</span>(tc_intronic,<span class="dv">5</span>,<span class="dv">15</span>)</a>
<a class="sourceLine" id="cb35-2" data-line-number="2">fitcspwm &lt;-<span class="st"> </span><span class="kw">generate_pwm</span>(<span class="kw">unlist</span>(GenomicRanges<span class="op">::</span><span class="kw">GRangesList</span>(fitcs<span class="op">$</span>sharp,fitcs<span class="op">$</span>broad)))</a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="kw">plot_frequencies</span>(fitcspwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>),<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">501</span>,<span class="dv">520</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-20&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;20&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw">plot_info</span>(fitcspwm<span class="op">$</span>pwm) <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">250</span>,<span class="dv">501</span>,<span class="dv">750</span>,<span class="dv">1001</span>),<span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;-500&quot;</span>,<span class="st">&quot;-250&quot;</span>,<span class="st">&quot;TSS&quot;</span>,<span class="st">&quot;250&quot;</span>,<span class="st">&quot;500&quot;</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="kw">plot_logo</span>(fitcspwm<span class="op">$</span>pwm,<span class="kw">c</span>(<span class="dv">480</span>,<span class="dv">520</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="nucleotide-composition-of-sharp-and-broad-promoters" class="section level2">
<h2>Nucleotide composition of sharp and broad promoters</h2>
<p>In order to determine whether sharp and broad promoters are of significantly different nucleotide compositions, we first need to divide them up into sharp and broad promoters, then generate random sequences of similar length distributions, and compare the nucleotide content of the actual promoters to those of the randomly generated ones.</p>
<p>First let’s create random intergenic sequences:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="co"># filter by total TPM</span></a>
<a class="sourceLine" id="cb38-2" data-line-number="2">filter_promoter_clusters2 &lt;-<span class="st"> </span><span class="cf">function</span>(pcs,tpm_threshold) {</a>
<a class="sourceLine" id="cb38-3" data-line-number="3">  <span class="co"># remove duplicates, add up total TPM, filter by threshold</span></a>
<a class="sourceLine" id="cb38-4" data-line-number="4">  fpcs &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(pcs) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,strand,full_end,name) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">tpm=</span><span class="kw">sum</span>(<span class="kw">as.numeric</span>(tpm)),<span class="dt">dominant_ctss=</span><span class="kw">max</span>(dominant_ctss)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-7" data-line-number="7"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm <span class="op">&gt;=</span><span class="st"> </span>tpm_threshold) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-8" data-line-number="8"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-9" data-line-number="9"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">end=</span>full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">end=</span><span class="kw">as.numeric</span>(end))</a>
<a class="sourceLine" id="cb38-11" data-line-number="11">  <span class="kw">return</span>(fpcs)</a>
<a class="sourceLine" id="cb38-12" data-line-number="12">}</a>
<a class="sourceLine" id="cb38-13" data-line-number="13"><span class="co"># split by an arbitrary width and extract broad and sharp sequences</span></a>
<a class="sourceLine" id="cb38-14" data-line-number="14">extract_cluster_seqs &lt;-<span class="st"> </span><span class="cf">function</span>(pcs, width) {</a>
<a class="sourceLine" id="cb38-15" data-line-number="15">  <span class="co"># split them by promoter width</span></a>
<a class="sourceLine" id="cb38-16" data-line-number="16">  broad_pcs &lt;-<span class="st"> </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>(dplyr<span class="op">::</span><span class="kw">filter</span>(pcs, end<span class="op">-</span>start <span class="op">&gt;=</span><span class="st"> </span>width))</a>
<a class="sourceLine" id="cb38-17" data-line-number="17">  sharp_pcs &lt;-<span class="st"> </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>(dplyr<span class="op">::</span><span class="kw">filter</span>(pcs, end<span class="op">-</span>start <span class="op">&lt;</span><span class="st"> </span>width))</a>
<a class="sourceLine" id="cb38-18" data-line-number="18">  <span class="co"># retrieve the sequences</span></a>
<a class="sourceLine" id="cb38-19" data-line-number="19">  broad_seqs &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24,broad_pcs)</a>
<a class="sourceLine" id="cb38-20" data-line-number="20">  sharp_seqs &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24,sharp_pcs)</a>
<a class="sourceLine" id="cb38-21" data-line-number="21">  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">broad=</span>broad_seqs,<span class="dt">sharp=</span>sharp_seqs))</a>
<a class="sourceLine" id="cb38-22" data-line-number="22">}</a>
<a class="sourceLine" id="cb38-23" data-line-number="23"><span class="co"># create random seqs to compare to</span></a>
<a class="sourceLine" id="cb38-24" data-line-number="24">extract_random_seqs2 &lt;-<span class="st"> </span><span class="cf">function</span>(seqs,widths) {</a>
<a class="sourceLine" id="cb38-25" data-line-number="25">  <span class="co"># start with the first sequence,</span></a>
<a class="sourceLine" id="cb38-26" data-line-number="26">  <span class="co"># filter by width to avoid errors,</span></a>
<a class="sourceLine" id="cb38-27" data-line-number="27">  <span class="co"># sample randomly from filtered sequences,</span></a>
<a class="sourceLine" id="cb38-28" data-line-number="28">  <span class="co"># grab random interval that matches the length of the</span></a>
<a class="sourceLine" id="cb38-29" data-line-number="29">  <span class="co"># promoter sequence</span></a>
<a class="sourceLine" id="cb38-30" data-line-number="30">  fseqs       &lt;-<span class="st"> </span>seqs[<span class="kw">width</span>(seqs) <span class="op">&gt;</span><span class="st"> </span>widths[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb38-31" data-line-number="31">  rsample     &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(fseqs),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb38-32" data-line-number="32">  rseq        &lt;-<span class="st"> </span>fseqs[rsample][[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb38-33" data-line-number="33">  rstart      &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(rseq)<span class="op">-</span>widths[<span class="dv">1</span>]),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb38-34" data-line-number="34">  random_seqs &lt;-<span class="st"> </span>rseq[rstart<span class="op">:</span>(rstart<span class="op">+</span>widths[<span class="dv">1</span>]<span class="op">-</span><span class="dv">1</span>)]</a>
<a class="sourceLine" id="cb38-35" data-line-number="35">  <span class="co"># do this for all sequences</span></a>
<a class="sourceLine" id="cb38-36" data-line-number="36">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">length</span>(widths)) {</a>
<a class="sourceLine" id="cb38-37" data-line-number="37">    fseqs       &lt;-<span class="st"> </span>seqs[<span class="kw">width</span>(seqs) <span class="op">&gt;</span><span class="st"> </span>widths[i]]</a>
<a class="sourceLine" id="cb38-38" data-line-number="38">    rsample     &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(fseqs),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb38-39" data-line-number="39">    rseq        &lt;-<span class="st"> </span>fseqs[rsample][[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb38-40" data-line-number="40">    rstart      &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span>(<span class="kw">length</span>(rseq)<span class="op">-</span>widths[i]<span class="op">-</span><span class="dv">1</span>),<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb38-41" data-line-number="41">    random_seqs &lt;-<span class="st"> </span><span class="kw">unlist</span>(Biostrings<span class="op">::</span><span class="kw">DNAStringSetList</span>(</a>
<a class="sourceLine" id="cb38-42" data-line-number="42">      Biostrings<span class="op">::</span><span class="kw">DNAStringSet</span>(random_seqs),</a>
<a class="sourceLine" id="cb38-43" data-line-number="43">      Biostrings<span class="op">::</span><span class="kw">DNAStringSet</span>(rseq[rstart<span class="op">:</span>(rstart<span class="op">+</span>widths[i]<span class="op">-</span><span class="dv">1</span>)])))</a>
<a class="sourceLine" id="cb38-44" data-line-number="44">  }</a>
<a class="sourceLine" id="cb38-45" data-line-number="45">  <span class="kw">return</span>(random_seqs)</a>
<a class="sourceLine" id="cb38-46" data-line-number="46">}</a>
<a class="sourceLine" id="cb38-47" data-line-number="47"></a>
<a class="sourceLine" id="cb38-48" data-line-number="48"><span class="co"># calculate promoter cluster nucleotide frequencies </span></a>
<a class="sourceLine" id="cb38-49" data-line-number="49">calculate_frequencies &lt;-<span class="st"> </span><span class="cf">function</span>(cluster_seqs,random_seqs) {</a>
<a class="sourceLine" id="cb38-50" data-line-number="50">  <span class="co"># calculate nucleotide frequencies for broad promoters</span></a>
<a class="sourceLine" id="cb38-51" data-line-number="51">  clengths &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">width</span>(cluster_seqs<span class="op">$</span>broad)</a>
<a class="sourceLine" id="cb38-52" data-line-number="52">  cfreqs   &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">oligonucleotideFrequency</span>(<span class="dt">x=</span>cluster_seqs<span class="op">$</span>broad,<span class="dt">width=</span><span class="dv">1</span>,<span class="dt">step=</span><span class="dv">1</span>,<span class="dt">as.prob=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb38-53" data-line-number="53">  rlengths &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">width</span>(random_seqs<span class="op">$</span>broad)</a>
<a class="sourceLine" id="cb38-54" data-line-number="54">  rfreqs   &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">oligonucleotideFrequency</span>(<span class="dt">x=</span>random_seqs<span class="op">$</span>broad,<span class="dt">width=</span><span class="dv">1</span>,<span class="dt">step=</span><span class="dv">1</span>,<span class="dt">as.prob=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb38-55" data-line-number="55">  cluster_broad_tibble &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">tibble</span>(<span class="dt">length=</span>clengths,</a>
<a class="sourceLine" id="cb38-56" data-line-number="56">                                 <span class="dt">AT=</span>(cfreqs[,<span class="dv">1</span>]<span class="op">+</span>cfreqs[,<span class="dv">4</span>]),</a>
<a class="sourceLine" id="cb38-57" data-line-number="57">                                 <span class="dt">GC=</span>(cfreqs[,<span class="dv">2</span>]<span class="op">+</span>cfreqs[,<span class="dv">3</span>]),</a>
<a class="sourceLine" id="cb38-58" data-line-number="58">                                 <span class="dt">shape=</span><span class="st">&quot;broad&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-59" data-line-number="59"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(length,AT,GC,shape)</a>
<a class="sourceLine" id="cb38-60" data-line-number="60">  random_broad_tibble &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">tibble</span>(<span class="dt">length=</span>rlengths,</a>
<a class="sourceLine" id="cb38-61" data-line-number="61">                                 <span class="dt">AT=</span>(rfreqs[,<span class="dv">1</span>]<span class="op">+</span>rfreqs[,<span class="dv">4</span>]),</a>
<a class="sourceLine" id="cb38-62" data-line-number="62">                                 <span class="dt">GC=</span>(rfreqs[,<span class="dv">2</span>]<span class="op">+</span>rfreqs[,<span class="dv">3</span>]),</a>
<a class="sourceLine" id="cb38-63" data-line-number="63">                                 <span class="dt">shape=</span><span class="st">&quot;broad&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-64" data-line-number="64"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(length,AT,GC,shape)</a>
<a class="sourceLine" id="cb38-65" data-line-number="65">  <span class="co"># calculate nucleotide frequencies for sharp promoters</span></a>
<a class="sourceLine" id="cb38-66" data-line-number="66">  clengths &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">width</span>(cluster_seqs<span class="op">$</span>sharp)</a>
<a class="sourceLine" id="cb38-67" data-line-number="67">  cfreqs   &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">oligonucleotideFrequency</span>(<span class="dt">x=</span>cluster_seqs<span class="op">$</span>sharp,<span class="dt">width=</span><span class="dv">1</span>,<span class="dt">step=</span><span class="dv">1</span>,<span class="dt">as.prob=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb38-68" data-line-number="68">  rlengths &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">width</span>(random_seqs<span class="op">$</span>sharp)</a>
<a class="sourceLine" id="cb38-69" data-line-number="69">  rfreqs   &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">oligonucleotideFrequency</span>(<span class="dt">x=</span>random_seqs<span class="op">$</span>sharp,<span class="dt">width=</span><span class="dv">1</span>,<span class="dt">step=</span><span class="dv">1</span>,<span class="dt">as.prob=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb38-70" data-line-number="70">  cluster_sharp_tibble &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">tibble</span>(<span class="dt">length=</span>clengths,</a>
<a class="sourceLine" id="cb38-71" data-line-number="71">                                 <span class="dt">AT=</span>(cfreqs[,<span class="dv">1</span>]<span class="op">+</span>cfreqs[,<span class="dv">4</span>]),</a>
<a class="sourceLine" id="cb38-72" data-line-number="72">                                 <span class="dt">GC=</span>(cfreqs[,<span class="dv">2</span>]<span class="op">+</span>cfreqs[,<span class="dv">3</span>]),</a>
<a class="sourceLine" id="cb38-73" data-line-number="73">                                 <span class="dt">shape=</span><span class="st">&quot;sharp&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-74" data-line-number="74"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(length,AT,GC,shape)</a>
<a class="sourceLine" id="cb38-75" data-line-number="75">  random_sharp_tibble &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">tibble</span>(<span class="dt">length=</span>rlengths,</a>
<a class="sourceLine" id="cb38-76" data-line-number="76">                                 <span class="dt">AT=</span>(rfreqs[,<span class="dv">1</span>]<span class="op">+</span>rfreqs[,<span class="dv">4</span>]),</a>
<a class="sourceLine" id="cb38-77" data-line-number="77">                                 <span class="dt">GC=</span>(rfreqs[,<span class="dv">2</span>]<span class="op">+</span>rfreqs[,<span class="dv">3</span>]),</a>
<a class="sourceLine" id="cb38-78" data-line-number="78">                                 <span class="dt">shape=</span><span class="st">&quot;sharp&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-79" data-line-number="79"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(length,AT,GC,shape)</a>
<a class="sourceLine" id="cb38-80" data-line-number="80">  <span class="co"># combine into one tibble</span></a>
<a class="sourceLine" id="cb38-81" data-line-number="81">  cluster_shape_tibble &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(cluster_broad_tibble,cluster_sharp_tibble)</a>
<a class="sourceLine" id="cb38-82" data-line-number="82">  random_shape_tibble &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(random_broad_tibble,random_sharp_tibble)</a>
<a class="sourceLine" id="cb38-83" data-line-number="83">  <span class="kw">return</span>(<span class="kw">list</span>(<span class="dt">cluster_shape_tibble=</span>cluster_shape_tibble,<span class="dt">random_shape_tibble=</span>random_shape_tibble))</a>
<a class="sourceLine" id="cb38-84" data-line-number="84">}</a>
<a class="sourceLine" id="cb38-85" data-line-number="85"></a>
<a class="sourceLine" id="cb38-86" data-line-number="86"><span class="co"># generate shape nucleotide frequency</span></a>
<a class="sourceLine" id="cb38-87" data-line-number="87">generate_shape_frequencies &lt;-<span class="st"> </span><span class="cf">function</span>(pcs,seqs,filter_threshold,split_width,freq_fun) {</a>
<a class="sourceLine" id="cb38-88" data-line-number="88">  <span class="co"># filter by TPM threshold</span></a>
<a class="sourceLine" id="cb38-89" data-line-number="89">  fpcs &lt;-<span class="st"> </span><span class="kw">filter_promoter_clusters2</span>(pcs,filter_threshold)</a>
<a class="sourceLine" id="cb38-90" data-line-number="90">  <span class="co"># split by arbitrary width</span></a>
<a class="sourceLine" id="cb38-91" data-line-number="91">  cluster_seqs &lt;-<span class="st"> </span><span class="kw">extract_cluster_seqs</span>(fpcs,split_width)</a>
<a class="sourceLine" id="cb38-92" data-line-number="92">  <span class="co"># generate random promoter clusters of similar widths</span></a>
<a class="sourceLine" id="cb38-93" data-line-number="93">  random_seqs &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">broad=</span><span class="kw">extract_random_seqs2</span>(<span class="dt">seqs=</span>seqs,<span class="dt">widths=</span><span class="kw">width</span>(cluster_seqs<span class="op">$</span>broad)),</a>
<a class="sourceLine" id="cb38-94" data-line-number="94">                      <span class="dt">sharp=</span><span class="kw">extract_random_seqs2</span>(<span class="dt">seqs=</span>seqs,<span class="dt">widths=</span><span class="kw">width</span>(cluster_seqs<span class="op">$</span>sharp))</a>
<a class="sourceLine" id="cb38-95" data-line-number="95">                  )</a>
<a class="sourceLine" id="cb38-96" data-line-number="96">  <span class="co"># normalize by nucleotide frequencies of random sequences</span></a>
<a class="sourceLine" id="cb38-97" data-line-number="97">  shape_tibble &lt;-<span class="st"> </span><span class="kw">do.call</span>(freq_fun,<span class="kw">list</span>(<span class="dt">cluster_seqs=</span>cluster_seqs,<span class="dt">random_seqs=</span><span class="kw">list</span>(<span class="dt">broad=</span>random_seqs<span class="op">$</span>broad,<span class="dt">sharp=</span>random_seqs<span class="op">$</span>sharp)))</a>
<a class="sourceLine" id="cb38-98" data-line-number="98">  <span class="kw">return</span>(shape_tibble)</a>
<a class="sourceLine" id="cb38-99" data-line-number="99">}</a></code></pre></div>
<p>First we can look at the nucleotide composition for intergenic promoters:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># first we can do this for intergenic sequences</span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2">genes      &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/annotations/genes_nuclear_3D7_v24.gff&quot;</span>)</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">telomeres  &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/annotations/Pf3D7_v3_subtelomeres.gff&quot;</span>)</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">intergenic &lt;-<span class="st"> </span>GenomicRanges<span class="op">::</span><span class="kw">gaps</span>(genes)</a>
<a class="sourceLine" id="cb39-5" data-line-number="5">intergenic &lt;-<span class="st"> </span>intergenic[<span class="kw">is.na</span>(GenomicRanges<span class="op">::</span><span class="kw">findOverlaps</span>(intergenic,telomeres,<span class="dt">select=</span><span class="st">&quot;arbitrary&quot;</span>))]</a>
<a class="sourceLine" id="cb39-6" data-line-number="6">intergenic_seqs &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24,intergenic)</a></code></pre></div>
<p>Now we’ll generate the frequencies without normalization:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">shape_tibble &lt;-<span class="st"> </span><span class="kw">generate_shape_frequencies</span>(<span class="dt">pcs=</span>pc_intergenic,</a>
<a class="sourceLine" id="cb40-2" data-line-number="2">                                           <span class="dt">seqs=</span>intergenic_seqs,</a>
<a class="sourceLine" id="cb40-3" data-line-number="3">                                           <span class="dt">filter_threshold=</span><span class="dv">5</span>,</a>
<a class="sourceLine" id="cb40-4" data-line-number="4">                                           <span class="dt">split_width=</span><span class="dv">15</span>,</a>
<a class="sourceLine" id="cb40-5" data-line-number="5">                                           <span class="dt">freq_fun=</span>calculate_frequencies)</a>
<a class="sourceLine" id="cb40-6" data-line-number="6"></a>
<a class="sourceLine" id="cb40-7" data-line-number="7">readr<span class="op">::</span><span class="kw">write_tsv</span>(<span class="dt">x=</span>shape_tibble<span class="op">$</span>cluster_shape_tibble,<span class="dt">path=</span><span class="st">&quot;../output/promoter_architecture/cluster_shape_frequencies.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb40-8" data-line-number="8">readr<span class="op">::</span><span class="kw">write_tsv</span>(<span class="dt">x=</span>shape_tibble<span class="op">$</span>random_shape_tibble,<span class="dt">path=</span><span class="st">&quot;../output/promoter_architecture/random_shape_frequencies.tsv&quot;</span>)</a></code></pre></div>
<p>But we’ll also generate them within normalization and repeat the random sampling 100 times:</p>
<p>Now we can look at some plots of the frequencies:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="co"># read in precalculated frequencies</span></a>
<a class="sourceLine" id="cb41-2" data-line-number="2">cluster_shape_tibble &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="dt">file=</span><span class="st">&quot;../output/promoter_architecture/cluster_shape_frequencies.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb41-3" data-line-number="3">random_shape_tibble &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="dt">file=</span><span class="st">&quot;../output/promoter_architecture/random_shape_frequencies.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb41-4" data-line-number="4"><span class="co"># plot nucleotide frequencies</span></a>
<a class="sourceLine" id="cb41-5" data-line-number="5">g &lt;-<span class="st"> </span>cluster_shape_tibble <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>shape,<span class="dt">y=</span>GC)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</a>
<a class="sourceLine" id="cb41-6" data-line-number="6"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">g &lt;-<span class="st"> </span>random_shape_tibble <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>shape,<span class="dt">y=</span>GC)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()</a>
<a class="sourceLine" id="cb42-2" data-line-number="2"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">g &lt;-<span class="st"> </span>cluster_shape_tibble <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>GC,<span class="dt">color=</span>shape)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">g &lt;-<span class="st"> </span>random_shape_tibble <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>GC,<span class="dt">color=</span>shape)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-21-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="manuscript-numbers" class="section level2">
<h2>Manuscript numbers</h2>
<div id="how-many-tag-clusters" class="section level3">
<h3>How many tag clusters?</h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1">tc_intergenic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb45-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm.dominant_ctss <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb45-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb45-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb45-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb45-7" data-line-number="7"><span class="st">  </span>nrow</a></code></pre></div>
<pre><code>[1] 14405</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">tc_exonic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-2" data-line-number="2"><span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm.dominant_ctss <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb47-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb47-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb47-7" data-line-number="7"><span class="st">  </span>nrow</a></code></pre></div>
<pre><code>[1] 970</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">tc_intronic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-2" data-line-number="2">tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm.dominant_ctss <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb49-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb49-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb49-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb49-7" data-line-number="7"><span class="st">  </span>nrow</a></code></pre></div>
<pre><code>[1] 150</code></pre>
</div>
<div id="how-many-promoter-clusters" class="section level3">
<h3>How many promoter clusters?</h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1">pc_intergenic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-2" data-line-number="2"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end,tp,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb51-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">tpm=</span><span class="kw">sum</span>(<span class="kw">as.double</span>(tpm))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb51-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb51-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb51-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb51-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb51-10" data-line-number="10"><span class="st">  </span>nrow</a></code></pre></div>
<pre><code>[1] 4144</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1">pc_exonic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-2" data-line-number="2"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end,tp,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">tpm=</span><span class="kw">sum</span>(<span class="kw">as.double</span>(tpm))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb53-10" data-line-number="10"><span class="st">  </span>nrow</a></code></pre></div>
<pre><code>[1] 316</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">pc_intronic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-2" data-line-number="2"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end,tp,name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">tpm=</span><span class="kw">sum</span>(<span class="kw">as.double</span>(tpm))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tpm <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb55-10" data-line-number="10"><span class="st">  </span>nrow</a></code></pre></div>
<pre><code>[1] 56</code></pre>
</div>
<div id="how-many-tag-clusters-per-gene-per-time-point" class="section level3">
<h3>How many tag clusters per gene per time point?</h3>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1">g &lt;-<span class="st"> </span>tc_intergenic <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-2" data-line-number="2"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,end,name,tp) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb57-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">as.numeric</span>(tpm.dominant_ctss) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb57-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(name, tp) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-8" data-line-number="8"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-9" data-line-number="9"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;count&quot;</span>,<span class="dt">color=</span><span class="st">&quot;black&quot;</span>,<span class="dt">fill=</span><span class="st">&quot;grey70&quot;</span>,<span class="dt">size=</span><span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-10" data-line-number="10"><span class="st">  </span><span class="kw">facet_grid</span>(.<span class="op">~</span>tp) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-11" data-line-number="11"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-12" data-line-number="12"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-13" data-line-number="13"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">9</span>,<span class="dv">13</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb57-14" data-line-number="14"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">500</span>,<span class="dv">1000</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-15" data-line-number="15"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>),</a>
<a class="sourceLine" id="cb57-16" data-line-number="16">        <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>,<span class="dt">face=</span><span class="st">&quot;bold&quot;</span>),</a>
<a class="sourceLine" id="cb57-17" data-line-number="17">        <span class="dt">axis.line.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>),</a>
<a class="sourceLine" id="cb57-18" data-line-number="18">        <span class="dt">axis.ticks.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>),</a>
<a class="sourceLine" id="cb57-19" data-line-number="19">        <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>),</a>
<a class="sourceLine" id="cb57-20" data-line-number="20">        <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>),</a>
<a class="sourceLine" id="cb57-21" data-line-number="21">        <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>))</a>
<a class="sourceLine" id="cb57-22" data-line-number="22"></a>
<a class="sourceLine" id="cb57-23" data-line-number="23">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">filename=</span><span class="st">&quot;../output/promoter_architecture/tagclusters_per_gene_per_timepoint.png&quot;</span>,<span class="dt">plot=</span>g,<span class="dt">base_width=</span><span class="dv">12</span>)</a>
<a class="sourceLine" id="cb57-24" data-line-number="24">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">filename=</span><span class="st">&quot;../output/promoter_architecture/tagclusters_per_gene_per_timepoint.svg&quot;</span>,<span class="dt">plot=</span>g,<span class="dt">base_width=</span><span class="dv">12</span>)</a>
<a class="sourceLine" id="cb57-25" data-line-number="25"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="is-there-anything-functionally-enriched-in-the-set-of-genes-with-many-tsss" class="section level3">
<h3>Is there anything functionally enriched in the set of genes with many TSSs?</h3>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1">core_genes &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/gene_lists/core_pf3d7_genes.txt&quot;</span>,<span class="dt">col_names=</span>F)<span class="op">$</span>X1</a>
<a class="sourceLine" id="cb58-2" data-line-number="2">out &lt;-<span class="st"> </span>tc_intergenic <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb58-3" data-line-number="3"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(seqnames,start,full_end,name,tp) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">as.numeric</span>(tpm.dominant_ctss) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb58-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(name,tp) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(n <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span> <span class="op">&amp;</span><span class="st"> </span>name <span class="op">%in%</span><span class="st"> </span>core_genes) <span class="op">%$%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-9" data-line-number="9"><span class="st">  </span><span class="kw">unique</span>(name)</a>
<a class="sourceLine" id="cb58-10" data-line-number="10"></a>
<a class="sourceLine" id="cb58-11" data-line-number="11">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span>out,<span class="dt">path=</span><span class="st">&quot;../output/promoter_architecture/5_annotated_tcs.txt&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="co"># Then run this from the command line</span></a>
<a class="sourceLine" id="cb59-2" data-line-number="2"><span class="ex">Rscript</span> code/topgo.R --gene_list output/promoter_architecture/5_annotated_tcs.txt --go_terms data/gene_ontology/Pf3D7_go_sept2014.txt --anno data/gene_ontology/Pf3D7_desc_sept2014.txt --out_prefix output/promoter_architecture/5_annotated_tcs</a></code></pre></div>
</div>
<div id="how-many-broadsharp-tsss-per-timepoint" class="section level3">
<h3>How many broad/sharp TSSs per timepoint?</h3>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1"><span class="co"># abundance estimates</span></a>
<a class="sourceLine" id="cb60-2" data-line-number="2">x3d7_abund &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../output/neighboring_genes/gene_reduced_3d7_abund.rds&quot;</span>)</a>
<a class="sourceLine" id="cb60-3" data-line-number="3">xhb3_abund &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../output/neighboring_genes/gene_reduced_hb3_abund.rds&quot;</span>)</a>
<a class="sourceLine" id="cb60-4" data-line-number="4">xit_abund  &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../output/neighboring_genes/gene_reduced_it_abund.rds&quot;</span>)</a>
<a class="sourceLine" id="cb60-5" data-line-number="5"></a>
<a class="sourceLine" id="cb60-6" data-line-number="6">pcg &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/annotations/PF3D7_codinggenes_for_bedtools.gff&quot;</span>))<span class="op">$</span>ID</a>
<a class="sourceLine" id="cb60-7" data-line-number="7">get_filtered_ids &lt;-<span class="st"> </span><span class="cf">function</span>(abund,tpm_threshold) {</a>
<a class="sourceLine" id="cb60-8" data-line-number="8">  fabund &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-9" data-line-number="9"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">f=</span><span class="kw">sum</span>(TPM<span class="op">&gt;=</span>tpm_threshold)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-11" data-line-number="11"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-12" data-line-number="12"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(f<span class="op">&gt;</span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>pcg)</a>
<a class="sourceLine" id="cb60-13" data-line-number="13">  <span class="kw">return</span>(fabund<span class="op">$</span>gene_id)</a>
<a class="sourceLine" id="cb60-14" data-line-number="14">}</a>
<a class="sourceLine" id="cb60-15" data-line-number="15"></a>
<a class="sourceLine" id="cb60-16" data-line-number="16">fx3d7 &lt;-<span class="st"> </span><span class="kw">get_filtered_ids</span>(x3d7_abund,<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb60-17" data-line-number="17">fxhb3 &lt;-<span class="st"> </span><span class="kw">get_filtered_ids</span>(xhb3_abund,<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb60-18" data-line-number="18">fxit  &lt;-<span class="st"> </span><span class="kw">get_filtered_ids</span>(xit_abund,<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb60-19" data-line-number="19"></a>
<a class="sourceLine" id="cb60-20" data-line-number="20">intertags &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(tibble<span class="op">::</span><span class="kw">as_tibble</span>(tc_intergenic) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tp=</span><span class="kw">as.integer</span>(tp)),x3d7_abund,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;name&quot;</span> =<span class="st"> &quot;gene_id&quot;</span>,<span class="st">&quot;tp&quot;</span>))</a>
<a class="sourceLine" id="cb60-21" data-line-number="21"></a>
<a class="sourceLine" id="cb60-22" data-line-number="22">intertags<span class="op">$</span>type &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">as.integer</span>(intertags<span class="op">$</span>interquantile_width) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">15</span>,<span class="st">&quot;broad&quot;</span>,<span class="st">&quot;sharp&quot;</span>)</a>
<a class="sourceLine" id="cb60-23" data-line-number="23">intertags <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-24" data-line-number="24"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="kw">as.numeric</span>(tpm.dominant_ctss) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">abs</span>(<span class="kw">as.numeric</span>(anno_start) <span class="op">-</span><span class="st"> </span><span class="kw">as.numeric</span>(dominant_ctss)) <span class="op">&lt;=</span><span class="st"> </span><span class="dv">2500</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-25" data-line-number="25"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>tp,<span class="dt">fill=</span>type)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-26" data-line-number="26"><span class="st">  </span><span class="kw">geom_bar</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-27" data-line-number="27"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette=</span><span class="st">&quot;Greys&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-28" data-line-number="28"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Frequency&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-29" data-line-number="29"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Timepoint&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-30" data-line-number="30"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">18</span>),</a>
<a class="sourceLine" id="cb60-31" data-line-number="31">        <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>,<span class="dt">face=</span><span class="st">&quot;bold&quot;</span>),</a>
<a class="sourceLine" id="cb60-32" data-line-number="32">        <span class="dt">axis.line.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb60-33" data-line-number="33">        <span class="dt">axis.ticks.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb60-34" data-line-number="34">        <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb60-35" data-line-number="35">        <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb60-36" data-line-number="36">        <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>),</a>
<a class="sourceLine" id="cb60-37" data-line-number="37">        <span class="dt">legend.title=</span><span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-38" data-line-number="38"><span class="st">  </span><span class="kw">labs</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb60-39" data-line-number="39"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>))</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>)) {</a>
<a class="sourceLine" id="cb61-2" data-line-number="2">  intertags <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-3" data-line-number="3"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tp<span class="op">==</span>i) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(seqnames,strand,q_<span class="fl">0.1</span>,q_<span class="fl">0.9</span>,</a>
<a class="sourceLine" id="cb61-5" data-line-number="5">                  dominant_ctss,tpm,tpm.dominant_ctss,name,TPM,type) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb61-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Chromosome=</span>seqnames,</a>
<a class="sourceLine" id="cb61-7" data-line-number="7">                  <span class="dt">Strand=</span>strand,</a>
<a class="sourceLine" id="cb61-8" data-line-number="8">                  <span class="dt">Quantile_10=</span>q_<span class="fl">0.1</span>,</a>
<a class="sourceLine" id="cb61-9" data-line-number="9">                  <span class="dt">Quantile_90=</span>q_<span class="fl">0.9</span>,</a>
<a class="sourceLine" id="cb61-10" data-line-number="10">                  <span class="dt">Quantile_TPM=</span>tpm,</a>
<a class="sourceLine" id="cb61-11" data-line-number="11">                  <span class="dt">Dominant_tag=</span>dominant_ctss,</a>
<a class="sourceLine" id="cb61-12" data-line-number="12">                  <span class="dt">Dominant_tag_TPM=</span>tpm.dominant_ctss,</a>
<a class="sourceLine" id="cb61-13" data-line-number="13">                  <span class="dt">Annotation=</span>name,</a>
<a class="sourceLine" id="cb61-14" data-line-number="14">                  <span class="dt">Annotation_TPM=</span>TPM,</a>
<a class="sourceLine" id="cb61-15" data-line-number="15">                  <span class="dt">Shape=</span>type) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb61-16" data-line-number="16"><span class="st">    </span>readr<span class="op">::</span><span class="kw">write_tsv</span>(<span class="kw">paste0</span>(<span class="st">&quot;../output/promoter_architecture/intertags_tp&quot;</span>,i,<span class="st">&quot;.tsv&quot;</span>))</a>
<a class="sourceLine" id="cb61-17" data-line-number="17">  </a>
<a class="sourceLine" id="cb61-18" data-line-number="18">  <span class="co">#googledrive::drive_upload(media=paste0(&quot;../output/promoter_architecture/intertags_tp&quot;,i,&quot;.tsv&quot;),</span></a>
<a class="sourceLine" id="cb61-19" data-line-number="19">  <span class="co">#                        path=&quot;Shared/Pf RNA-seq manuscript 2017/Supplementary tables/Named tables&quot;,</span></a>
<a class="sourceLine" id="cb61-20" data-line-number="20">  <span class="co">#                        name=paste0(&quot;intertags_tp&quot;,i,&quot;.tsv&quot;),</span></a>
<a class="sourceLine" id="cb61-21" data-line-number="21">  <span class="co">#                        type=&quot;spreadsheet&quot;)</span></a>
<a class="sourceLine" id="cb61-22" data-line-number="22">}</a></code></pre></div>
</div>
<div id="is-there-actually-an-expression-difference-between-genes-annotated-with-sharp-and-genes-annotated-with-broad-tcs" class="section level3">
<h3>Is there actually an expression difference between genes annotated with sharp and genes annotated with broad TCs?</h3>
<p>We look at tag clusters here rather than promoter clusters since the width of a promoter cluster could be due to the fact that TSSs can be dynamic throughout the falciparum IDC, not just that the TSS at a certain time point is “sharp” or “broad.”</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1">intertags &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(intertags <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">tp=</span><span class="kw">as.integer</span>(tp)),x3d7_abund,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;name&quot;</span>=<span class="st">&quot;gene_id&quot;</span>,<span class="st">&quot;tp&quot;</span>))</a>
<a class="sourceLine" id="cb62-2" data-line-number="2"></a>
<a class="sourceLine" id="cb62-3" data-line-number="3">s &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(intertags, type<span class="op">==</span><span class="st">&quot;sharp&quot;</span>)</a>
<a class="sourceLine" id="cb62-4" data-line-number="4">b &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(intertags, type<span class="op">==</span><span class="st">&quot;broad&quot;</span>)</a>
<a class="sourceLine" id="cb62-5" data-line-number="5"></a>
<a class="sourceLine" id="cb62-6" data-line-number="6">sabund &lt;-<span class="st"> </span>x3d7_abund <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>s<span class="op">$</span>name) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">type=</span><span class="st">&quot;sharp&quot;</span>)</a>
<a class="sourceLine" id="cb62-7" data-line-number="7">babund &lt;-<span class="st"> </span>x3d7_abund <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>b<span class="op">$</span>name) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">type=</span><span class="st">&quot;broad&quot;</span>)</a>
<a class="sourceLine" id="cb62-8" data-line-number="8"></a>
<a class="sourceLine" id="cb62-9" data-line-number="9">nabund &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(sabund,babund)</a>
<a class="sourceLine" id="cb62-10" data-line-number="10"></a>
<a class="sourceLine" id="cb62-11" data-line-number="11">g &lt;-<span class="st"> </span>nabund <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-12" data-line-number="12"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(strain <span class="op">==</span><span class="st"> &quot;3d7&quot;</span> <span class="op">&amp;</span><span class="st"> </span>TPM <span class="op">&gt;</span><span class="st"> </span><span class="dv">5</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-13" data-line-number="13"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>type,<span class="dt">y=</span><span class="kw">log10</span>(TPM<span class="op">+</span><span class="dv">1</span>),<span class="dt">fill=</span>type)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-14" data-line-number="14"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-15" data-line-number="15"><span class="st">  </span><span class="kw">facet_grid</span>(<span class="op">~</span>tp) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-16" data-line-number="16"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Log10 TPM&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-17" data-line-number="17"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-18" data-line-number="18"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>),</a>
<a class="sourceLine" id="cb62-19" data-line-number="19">        <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>,<span class="dt">face=</span><span class="st">&quot;bold&quot;</span>),</a>
<a class="sourceLine" id="cb62-20" data-line-number="20">        <span class="dt">axis.line.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb62-21" data-line-number="21">        <span class="dt">axis.ticks.x=</span><span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb62-22" data-line-number="22">        <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>),</a>
<a class="sourceLine" id="cb62-23" data-line-number="23">        <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.0</span>),</a>
<a class="sourceLine" id="cb62-24" data-line-number="24">        <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">16</span>),</a>
<a class="sourceLine" id="cb62-25" data-line-number="25">        <span class="dt">axis.text.x=</span><span class="kw">element_blank</span>()) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-26" data-line-number="26"><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;sharp&quot;</span>=<span class="st">&quot;#b29ed1&quot;</span>,<span class="st">&quot;broad&quot;</span>=<span class="st">&quot;#edc074&quot;</span>))</a>
<a class="sourceLine" id="cb62-27" data-line-number="27"></a>
<a class="sourceLine" id="cb62-28" data-line-number="28">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">filename=</span><span class="st">&quot;../output/promoter_architecture/shape_gene_expression.png&quot;</span>,<span class="dt">plot=</span>g,<span class="dt">base_width=</span><span class="dv">12</span>)</a>
<a class="sourceLine" id="cb62-29" data-line-number="29">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">filename=</span><span class="st">&quot;../output/promoter_architecture/shape_gene_expression.svg&quot;</span>,<span class="dt">plot=</span>g,<span class="dt">base_width=</span><span class="dv">12</span>)</a>
<a class="sourceLine" id="cb62-30" data-line-number="30"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/promoter_architecture.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>)) {</a>
<a class="sourceLine" id="cb63-2" data-line-number="2">  h1 &lt;-<span class="st"> </span>sabund <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tp<span class="op">==</span>i)</a>
<a class="sourceLine" id="cb63-3" data-line-number="3">  h2 &lt;-<span class="st"> </span>babund <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(tp<span class="op">==</span>i)</a>
<a class="sourceLine" id="cb63-4" data-line-number="4">  <span class="kw">print</span>(<span class="kw">t.test</span>(<span class="kw">log2</span>(h1<span class="op">$</span>TPM<span class="op">+</span><span class="dv">1</span>),<span class="kw">log2</span>(h2<span class="op">$</span>TPM<span class="op">+</span><span class="dv">1</span>)))</a>
<a class="sourceLine" id="cb63-5" data-line-number="5">}</a></code></pre></div>
<pre><code>
    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.7821, df = 5416, p-value = 1.311e-11
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3861753 -0.2129839
sample estimates:
mean of x mean of y 
  6.72247   7.02205 


    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.9608, df = 5400.3, p-value = 3.787e-12
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4155964 -0.2329437
sample estimates:
mean of x mean of y 
 6.605683  6.929953 


    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.2211, df = 5351.9, p-value = 5.311e-10
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.4182292 -0.2178022
sample estimates:
mean of x mean of y 
 6.402663  6.720679 


    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.0286, df = 5402.4, p-value = 1.763e-09
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3682973 -0.1875464
sample estimates:
mean of x mean of y 
 6.788421  7.066343 


    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.0978, df = 5450.1, p-value = 1.149e-09
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3674217 -0.1886488
sample estimates:
mean of x mean of y 
 6.745237  7.023272 


    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.0034, df = 5419.4, p-value = 2.058e-09
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3623882 -0.1839739
sample estimates:
mean of x mean of y 
 6.679580  6.952761 


    Welch Two Sample t-test

data:  log2(h1$TPM + 1) and log2(h2$TPM + 1)
t = -6.3478, df = 5389.4, p-value = 2.362e-10
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3830553 -0.2022844
sample estimates:
mean of x mean of y 
 6.585352  6.878021 </code></pre>
</div>
<div id="is-there-anything-significant-about-gene-annotated-with-shifting-tsss" class="section level3">
<h3>Is there anything significant about gene annotated with shifting TSSs?</h3>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" data-line-number="1">shifting &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="dt">file=</span><span class="st">&quot;../output/ctss_clustering/modified/annotated_shifting.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb65-2" data-line-number="2">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(shifting<span class="op">$</span>name),<span class="dt">path=</span><span class="st">&quot;../output/ctss_clustering/modified/annotated_shifting_genes.txt&quot;</span>)</a></code></pre></div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Gentoo/Linux

Matrix products: default
BLAS: /usr/local/lib64/R/lib/libRblas.so
LAPACK: /usr/local/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gdtools_0.1.7                        
 [2] bindrcpp_0.2.2                       
 [3] BSgenome.Pfalciparum.PlasmoDB.v24_1.0
 [4] BSgenome_1.48.0                      
 [5] rtracklayer_1.40.6                   
 [6] Biostrings_2.48.0                    
 [7] XVector_0.20.0                       
 [8] GenomicRanges_1.32.7                 
 [9] GenomeInfoDb_1.16.0                  
[10] org.Pf.plasmo.db_3.6.0               
[11] AnnotationDbi_1.42.1                 
[12] IRanges_2.14.12                      
[13] S4Vectors_0.18.3                     
[14] Biobase_2.40.0                       
[15] BiocGenerics_0.26.0                  
[16] scales_1.0.0                         
[17] cowplot_0.9.3                        
[18] magrittr_1.5                         
[19] forcats_0.3.0                        
[20] stringr_1.3.1                        
[21] dplyr_0.7.6                          
[22] purrr_0.2.5                          
[23] readr_1.1.1                          
[24] tidyr_0.8.1                          
[25] tibble_1.4.2                         
[26] ggplot2_3.0.0                        
[27] tidyverse_1.2.1                      

loaded via a namespace (and not attached):
 [1] nlme_3.1-137                bitops_1.0-6               
 [3] matrixStats_0.54.0          lubridate_1.7.4            
 [5] bit64_0.9-7                 RColorBrewer_1.1-2         
 [7] httr_1.3.1                  rprojroot_1.3-2            
 [9] tools_3.5.0                 backports_1.1.2            
[11] R6_2.3.0                    seqLogo_1.46.0             
[13] DBI_1.0.0                   lazyeval_0.2.1             
[15] colorspace_1.3-2            withr_2.1.2                
[17] tidyselect_0.2.4            bit_1.1-14                 
[19] compiler_3.5.0              git2r_0.23.0               
[21] cli_1.0.1                   rvest_0.3.2                
[23] xml2_1.2.0                  DelayedArray_0.6.6         
[25] labeling_0.3                digest_0.6.17              
[27] Rsamtools_1.32.3            svglite_1.2.1              
[29] rmarkdown_1.10              R.utils_2.7.0              
[31] pkgconfig_2.0.2             htmltools_0.3.6            
[33] rlang_0.2.2                 readxl_1.1.0               
[35] rstudioapi_0.8              RSQLite_2.1.1              
[37] bindr_0.1.1                 jsonlite_1.5               
[39] BiocParallel_1.14.2         R.oo_1.22.0                
[41] RCurl_1.95-4.11             GenomeInfoDbData_1.1.0     
[43] Matrix_1.2-14               Rcpp_0.12.19               
[45] munsell_0.5.0               R.methodsS3_1.7.1          
[47] stringi_1.2.4               whisker_0.3-2              
[49] yaml_2.2.0                  SummarizedExperiment_1.10.1
[51] zlibbioc_1.26.0             plyr_1.8.4                 
[53] grid_3.5.0                  blob_1.1.1                 
[55] crayon_1.3.4                lattice_0.20-35            
[57] haven_1.1.2                 hms_0.4.2                  
[59] knitr_1.20                  pillar_1.3.0               
[61] reshape2_1.4.3              XML_3.98-1.16              
[63] glue_1.3.0                  evaluate_0.11              
[65] modelr_0.1.2                cellranger_1.1.0           
[67] gtable_0.2.0                assertthat_0.2.0           
[69] broom_0.5.0                 GenomicAlignments_1.16.0   
[71] memoise_1.1.0               workflowr_1.1.1            </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
