<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2018-09-25" />

<title>Differential transcript expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential transcript expression</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2018-09-25</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-10-05</p>
<p><strong>Code version:</strong> 5584d21</p>
<div id="process-data" class="section level2">
<h2>Process data</h2>
<p>First we need to actually calculate the differentially expressed genes. Weâ€™ll need to import some data first:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">core_genes &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/gene_lists/core_pf3d7_genes.txt&quot;</span>,<span class="dt">col_names=</span>F)<span class="op">$</span>X1</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">x3d7exp &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../output/neighboring_genes/gene_reduced_3d7_abund.rds&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">xhb3exp &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../output/neighboring_genes/gene_reduced_hb3_abund.rds&quot;</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">xitexp  &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../output/neighboring_genes/gene_reduced_it_abund.rds&quot;</span>)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">abund &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(x3d7exp,xhb3exp,xitexp)</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="kw">rm</span>(x3d7exp,xhb3exp,xitexp)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co"># unique gene IDs</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12">ug &lt;-<span class="st"> </span><span class="kw">unique</span>(abund<span class="op">$</span>gene_id)</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co"># Create data frame of genes that are &quot;on&quot; in each strain</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">detected &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">n =</span> <span class="kw">sum</span>(TPM <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">on =</span> <span class="kw">ifelse</span>(n <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>n)</a>
<a class="sourceLine" id="cb1-21" data-line-number="21"></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co"># What genes are &quot;on&quot; in all strains</span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23">onall &lt;-<span class="st"> </span>detected <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(on <span class="op">==</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">length</span>(<span class="kw">unique</span>(strain))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(count <span class="op">==</span><span class="st"> </span><span class="dv">3</span>) <span class="op">%$%</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="st">  </span>gene_id</a>
<a class="sourceLine" id="cb1-29" data-line-number="29"></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co"># What genes are &quot;off&quot; in all strains</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31">offall &lt;-<span class="st"> </span>detected <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(on <span class="op">==</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">count =</span> <span class="kw">length</span>(<span class="kw">unique</span>(strain))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(count <span class="op">==</span><span class="st"> </span><span class="dv">3</span>) <span class="op">%$%</span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="st">  </span>gene_id</a>
<a class="sourceLine" id="cb1-37" data-line-number="37"></a>
<a class="sourceLine" id="cb1-38" data-line-number="38">on3d7 &lt;-<span class="st"> </span>detected <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(strain <span class="op">==</span><span class="st"> &quot;3d7&quot;</span> <span class="op">&amp;</span><span class="st"> </span>on <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>core_genes) <span class="op">%$%</span><span class="st"> </span>gene_id</a>
<a class="sourceLine" id="cb1-39" data-line-number="39">onhb3 &lt;-<span class="st"> </span>detected <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(strain <span class="op">==</span><span class="st"> &quot;hb3&quot;</span> <span class="op">&amp;</span><span class="st"> </span>on <span class="op">==</span><span class="st"> </span><span class="dv">1</span><span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>core_genes) <span class="op">%$%</span><span class="st"> </span>gene_id</a>
<a class="sourceLine" id="cb1-40" data-line-number="40">onit  &lt;-<span class="st"> </span>detected <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(strain <span class="op">==</span><span class="st"> &quot;it&quot;</span> <span class="op">&amp;</span><span class="st"> </span>on <span class="op">==</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>core_genes) <span class="op">%$%</span><span class="st"> </span>gene_id</a>
<a class="sourceLine" id="cb1-41" data-line-number="41"></a>
<a class="sourceLine" id="cb1-42" data-line-number="42">on3d7hb3 &lt;-<span class="st"> </span><span class="kw">intersect</span>(on3d7,onhb3)</a>
<a class="sourceLine" id="cb1-43" data-line-number="43">on3d7it  &lt;-<span class="st"> </span><span class="kw">intersect</span>(on3d7,onit)</a>
<a class="sourceLine" id="cb1-44" data-line-number="44">onhb3it  &lt;-<span class="st"> </span><span class="kw">intersect</span>(onhb3,onit)</a>
<a class="sourceLine" id="cb1-45" data-line-number="45"></a>
<a class="sourceLine" id="cb1-46" data-line-number="46">comparable &lt;-<span class="st"> </span><span class="kw">intersect</span>(core_genes, onall)</a></code></pre></div>
<p>Now we can look to calculate the differentially expressed genes.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># First calculate differential expression</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">diff_exp_3d7hb3 &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="st">  </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>, <span class="st">&quot;hb3&quot;</span>) <span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>comparable) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">  </span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="st">  </span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">t.test</span>(<span class="kw">log2</span>(TPM<span class="op">+</span><span class="dv">1</span>)<span class="op">~</span>strain,<span class="dt">data=</span>.))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_id, p.value)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">diff_exp_3d7hb3<span class="op">$</span>strain_pair &lt;-<span class="st"> &quot;3d7-hb3&quot;</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"></a>
<a class="sourceLine" id="cb2-11" data-line-number="11">diff_exp_3d7it &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="st">  </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>, <span class="st">&quot;it&quot;</span>) <span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>comparable) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="st">  </span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="st">  </span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">t.test</span>(<span class="kw">log2</span>(TPM<span class="op">+</span><span class="dv">1</span>)<span class="op">~</span>strain,<span class="dt">data=</span>.))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_id, p.value)</a>
<a class="sourceLine" id="cb2-17" data-line-number="17"></a>
<a class="sourceLine" id="cb2-18" data-line-number="18">diff_exp_3d7it<span class="op">$</span>strain_pair &lt;-<span class="st"> &quot;3d7-it&quot;</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"></a>
<a class="sourceLine" id="cb2-20" data-line-number="20">diff_exp_hb3it &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="st">  </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hb3&quot;</span>, <span class="st">&quot;it&quot;</span>) <span class="op">&amp;</span><span class="st"> </span>gene_id <span class="op">%in%</span><span class="st"> </span>comparable) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="st">  </span><span class="kw">group_by</span>(gene_id) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="st">  </span><span class="kw">do</span>(broom<span class="op">::</span><span class="kw">tidy</span>(<span class="kw">t.test</span>(<span class="kw">log2</span>(TPM<span class="op">+</span><span class="dv">1</span>)<span class="op">~</span>strain,<span class="dt">data=</span>.))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_id, p.value)</a>
<a class="sourceLine" id="cb2-26" data-line-number="26"></a>
<a class="sourceLine" id="cb2-27" data-line-number="27">diff_exp_hb3it<span class="op">$</span>strain_pair &lt;-<span class="st"> &quot;hb3-it&quot;</span></a>
<a class="sourceLine" id="cb2-28" data-line-number="28"></a>
<a class="sourceLine" id="cb2-29" data-line-number="29">diff_exp &lt;-<span class="st"> </span><span class="kw">rbind</span>(diff_exp_3d7hb3, diff_exp_3d7it, diff_exp_hb3it)</a>
<a class="sourceLine" id="cb2-30" data-line-number="30"></a>
<a class="sourceLine" id="cb2-31" data-line-number="31">diff3d7hb3 &lt;-<span class="st"> </span>diff_exp <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(strain_pair <span class="op">==</span><span class="st"> &quot;3d7-hb3&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(strain_pair, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&quot;strain1&quot;</span>,<span class="st">&quot;strain2&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb2-32" data-line-number="32">diff3d7it  &lt;-<span class="st"> </span>diff_exp <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(strain_pair <span class="op">==</span><span class="st"> &quot;3d7-it&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(strain_pair, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&quot;strain1&quot;</span>,<span class="st">&quot;strain2&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb2-33" data-line-number="33">diffhb3it  &lt;-<span class="st"> </span>diff_exp <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(strain_pair <span class="op">==</span><span class="st"> &quot;hb3-it&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(strain_pair, <span class="dt">into =</span> <span class="kw">c</span>(<span class="st">&quot;strain1&quot;</span>,<span class="st">&quot;strain2&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>)</a>
<a class="sourceLine" id="cb2-34" data-line-number="34"></a>
<a class="sourceLine" id="cb2-35" data-line-number="35">diff3d7hb3<span class="op">$</span>qvalue &lt;-<span class="st"> </span>qvalue<span class="op">::</span><span class="kw">qvalue</span>(diff3d7hb3<span class="op">$</span>p.value)<span class="op">$</span>qvalues</a>
<a class="sourceLine" id="cb2-36" data-line-number="36">diff3d7it<span class="op">$</span>qvalue  &lt;-<span class="st"> </span>qvalue<span class="op">::</span><span class="kw">qvalue</span>(diff3d7it<span class="op">$</span>p.value)<span class="op">$</span>qvalues</a>
<a class="sourceLine" id="cb2-37" data-line-number="37">diffhb3it<span class="op">$</span>qvalue  &lt;-<span class="st"> </span>qvalue<span class="op">::</span><span class="kw">qvalue</span>(diffhb3it<span class="op">$</span>p.value)<span class="op">$</span>qvalues</a></code></pre></div>
<p>And finally, we can plot the results and write everything to a file:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Create a scatter plot comparing 3D7 to HB3</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">plot_3d7_hb3_scatter &lt;-<span class="st"> </span><span class="cf">function</span>() {</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">  overunder &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="st">    </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7hb3) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="st">    </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="st">    </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="st">    </span><span class="kw">mutate</span>(<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> =<span class="st"> </span><span class="kw">log2</span>(<span class="st">`</span><span class="dt">3d7</span><span class="st">`</span> <span class="op">/</span><span class="st"> </span>hb3)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>, <span class="op">-</span>hb3, <span class="op">-</span>it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="st">    </span><span class="kw">inner_join</span>(diff3d7hb3, <span class="dt">by =</span> <span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="st">    </span><span class="kw">gather</span>(strain1, strain2,<span class="op">-</span><span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span>,<span class="op">-</span>gene_id,<span class="op">-</span>p.value,<span class="op">-</span>qvalue) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>strain1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">strain=</span>strain2)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"></a>
<a class="sourceLine" id="cb3-16" data-line-number="16">  df &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="st">    </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7hb3) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="st">    </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="st">    </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>,<span class="st">&quot;hb3&quot;</span>))</a>
<a class="sourceLine" id="cb3-22" data-line-number="22"></a>
<a class="sourceLine" id="cb3-23" data-line-number="23">  df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, overunder, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="st">&quot;strain&quot;</span>))</a>
<a class="sourceLine" id="cb3-24" data-line-number="24"></a>
<a class="sourceLine" id="cb3-25" data-line-number="25">  df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="st">    </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>hb3,<span class="dt">y=</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>,<span class="dt">label=</span>gene_id)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">color=</span><span class="st">&quot;grey80&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">data =</span> <span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">spread</span>(strain, m), <span class="dt">color =</span> <span class="st">&quot;#E41A1C&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">data =</span> <span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">spread</span>(strain, m), <span class="dt">color =</span> <span class="st">&quot;#377EB8&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Log10(TPM) 3D7&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Log10(TPM) HB3&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans=</span><span class="st">&quot;log10&quot;</span>,<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>,<span class="dv">1000</span>,<span class="dv">10000</span>),<span class="dt">labels=</span><span class="kw">trans_format</span>(<span class="st">&#39;log10&#39;</span>,<span class="kw">math_format</span>(<span class="dv">10</span><span class="op">^</span>.x)),<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans=</span><span class="st">&quot;log10&quot;</span>,<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>,<span class="dv">1000</span>,<span class="dv">10000</span>),<span class="dt">labels=</span><span class="kw">trans_format</span>(<span class="st">&#39;log10&#39;</span>,<span class="kw">math_format</span>(<span class="dv">10</span><span class="op">^</span>.x)),<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.line.x=</span><span class="kw">element_line</span>(<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-36" data-line-number="36">          <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-37" data-line-number="37">          <span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb3-38" data-line-number="38">          <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb3-39" data-line-number="39">          <span class="dt">axis.ticks.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-40" data-line-number="40">          <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>))</a>
<a class="sourceLine" id="cb3-41" data-line-number="41">}</a>
<a class="sourceLine" id="cb3-42" data-line-number="42"></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"><span class="co"># Create a scatter plot comparing 3D7 to IT</span></a>
<a class="sourceLine" id="cb3-44" data-line-number="44">plot_3d7_it_scatter &lt;-<span class="st"> </span><span class="cf">function</span>() {</a>
<a class="sourceLine" id="cb3-45" data-line-number="45"></a>
<a class="sourceLine" id="cb3-46" data-line-number="46">  overunder &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"><span class="st">    </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="st">    </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="st">    </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="st">    </span><span class="kw">mutate</span>(<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> =<span class="st"> </span><span class="kw">log2</span>(<span class="st">`</span><span class="dt">3d7</span><span class="st">`</span> <span class="op">/</span><span class="st"> </span>it)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>, <span class="op">-</span>hb3, <span class="op">-</span>it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-53" data-line-number="53"><span class="st">    </span><span class="kw">inner_join</span>(diff3d7it, <span class="dt">by =</span> <span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="st">    </span><span class="kw">gather</span>(strain1, strain2,<span class="op">-</span><span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span>,<span class="op">-</span>gene_id,<span class="op">-</span>p.value,<span class="op">-</span>qvalue) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>strain1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-56" data-line-number="56"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">strain=</span>strain2)</a>
<a class="sourceLine" id="cb3-57" data-line-number="57"></a>
<a class="sourceLine" id="cb3-58" data-line-number="58">  df &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-59" data-line-number="59"><span class="st">    </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-60" data-line-number="60"><span class="st">    </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-61" data-line-number="61"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-62" data-line-number="62"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-63" data-line-number="63"><span class="st">    </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>,<span class="st">&quot;it&quot;</span>))</a>
<a class="sourceLine" id="cb3-64" data-line-number="64"></a>
<a class="sourceLine" id="cb3-65" data-line-number="65">  df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, overunder, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="st">&quot;strain&quot;</span>))</a>
<a class="sourceLine" id="cb3-66" data-line-number="66"></a>
<a class="sourceLine" id="cb3-67" data-line-number="67">  df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-68" data-line-number="68"><span class="st">    </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-69" data-line-number="69"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>it,<span class="dt">y=</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>,<span class="dt">label=</span>gene_id)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-70" data-line-number="70"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">color=</span><span class="st">&quot;grey80&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-71" data-line-number="71"><span class="st">    </span><span class="co">#geom_point(data = subset(df, (`3d7_over_it` &lt;= -2) &amp; qvalue &lt; 0.05) %&gt;% ungroup() %&gt;% spread(strain, m), color = &quot;#4DAF4A&quot;) +</span></a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">data =</span> <span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">spread</span>(strain, m), <span class="dt">color =</span> <span class="st">&quot;#377EB8&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-73" data-line-number="73"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Log10(TPM) 3D7&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-74" data-line-number="74"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Log10(TPM) IT&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-75" data-line-number="75"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans=</span><span class="st">&quot;log10&quot;</span>,<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>,<span class="dv">1000</span>,<span class="dv">10000</span>),<span class="dt">labels=</span><span class="kw">trans_format</span>(<span class="st">&#39;log10&#39;</span>,<span class="kw">math_format</span>(<span class="dv">10</span><span class="op">^</span>.x)),<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-76" data-line-number="76"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans=</span><span class="st">&quot;log10&quot;</span>,<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>,<span class="dv">1000</span>,<span class="dv">10000</span>),<span class="dt">labels=</span><span class="kw">trans_format</span>(<span class="st">&#39;log10&#39;</span>,<span class="kw">math_format</span>(<span class="dv">10</span><span class="op">^</span>.x)),<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-77" data-line-number="77"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.line.x=</span><span class="kw">element_line</span>(<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-78" data-line-number="78">          <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-79" data-line-number="79">          <span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb3-80" data-line-number="80">          <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb3-81" data-line-number="81">          <span class="dt">axis.ticks.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-82" data-line-number="82">          <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>))</a>
<a class="sourceLine" id="cb3-83" data-line-number="83">}</a>
<a class="sourceLine" id="cb3-84" data-line-number="84"></a>
<a class="sourceLine" id="cb3-85" data-line-number="85"><span class="co"># Create a scatter plot comparing HB3 to IT</span></a>
<a class="sourceLine" id="cb3-86" data-line-number="86">plot_hb3_it_scatter &lt;-<span class="st"> </span><span class="cf">function</span>() {</a>
<a class="sourceLine" id="cb3-87" data-line-number="87"></a>
<a class="sourceLine" id="cb3-88" data-line-number="88">  overunder &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-89" data-line-number="89"><span class="st">    </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>onhb3it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-90" data-line-number="90"><span class="st">    </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-91" data-line-number="91"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-92" data-line-number="92"><span class="st">    </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-93" data-line-number="93"><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">hb3_over_it =</span> <span class="kw">log2</span>(hb3 <span class="op">/</span><span class="st"> </span>it)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-94" data-line-number="94"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>, <span class="op">-</span>hb3, <span class="op">-</span>it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-95" data-line-number="95"><span class="st">    </span><span class="kw">inner_join</span>(diffhb3it, <span class="dt">by =</span> <span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-96" data-line-number="96"><span class="st">    </span><span class="kw">gather</span>(strain1, strain2,<span class="op">-</span>hb3_over_it,<span class="op">-</span>gene_id,<span class="op">-</span>p.value,<span class="op">-</span>qvalue) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-97" data-line-number="97"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>strain1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-98" data-line-number="98"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">strain=</span>strain2)</a>
<a class="sourceLine" id="cb3-99" data-line-number="99"></a>
<a class="sourceLine" id="cb3-100" data-line-number="100">  df &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-101" data-line-number="101"><span class="st">    </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>onhb3it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-102" data-line-number="102"><span class="st">    </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-103" data-line-number="103"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-104" data-line-number="104"><span class="st">    </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-105" data-line-number="105"><span class="st">    </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hb3&quot;</span>,<span class="st">&quot;it&quot;</span>))</a>
<a class="sourceLine" id="cb3-106" data-line-number="106"></a>
<a class="sourceLine" id="cb3-107" data-line-number="107">  df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, overunder, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="st">&quot;strain&quot;</span>))</a>
<a class="sourceLine" id="cb3-108" data-line-number="108"></a>
<a class="sourceLine" id="cb3-109" data-line-number="109">  df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-110" data-line-number="110"><span class="st">    </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-111" data-line-number="111"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>hb3,<span class="dt">y=</span>it,<span class="dt">label=</span>gene_id)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-112" data-line-number="112"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">color=</span><span class="st">&quot;grey80&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-113" data-line-number="113"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">data =</span> <span class="kw">subset</span>(df, (hb3_over_it <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">spread</span>(strain, m), <span class="dt">color =</span> <span class="st">&quot;#E41A1C&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-114" data-line-number="114"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">data =</span> <span class="kw">subset</span>(df, (hb3_over_it <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">spread</span>(strain, m), <span class="dt">color =</span> <span class="st">&quot;#4DAF4A&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-115" data-line-number="115"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Log10(TPM) HB3&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-116" data-line-number="116"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;Log10(TPM) IT&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-117" data-line-number="117"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans=</span><span class="st">&quot;log10&quot;</span>,<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>,<span class="dv">1000</span>,<span class="dv">10000</span>),<span class="dt">labels=</span><span class="kw">trans_format</span>(<span class="st">&#39;log10&#39;</span>,<span class="kw">math_format</span>(<span class="dv">10</span><span class="op">^</span>.x)),<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-118" data-line-number="118"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">trans=</span><span class="st">&quot;log10&quot;</span>,<span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>,<span class="dv">1000</span>,<span class="dv">10000</span>),<span class="dt">labels=</span><span class="kw">trans_format</span>(<span class="st">&#39;log10&#39;</span>,<span class="kw">math_format</span>(<span class="dv">10</span><span class="op">^</span>.x)),<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-119" data-line-number="119"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.line.x=</span><span class="kw">element_line</span>(<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-120" data-line-number="120">          <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-121" data-line-number="121">          <span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb3-122" data-line-number="122">          <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb3-123" data-line-number="123">          <span class="dt">axis.ticks.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>),</a>
<a class="sourceLine" id="cb3-124" data-line-number="124">          <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>))</a>
<a class="sourceLine" id="cb3-125" data-line-number="125">}</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">g &lt;-<span class="st"> </span><span class="kw">plot_3d7_hb3_scatter</span>()</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="st">&quot;../output/differential_expression/3d7_hb3_scatter.svg&quot;</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="st">&quot;../output/differential_expression/3d7_hb3_scatter.png&quot;</span>)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">g &lt;-<span class="st"> </span><span class="kw">plot_3d7_it_scatter</span>()</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="st">&quot;../output/differential_expression/3d7_it_scatter.svg&quot;</span>)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="st">&quot;../output/differential_expression/3d7_it_scatter.png&quot;</span>)</a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">g &lt;-<span class="st"> </span><span class="kw">plot_hb3_it_scatter</span>()</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="st">&quot;../output/differential_expression/hb3_it_scatter.svg&quot;</span>)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="st">&quot;../output/differential_expression/hb3_it_scatter.png&quot;</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="kw">print</span>(g)</a></code></pre></div>
<p><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="manuscript-numbers" class="section level2">
<h2>Manuscript numbers</h2>
<div id="how-many-differentially-expressed-genes-do-we-find" class="section level3">
<h3>How many differentially expressed genes do we find?</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">overunder &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7hb3) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="st">  </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="st">  </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="st">  </span><span class="kw">mutate</span>(<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> =<span class="st"> </span><span class="kw">log2</span>(<span class="st">`</span><span class="dt">3d7</span><span class="st">`</span> <span class="op">/</span><span class="st"> </span>hb3)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>, <span class="op">-</span>hb3, <span class="op">-</span>it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="st">  </span><span class="kw">inner_join</span>(diff3d7hb3, <span class="dt">by =</span> <span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="st">  </span><span class="kw">gather</span>(strain1, strain2,<span class="op">-</span><span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span>,<span class="op">-</span>gene_id,<span class="op">-</span>p.value,<span class="op">-</span>qvalue) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>strain1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">strain=</span>strain2)</a>
<a class="sourceLine" id="cb7-12" data-line-number="12"></a>
<a class="sourceLine" id="cb7-13" data-line-number="13">df &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="st">  </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7hb3) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="st">  </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="st">  </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>,<span class="st">&quot;hb3&quot;</span>))</a>
<a class="sourceLine" id="cb7-19" data-line-number="19"></a>
<a class="sourceLine" id="cb7-20" data-line-number="20">df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, overunder, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="st">&quot;strain&quot;</span>))</a>
<a class="sourceLine" id="cb7-21" data-line-number="21"></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="kw">print</span>(<span class="st">&quot;Number of genes overexpressed in HB3:&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;Number of genes overexpressed in HB3:&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id)))</a></code></pre></div>
<pre><code>[1] 28</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id),<span class="dt">path=</span><span class="st">&quot;../output/differential_expression/hb3_over_3d7_genes.txt&quot;</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">print</span>(<span class="st">&quot;Number of genes overexpressed in 3D7:&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;Number of genes overexpressed in 3D7:&quot;</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id)))</a></code></pre></div>
<pre><code>[1] 408</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_hb3</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id),<span class="dt">path=</span><span class="st">&quot;../output/differential_expression/3d7_over_hb3_genes.txt&quot;</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">###########################################################</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"></a>
<a class="sourceLine" id="cb15-5" data-line-number="5">overunder &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="st">  </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="st">  </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="st">  </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="st">  </span><span class="kw">mutate</span>(<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> =<span class="st"> </span><span class="kw">log2</span>(<span class="st">`</span><span class="dt">3d7</span><span class="st">`</span> <span class="op">/</span><span class="st"> </span>it)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>, <span class="op">-</span>hb3, <span class="op">-</span>it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="st">  </span><span class="kw">inner_join</span>(diff3d7it, <span class="dt">by =</span> <span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="st">  </span><span class="kw">gather</span>(strain1, strain2,<span class="op">-</span><span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span>,<span class="op">-</span>gene_id,<span class="op">-</span>p.value,<span class="op">-</span>qvalue) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-14" data-line-number="14"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>strain1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">strain=</span>strain2)</a>
<a class="sourceLine" id="cb15-16" data-line-number="16"></a>
<a class="sourceLine" id="cb15-17" data-line-number="17">df &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-18" data-line-number="18"><span class="st">  </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>on3d7it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-19" data-line-number="19"><span class="st">  </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-20" data-line-number="20"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-21" data-line-number="21"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-22" data-line-number="22"><span class="st">  </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>,<span class="st">&quot;it&quot;</span>))</a>
<a class="sourceLine" id="cb15-23" data-line-number="23"></a>
<a class="sourceLine" id="cb15-24" data-line-number="24">df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, overunder, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="st">&quot;strain&quot;</span>))</a>
<a class="sourceLine" id="cb15-25" data-line-number="25"></a>
<a class="sourceLine" id="cb15-26" data-line-number="26"><span class="kw">print</span>(<span class="st">&quot;Number of genes overexpressed in IT:&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;Number of genes overexpressed in IT:&quot;</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id)))</a></code></pre></div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id),<span class="dt">path=</span><span class="st">&quot;../output/differential_expression/it_over_3d7_genes.txt&quot;</span>)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw">print</span>(<span class="st">&quot;Number of genes overexpressed in 3D7:&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;Number of genes overexpressed in 3D7:&quot;</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id)))</a></code></pre></div>
<pre><code>[1] 129</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">3d7_over_it</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id),<span class="dt">path=</span><span class="st">&quot;../output/differential_expression/3d7_over_it_genes.txt&quot;</span>)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2"></a>
<a class="sourceLine" id="cb23-3" data-line-number="3">###########################################################</a>
<a class="sourceLine" id="cb23-4" data-line-number="4"></a>
<a class="sourceLine" id="cb23-5" data-line-number="5">overunder &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="st">  </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>onhb3it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="st">  </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="st">  </span><span class="kw">spread</span>(strain, m) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="st">  </span><span class="kw">mutate</span>(<span class="st">`</span><span class="dt">hb3_over_it</span><span class="st">`</span> =<span class="st"> </span><span class="kw">log2</span>(<span class="st">`</span><span class="dt">hb3</span><span class="st">`</span> <span class="op">/</span><span class="st"> </span>it)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">3d7</span><span class="st">`</span>, <span class="op">-</span>hb3, <span class="op">-</span>it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-12" data-line-number="12"><span class="st">  </span><span class="kw">inner_join</span>(diffhb3it, <span class="dt">by =</span> <span class="st">&quot;gene_id&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-13" data-line-number="13"><span class="st">  </span><span class="kw">gather</span>(strain1, strain2,<span class="op">-</span><span class="st">`</span><span class="dt">hb3_over_it</span><span class="st">`</span>,<span class="op">-</span>gene_id,<span class="op">-</span>p.value,<span class="op">-</span>qvalue) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-14" data-line-number="14"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>strain1) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">strain=</span>strain2)</a>
<a class="sourceLine" id="cb23-16" data-line-number="16"></a>
<a class="sourceLine" id="cb23-17" data-line-number="17">df &lt;-<span class="st"> </span>abund <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-18" data-line-number="18"><span class="st">  </span><span class="kw">filter</span>(gene_id <span class="op">%in%</span><span class="st"> </span>onhb3it) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-19" data-line-number="19"><span class="st">  </span><span class="kw">group_by</span>(gene_id, strain) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-20" data-line-number="20"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">m =</span> <span class="kw">mean</span>(TPM)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-21" data-line-number="21"><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb23-22" data-line-number="22"><span class="st">  </span><span class="kw">filter</span>(strain <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;hb3&quot;</span>,<span class="st">&quot;it&quot;</span>))</a>
<a class="sourceLine" id="cb23-23" data-line-number="23"></a>
<a class="sourceLine" id="cb23-24" data-line-number="24">df &lt;-<span class="st"> </span><span class="kw">inner_join</span>(df, overunder, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;gene_id&quot;</span>, <span class="st">&quot;strain&quot;</span>))</a>
<a class="sourceLine" id="cb23-25" data-line-number="25"></a>
<a class="sourceLine" id="cb23-26" data-line-number="26"><span class="kw">print</span>(<span class="st">&quot;Number of genes overexpressed in HB3:&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;Number of genes overexpressed in HB3:&quot;</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">hb3_over_it</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id)))</a></code></pre></div>
<pre><code>[1] 161</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">hb3_over_it</span><span class="st">`</span> <span class="op">&lt;=</span><span class="st"> </span><span class="dv">-2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id),<span class="dt">path=</span><span class="st">&quot;../output/differential_expression/it_over_hb3_genes.txt&quot;</span>)</a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="kw">print</span>(<span class="st">&quot;Number of genes overexpressed in IT:&quot;</span>)</a></code></pre></div>
<pre><code>[1] &quot;Number of genes overexpressed in IT:&quot;</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="kw">print</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">hb3_over_it</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id)))</a></code></pre></div>
<pre><code>[1] 34</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">readr<span class="op">::</span><span class="kw">write_lines</span>(<span class="dt">x=</span><span class="kw">unique</span>(<span class="kw">subset</span>(df, (<span class="st">`</span><span class="dt">hb3_over_it</span><span class="st">`</span> <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>) <span class="op">&amp;</span><span class="st"> </span>qvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)<span class="op">$</span>gene_id),<span class="dt">path=</span><span class="st">&quot;../output/differential_expression/hb3_over_it_genes.txt&quot;</span>)</a></code></pre></div>
<p>Now run this from the command line:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw">for</span> <span class="fu">file</span> in <span class="va">$(</span><span class="fu">ls</span> ../output/differential_expression/*.txt<span class="va">)</span><span class="kw">;</span> <span class="kw">do</span> <span class="ex">Rscript</span> code/topgo.R --gene_list ../output/differential_expression/<span class="va">$file</span> --go_terms ../data/gene_ontology/Pf3D7_go_sept2014.txt --anno ../data/gene_ontology/Pf3D7_desc_sept2014.txt --out_prefix ../output/differential_expression/<span class="va">${file%%</span>.*<span class="va">}</span><span class="kw">;</span> <span class="kw">done</span></a></code></pre></div>
<p>Plot the abundances for the following ApiAP2 transcriptsâ€¦</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">what &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;PF3D7_0604100&quot;</span>,<span class="st">&quot;PF3D7_1222400&quot;</span>,<span class="st">&quot;PF3D7_1408200&quot;</span>,</a>
<a class="sourceLine" id="cb33-2" data-line-number="2">          <span class="st">&quot;PF3D7_1143100&quot;</span>,<span class="st">&quot;PF3D7_0516800&quot;</span>,<span class="st">&quot;PF3D7_1466400&quot;</span>,</a>
<a class="sourceLine" id="cb33-3" data-line-number="3">          <span class="st">&quot;PF3D7_0420300&quot;</span>,<span class="st">&quot;PF3D7_0613800&quot;</span>,<span class="st">&quot;PF3D7_0802100&quot;</span>,</a>
<a class="sourceLine" id="cb33-4" data-line-number="4">          <span class="st">&quot;PF3D7_1107800&quot;</span>,<span class="st">&quot;PF3D7_1139300&quot;</span>,<span class="st">&quot;PF3D7_1239200&quot;</span>,</a>
<a class="sourceLine" id="cb33-5" data-line-number="5">          <span class="st">&quot;PF3D7_1317200&quot;</span>,<span class="st">&quot;PF3D7_1342900&quot;</span>,<span class="st">&quot;PF3D7_1449500&quot;</span>)</a>
<a class="sourceLine" id="cb33-6" data-line-number="6"></a>
<a class="sourceLine" id="cb33-7" data-line-number="7"><span class="cf">for</span> (w <span class="cf">in</span> what) {</a>
<a class="sourceLine" id="cb33-8" data-line-number="8">  g &lt;-<span class="st"> </span><span class="kw">plot_strain_abundances</span>(abund,w)</a>
<a class="sourceLine" id="cb33-9" data-line-number="9">  <span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="kw">paste0</span>(<span class="st">&quot;../output/differential_expression/&quot;</span>,w,<span class="st">&quot;.svg&quot;</span>),<span class="dt">height=</span><span class="dv">4</span>,<span class="dt">width=</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb33-10" data-line-number="10">  <span class="kw">ggsave</span>(<span class="dt">plot=</span>g,<span class="dt">filename=</span><span class="kw">paste0</span>(<span class="st">&quot;../output/differential_expression/&quot;</span>,w,<span class="st">&quot;.png&quot;</span>),<span class="dt">height=</span><span class="dv">4</span>,<span class="dt">width=</span><span class="dv">4</span>)</a>
<a class="sourceLine" id="cb33-11" data-line-number="11">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb33-12" data-line-number="12">}</a></code></pre></div>
<p><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/strain_differential_expression.Rmd/unnamed-chunk-7-15.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Gentoo/Linux

Matrix products: default
BLAS: /usr/local/lib64/R/lib/libRblas.so
LAPACK: /usr/local/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gdtools_0.1.7                        
 [2] bindrcpp_0.2.2                       
 [3] BSgenome.Pfalciparum.PlasmoDB.v24_1.0
 [4] BSgenome_1.48.0                      
 [5] rtracklayer_1.40.6                   
 [6] Biostrings_2.48.0                    
 [7] XVector_0.20.0                       
 [8] GenomicRanges_1.32.7                 
 [9] GenomeInfoDb_1.16.0                  
[10] org.Pf.plasmo.db_3.6.0               
[11] AnnotationDbi_1.42.1                 
[12] IRanges_2.14.12                      
[13] S4Vectors_0.18.3                     
[14] Biobase_2.40.0                       
[15] BiocGenerics_0.26.0                  
[16] scales_1.0.0                         
[17] cowplot_0.9.3                        
[18] magrittr_1.5                         
[19] forcats_0.3.0                        
[20] stringr_1.3.1                        
[21] dplyr_0.7.6                          
[22] purrr_0.2.5                          
[23] readr_1.1.1                          
[24] tidyr_0.8.1                          
[25] tibble_1.4.2                         
[26] ggplot2_3.0.0                        
[27] tidyverse_1.2.1                      

loaded via a namespace (and not attached):
 [1] nlme_3.1-137                bitops_1.0-6               
 [3] matrixStats_0.54.0          lubridate_1.7.4            
 [5] bit64_0.9-7                 httr_1.3.1                 
 [7] rprojroot_1.3-2             tools_3.5.0                
 [9] backports_1.1.2             R6_2.3.0                   
[11] DBI_1.0.0                   lazyeval_0.2.1             
[13] colorspace_1.3-2            withr_2.1.2                
[15] tidyselect_0.2.4            bit_1.1-14                 
[17] compiler_3.5.0              git2r_0.23.0               
[19] cli_1.0.1                   rvest_0.3.2                
[21] xml2_1.2.0                  DelayedArray_0.6.6         
[23] labeling_0.3                digest_0.6.17              
[25] Rsamtools_1.32.3            svglite_1.2.1              
[27] rmarkdown_1.10              R.utils_2.7.0              
[29] pkgconfig_2.0.2             htmltools_0.3.6            
[31] rlang_0.2.2                 readxl_1.1.0               
[33] rstudioapi_0.8              RSQLite_2.1.1              
[35] bindr_0.1.1                 jsonlite_1.5               
[37] BiocParallel_1.14.2         R.oo_1.22.0                
[39] RCurl_1.95-4.11             GenomeInfoDbData_1.1.0     
[41] Matrix_1.2-14               Rcpp_0.12.19               
[43] munsell_0.5.0               R.methodsS3_1.7.1          
[45] stringi_1.2.4               yaml_2.2.0                 
[47] SummarizedExperiment_1.10.1 zlibbioc_1.26.0            
[49] qvalue_2.12.0               plyr_1.8.4                 
[51] grid_3.5.0                  blob_1.1.1                 
[53] crayon_1.3.4                lattice_0.20-35            
[55] splines_3.5.0               haven_1.1.2                
[57] hms_0.4.2                   knitr_1.20                 
[59] pillar_1.3.0                reshape2_1.4.3             
[61] XML_3.98-1.16               glue_1.3.0                 
[63] evaluate_0.11               modelr_0.1.2               
[65] cellranger_1.1.0            gtable_0.2.0               
[67] assertthat_0.2.0            broom_0.5.0                
[69] GenomicAlignments_1.16.0    memoise_1.1.0              
[71] workflowr_1.1.1            </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
