<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Philipp Ross" />

<meta name="date" content="2018-09-25" />

<title>Transcription Factor Binding Site Analysis</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.4/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sanger RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Transcription Factor Binding Site Analysis</h1>
<h4 class="author"><em>Philipp Ross</em></h4>
<h4 class="date"><em>2018-09-25</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-10-26</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/llinaslab/sanger_rnaseq/tree/dd9d56a8fbb07e41b641cd5e2fda6fe162519c0a" target="_blank">dd9d56a</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.httr-oauth
    Ignored:    code/.DS_Store
    Ignored:    code/differential_expression/
    Ignored:    code/differential_phase/
    Ignored:    data/
    Ignored:    docs/.DS_Store
    Ignored:    docs/figure/.DS_Store
    Ignored:    docs/figure/neighboring_genes.Rmd/.DS_Store
    Ignored:    output/ctss_clustering/
    Ignored:    output/differential_detection/
    Ignored:    output/differential_expression/
    Ignored:    output/differential_phase/
    Ignored:    output/extensive_transcription/
    Ignored:    output/final_utrs/
    Ignored:    output/gcbias/
    Ignored:    output/homopolymer_analysis/
    Ignored:    output/neighboring_genes/
    Ignored:    output/promoter_architecture/
    Ignored:    output/tfbs_analysis/
    Ignored:    output/transcript_abundance/

Untracked files:
    Untracked:  _workflowr.yml
    Untracked:  figures/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/analyze_neighboring_genes.Rmd
    Modified:   analysis/array_correlations.Rmd
    Modified:   analysis/calculate_transcript_abundance.Rmd
    Deleted:    analysis/chunks.R
    Modified:   analysis/comparing_utrs.Rmd
    Modified:   analysis/ctss_clustering.Rmd
    Modified:   analysis/dynamic_tss.Rmd
    Modified:   analysis/extensive_transcription.Rmd
    Modified:   analysis/final_utrs.Rmd
    Modified:   analysis/gcbias.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    Modified:   analysis/process_neighboring_genes.Rmd
    Modified:   analysis/promoter_architecture.Rmd
    Modified:   analysis/strain_differential_detection.Rmd
    Modified:   analysis/strain_differential_expression.Rmd
    Modified:   analysis/strain_differential_phase.Rmd
    Modified:   analysis/tfbs_analysis.Rmd
    Deleted:    docs/Rplots.pdf

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/35726e6ac81c7089fbc841edd0497fc30c1b076e/analysis/tfbs_analysis.Rmd" target="_blank">35726e6</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-10-04
</td>
<td style="text-align:left;">
finished tfbs analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/llinaslab/sanger_rnaseq/35726e6ac81c7089fbc841edd0497fc30c1b076e/docs/tfbs_analysis.html" target="_blank">35726e6</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-10-04
</td>
<td style="text-align:left;">
finished tfbs analysis
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/b1b89808d44aab66572a52174ef09c607bec1f18/analysis/tfbs_analysis.Rmd" target="_blank">b1b8980</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-30
</td>
<td style="text-align:left;">
you’re what message
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/llinaslab/sanger_rnaseq/blob/f59e2e352af3a95d97e00cda9084806c9d33bd03/analysis/tfbs_analysis.Rmd" target="_blank">f59e2e3</a>
</td>
<td style="text-align:left;">
Philipp Ross
</td>
<td style="text-align:left;">
2018-09-22
</td>
<td style="text-align:left;">
hellooooo
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<div id="refined-motif-search" class="section level2">
<h2>Refined motif search</h2>
<p>Now that we have predicted TSSs, we can refine our motif binding site search space and look at smaller regions of sequence just upstream of our predicted TSSs. We will use our most commonly used TSSs for this.</p>
<p>First we need to create promoter regions to analyze:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">promoter_region &lt;-<span class="st"> </span><span class="dv">1000</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">core_genes &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/gene_lists/core_pf3d7_genes.txt&quot;</span>,<span class="dt">col_names=</span>F)<span class="op">$</span>X1</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4">upstream_ORF &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../data/annotations/genes_3D7_v24.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(ID <span class="op">%in%</span><span class="st"> </span>core_genes) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start,end<span class="op">+</span>promoter_region),</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>promoter_region,end)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-12" data-line-number="12"></a>
<a class="sourceLine" id="cb1-13" data-line-number="13">promoters_3d7 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_3d7_plasmodb_compatible.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Parent=</span>stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(<span class="kw">unlist</span>(Parent), <span class="st">&quot;rna_&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;&quot;</span>),<span class="st">&quot;[.][0-9]&quot;</span>,<span class="st">&quot;&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">distinct</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start,end<span class="op">+</span>promoter_region),</a>
<a class="sourceLine" id="cb1-18" data-line-number="18">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>promoter_region,end)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">start=</span>newstart,<span class="dt">end=</span>newend) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-23" data-line-number="23"></a>
<a class="sourceLine" id="cb1-24" data-line-number="24">promoters_hb3 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_hb3_plasmodb_compatible.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Parent=</span>stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(<span class="kw">unlist</span>(Parent), <span class="st">&quot;rna_&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;&quot;</span>),<span class="st">&quot;[.][0-9]&quot;</span>,<span class="st">&quot;&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">distinct</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start,end<span class="op">+</span>promoter_region),</a>
<a class="sourceLine" id="cb1-29" data-line-number="29">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>promoter_region,end)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-33" data-line-number="33"></a>
<a class="sourceLine" id="cb1-34" data-line-number="34">promoters_it &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(<span class="st">&quot;../output/final_utrs/longest_utrs_it_plasmodb_compatible.gff&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Parent=</span>stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(<span class="kw">unlist</span>(Parent), <span class="st">&quot;rna_&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;&quot;</span>),<span class="st">&quot;[.][0-9]&quot;</span>,<span class="st">&quot;&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">distinct</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">newend=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start,end<span class="op">+</span>promoter_region),</a>
<a class="sourceLine" id="cb1-39" data-line-number="39">                <span class="dt">newstart=</span><span class="kw">ifelse</span>(strand<span class="op">==</span><span class="st">&quot;+&quot;</span>,start<span class="op">-</span>promoter_region,end)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>start,<span class="op">-</span>end) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(type<span class="op">==</span><span class="st">&quot;5UTR&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="st">  </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>()</a>
<a class="sourceLine" id="cb1-43" data-line-number="43"></a>
<a class="sourceLine" id="cb1-44" data-line-number="44">rtracklayer<span class="op">::</span><span class="kw">export.gff3</span>(<span class="dt">object=</span>upstream_ORF,<span class="dt">con=</span><span class="st">&quot;../output/tfbs_analysis/upstream_ORF.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-45" data-line-number="45">rtracklayer<span class="op">::</span><span class="kw">export.gff3</span>(<span class="dt">object=</span>promoters_3d7,<span class="dt">con=</span><span class="st">&quot;../output/tfbs_analysis/promoters_3d7.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-46" data-line-number="46">rtracklayer<span class="op">::</span><span class="kw">export.gff3</span>(<span class="dt">object=</span>promoters_hb3,<span class="dt">con=</span><span class="st">&quot;../output/tfbs_analysis/promoters_hb3.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-47" data-line-number="47">rtracklayer<span class="op">::</span><span class="kw">export.gff3</span>(<span class="dt">object=</span>promoters_it,<span class="dt">con=</span><span class="st">&quot;../output/tfbs_analysis/promoters_it.gff&quot;</span>)</a>
<a class="sourceLine" id="cb1-48" data-line-number="48"></a>
<a class="sourceLine" id="cb1-49" data-line-number="49">seqs_upstream_ORF &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24, upstream_ORF)</a>
<a class="sourceLine" id="cb1-50" data-line-number="50"><span class="kw">names</span>(seqs_upstream_ORF) &lt;-<span class="st"> </span><span class="kw">unlist</span>(upstream_ORF<span class="op">$</span>ID)</a>
<a class="sourceLine" id="cb1-51" data-line-number="51">Biostrings<span class="op">::</span><span class="kw">writeXStringSet</span>(seqs_upstream_ORF,<span class="st">&quot;../output/tfbs_analysis/upstream_ORF.fasta&quot;</span>)</a>
<a class="sourceLine" id="cb1-52" data-line-number="52">seqs3d7 &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24, promoters_3d7)</a>
<a class="sourceLine" id="cb1-53" data-line-number="53"><span class="kw">names</span>(seqs3d7) &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(<span class="kw">unlist</span>(promoters_3d7<span class="op">$</span>Parent), <span class="st">&quot;rna_&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb1-54" data-line-number="54">Biostrings<span class="op">::</span><span class="kw">writeXStringSet</span>(seqs3d7,<span class="st">&quot;../output/tfbs_analysis/promoters_3d7.fasta&quot;</span>)</a>
<a class="sourceLine" id="cb1-55" data-line-number="55">seqshb3 &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24, promoters_hb3)</a>
<a class="sourceLine" id="cb1-56" data-line-number="56"><span class="kw">names</span>(seqshb3) &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(<span class="kw">unlist</span>(promoters_hb3<span class="op">$</span>Parent), <span class="st">&quot;rna_&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb1-57" data-line-number="57">Biostrings<span class="op">::</span><span class="kw">writeXStringSet</span>(seqshb3,<span class="st">&quot;../output/tfbs_analysis/promoters_hb3.fasta&quot;</span>)</a>
<a class="sourceLine" id="cb1-58" data-line-number="58">seqsit &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24, promoters_it)</a>
<a class="sourceLine" id="cb1-59" data-line-number="59"><span class="kw">names</span>(seqsit) &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_replace</span>(stringr<span class="op">::</span><span class="kw">str_replace</span>(<span class="kw">unlist</span>(promoters_it<span class="op">$</span>Parent), <span class="st">&quot;rna_&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="st">&quot;-1&quot;</span>, <span class="st">&quot;&quot;</span>)</a>
<a class="sourceLine" id="cb1-60" data-line-number="60">Biostrings<span class="op">::</span><span class="kw">writeXStringSet</span>(seqsit,<span class="st">&quot;../output/tfbs_analysis/promoters_it.fasta&quot;</span>)</a></code></pre></div>
<p>We should also create files that parse the genomic coordinates so as to be able to load them into a genome browser:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">for</span> <span class="ex">strain</span> in 3d7 hb3 it<span class="kw">;</span> <span class="kw">do</span> <span class="ex">bedtools</span> getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>.gff -fo ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>_genome_coords.fasta -s<span class="kw">;</span> <span class="kw">done</span></a></code></pre></div>
<p>Now we need to run fimo on these regions and search for our motifs:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># generate background files</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">for</span> <span class="ex">strain</span> in 3d7 hb3 it<span class="kw">;</span> <span class="kw">do</span> <span class="ex">fasta-get-markov</span> -m 3 ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>.fasta ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>.background<span class="kw">;</span> <span class="kw">done</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="ex">fasta-get-markov</span> -m 3 ../output/tfbs_analysis/upstream_ORF.fasta ../output/tfbs_analysis/upstream_ORF.background</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co"># run fimo for promoters with gene names</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="kw">for</span> <span class="ex">strain</span> in 3d7 hb3 it<span class="kw">;</span> <span class="kw">do</span> <span class="ex">fimo</span> --bgfile ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>.background --no-qvalue -oc ../output/tfbs_analysis/promoters_<span class="va">${strain}</span> --thresh 1e-2 ../data/motif_cores/ap2_pbm_cores.meme ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>.fasta<span class="kw">;</span> <span class="kw">done</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="ex">fimo</span> --bgfile ../output/tfbs_analysis/upstream_ORF.background --no-qvalue -oc ../output/tfbs_analysis/upstream_ORF --thresh 1e-2 ../data/motif_cores/ap2_pbm_cores.meme ../output/tfbs_analysis/upstream_ORF.fasta</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co"># and for promoters with genome coordinates</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="kw">for</span> <span class="ex">strain</span> in 3d7 hb3 it<span class="kw">;</span> <span class="kw">do</span> <span class="ex">fimo</span> --bgfile ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>.background --no-qvalue --parse-genomic-coord -oc ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>_genome_coords --thresh 1e-2 ../data/motif_cores/ap2_pbm_cores.meme ../output/tfbs_analysis/promoters_<span class="va">${strain}</span>_genome_coords.fasta<span class="kw">;</span> <span class="kw">done</span></a></code></pre></div>
<p>Now we can import that data and analyze it for positional information:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">fig_theme &lt;-<span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">18</span>),</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">          <span class="dt">axis.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">26</span>,<span class="dt">face=</span><span class="st">&quot;bold&quot;</span>),</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">          <span class="dt">axis.line.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">          <span class="dt">axis.ticks.x=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">          <span class="dt">axis.line.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">          <span class="dt">axis.ticks.y=</span><span class="kw">element_line</span>(<span class="dt">colour=</span><span class="st">&quot;black&quot;</span>,<span class="dt">size=</span><span class="fl">1.5</span>),</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">          <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">20</span>),</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">          <span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">26</span>,<span class="dt">face=</span><span class="st">&quot;bold&quot;</span>))</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"></a>
<a class="sourceLine" id="cb4-10" data-line-number="10">motifs_upstream_ORF &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/upstream_ORF/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb4-12" data-line-number="12"></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="cf">for</span> (motif <span class="cf">in</span> <span class="kw">unique</span>(motifs_upstream_ORF<span class="op">$</span>motif_id)) {</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">  n &lt;-<span class="st"> </span>motifs_upstream_ORF <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(motif_id<span class="op">==</span>motif)</a>
<a class="sourceLine" id="cb4-16" data-line-number="16">  <span class="co">#print(ks.test(n$start,runif(10000),alternative=&quot;less&quot;))</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17">  g &lt;-<span class="st"> </span>n <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>start)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="st">    </span><span class="kw">ggtitle</span>(motif) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="st">    </span>fig_theme</a>
<a class="sourceLine" id="cb4-23" data-line-number="23">  <span class="kw">assign</span>(<span class="dt">x=</span>motif,<span class="dt">value=</span>g)</a>
<a class="sourceLine" id="cb4-24" data-line-number="24">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb4-25" data-line-number="25">}</a></code></pre></div>
<p><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-16.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-18.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-20.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-21.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-22.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-23.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">plots &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="st">`</span><span class="dt">PF3D7_1342900-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1408200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1222600</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1143100</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">                            <span class="st">`</span><span class="dt">PF3D7_1466400</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0802100</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0604100-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D3</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                            <span class="st">`</span><span class="dt">PF3D7_0420300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1107800</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1350900</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1456000</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                            <span class="st">`</span><span class="dt">PF3D7_1305200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1317200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0516800-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D2ext</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0420300-D2</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                            <span class="st">`</span><span class="dt">PF3D7_1239200-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1139300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0730300-DsL</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">plot=</span>plots,<span class="dt">filename=</span><span class="st">&quot;../output/tfbs_analysis/upstream_ORF/all_motifs.pdf&quot;</span>,<span class="dt">base_width=</span><span class="dv">26</span>,<span class="dt">base_height=</span><span class="dv">20</span>)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="kw">rm</span>(motifs_upstream_ORF)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"></a>
<a class="sourceLine" id="cb5-9" data-line-number="9">motifs_3d7 &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/promoters_3d7/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb5-11" data-line-number="11"></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="cf">for</span> (motif <span class="cf">in</span> <span class="kw">unique</span>(motifs_3d7<span class="op">$</span>motif_id)) {</a>
<a class="sourceLine" id="cb5-13" data-line-number="13">  n &lt;-<span class="st"> </span>motifs_3d7 <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(motif_id<span class="op">==</span>motif)</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">  <span class="co">#print(ks.test(n$start,runif(10000),alternative=&quot;less&quot;))</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16">  g &lt;-<span class="st"> </span>n <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>start)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="st">    </span><span class="kw">ggtitle</span>(motif) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="st">    </span>fig_theme</a>
<a class="sourceLine" id="cb5-22" data-line-number="22">  <span class="kw">assign</span>(<span class="dt">x=</span>motif,<span class="dt">value=</span>g)</a>
<a class="sourceLine" id="cb5-23" data-line-number="23">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb5-24" data-line-number="24">}</a></code></pre></div>
<p><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-24.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-25.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-26.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-27.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-28.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-29.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-30.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-31.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-32.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-33.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-34.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-35.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-36.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-37.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-38.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-39.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-40.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-41.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-42.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-43.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-44.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-45.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-46.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">plots &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="st">`</span><span class="dt">PF3D7_1342900-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1408200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1222600</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1143100</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                            <span class="st">`</span><span class="dt">PF3D7_1466400</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0802100</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0604100-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D3</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                            <span class="st">`</span><span class="dt">PF3D7_0420300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1107800</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1350900</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1456000</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                            <span class="st">`</span><span class="dt">PF3D7_1305200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1317200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0516800-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D2ext</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0420300-D2</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">                            <span class="st">`</span><span class="dt">PF3D7_1239200-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1139300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0730300-DsL</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">plot=</span>plots,<span class="dt">filename=</span><span class="st">&quot;../output/tfbs_analysis/promoters_3d7/all_motifs.pdf&quot;</span>,<span class="dt">base_width=</span><span class="dv">26</span>,<span class="dt">base_height=</span><span class="dv">20</span>)</a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="kw">rm</span>(motifs_3d7)</a>
<a class="sourceLine" id="cb6-8" data-line-number="8"></a>
<a class="sourceLine" id="cb6-9" data-line-number="9">motifs_hb3 &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/promoters_hb3/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="cf">for</span> (motif <span class="cf">in</span> <span class="kw">unique</span>(motifs_hb3<span class="op">$</span>motif_id)) {</a>
<a class="sourceLine" id="cb6-13" data-line-number="13">  n &lt;-<span class="st"> </span>motifs_hb3 <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(motif_id<span class="op">==</span>motif)</a>
<a class="sourceLine" id="cb6-15" data-line-number="15">  <span class="co">#print(ks.test(n$start,runif(10000),alternative=&quot;less&quot;))</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16">  g &lt;-<span class="st"> </span>n <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>start)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="st">    </span><span class="kw">ggtitle</span>(motif) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="st">    </span>fig_theme</a>
<a class="sourceLine" id="cb6-22" data-line-number="22">  <span class="kw">assign</span>(<span class="dt">x=</span>motif,<span class="dt">value=</span>g)</a>
<a class="sourceLine" id="cb6-23" data-line-number="23">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb6-24" data-line-number="24">}</a></code></pre></div>
<p><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-47.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-48.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-49.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-50.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-51.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-52.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-53.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-54.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-55.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-56.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-57.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-58.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-59.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-60.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-61.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-62.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-63.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-64.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-65.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-66.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-67.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-68.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-69.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">plots &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="st">`</span><span class="dt">PF3D7_1342900-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1408200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1222600</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1143100</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">                            <span class="st">`</span><span class="dt">PF3D7_1466400</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0802100</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0604100-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D3</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">                            <span class="st">`</span><span class="dt">PF3D7_0420300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1107800</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1350900</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1456000</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">                            <span class="st">`</span><span class="dt">PF3D7_1305200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1317200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0516800-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D2ext</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0420300-D2</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                            <span class="st">`</span><span class="dt">PF3D7_1239200-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1139300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0730300-DsL</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">plot=</span>plots,<span class="dt">filename=</span><span class="st">&quot;../output/tfbs_analysis/promoters_hb3/all_motifs.pdf&quot;</span>,<span class="dt">base_width=</span><span class="dv">26</span>,<span class="dt">base_height=</span><span class="dv">20</span>)</a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="kw">rm</span>(motifs_hb3)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8"></a>
<a class="sourceLine" id="cb7-9" data-line-number="9">motifs_it &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/promoters_it/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb7-11" data-line-number="11"></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="cf">for</span> (motif <span class="cf">in</span> <span class="kw">unique</span>(motifs_it<span class="op">$</span>motif_id)) {</a>
<a class="sourceLine" id="cb7-13" data-line-number="13">  n &lt;-<span class="st"> </span>motifs_it <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(motif_id<span class="op">==</span>motif)</a>
<a class="sourceLine" id="cb7-15" data-line-number="15">  <span class="co">#print(ks.test(n$start,runif(10000),alternative=&quot;less&quot;))</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16">  g &lt;-<span class="st"> </span>n <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>start)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Density&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="st">    </span><span class="kw">ggtitle</span>(motif) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="st">    </span>fig_theme</a>
<a class="sourceLine" id="cb7-22" data-line-number="22">  <span class="kw">assign</span>(<span class="dt">x=</span>motif,<span class="dt">value=</span>g)</a>
<a class="sourceLine" id="cb7-23" data-line-number="23">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb7-24" data-line-number="24">}</a></code></pre></div>
<p><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-70.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-71.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-72.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-73.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-74.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-75.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-76.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-77.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-78.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-79.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-80.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-81.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-82.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-83.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-84.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-85.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-86.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-87.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-88.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-89.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-90.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-91.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-5-92.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">plots &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="st">`</span><span class="dt">PF3D7_1342900-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1408200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1222600</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1143100</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">                            <span class="st">`</span><span class="dt">PF3D7_1466400</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0802100</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0604100-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D3</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                            <span class="st">`</span><span class="dt">PF3D7_0420300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1107800</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1350900</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1456000</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                            <span class="st">`</span><span class="dt">PF3D7_1305200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1317200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0516800-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D2ext</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0420300-D2</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">                            <span class="st">`</span><span class="dt">PF3D7_1239200-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1139300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0730300-DsL</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">plot=</span>plots,<span class="dt">filename=</span><span class="st">&quot;../output/tfbs_analysis/promoters_it/all_motifs.pdf&quot;</span>,<span class="dt">base_width=</span><span class="dv">26</span>,<span class="dt">base_height=</span><span class="dv">20</span>)</a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="kw">rm</span>(motifs_it)</a></code></pre></div>
</div>
<div id="apiap2-activity-predictions" class="section level2">
<h2>ApiAP2 Activity Predictions</h2>
<p>Here we will redo the analysis performed in Campbell et al. using motif hits within newly defined promoter regions. To perform this analysis, two R scripts need to be sourced from within the analysis working directory.</p>
<p>First <code>generate_inputs.R</code>, then <code>estimate_apiap2_activity.R</code>.</p>
</div>
<div id="bidirectional-promoters" class="section level2">
<h2>Bidirectional promoters</h2>
<p>It would be interesting to see whether there is an enrichment for a particular motif found within bidirectional promoters. First let’s extract bidirectional promoter sequences:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="cf">for</span> (strain <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;3d7&quot;</span>,<span class="st">&quot;hb3&quot;</span>,<span class="st">&quot;it&quot;</span>)) {</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">  divergent &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="kw">paste0</span>(<span class="st">&quot;../output/neighboring_genes/&quot;</span>,strain,<span class="st">&quot;_divergent.tsv&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">filter</span>(dist <span class="op">&lt;</span><span class="st"> </span><span class="dv">1000</span> <span class="op">&amp;</span><span class="st"> </span>dist <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>cor <span class="op">&gt;=</span><span class="st"> </span><span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5"></a>
<a class="sourceLine" id="cb9-6" data-line-number="6">  transcripts &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>(</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">    rtracklayer<span class="op">::</span><span class="kw">import.gff3</span>(</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">      <span class="kw">paste0</span>(<span class="st">&quot;../output/neighboring_genes/full_transcripts_&quot;</span>,strain,<span class="st">&quot;.gff&quot;</span>)))</a>
<a class="sourceLine" id="cb9-9" data-line-number="9">  </a>
<a class="sourceLine" id="cb9-10" data-line-number="10">  biprom &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">tibble</span>(<span class="dt">seqnames=</span><span class="kw">character</span>(),</a>
<a class="sourceLine" id="cb9-11" data-line-number="11">                           <span class="dt">start=</span><span class="kw">integer</span>(),</a>
<a class="sourceLine" id="cb9-12" data-line-number="12">                           <span class="dt">end=</span><span class="kw">integer</span>(),</a>
<a class="sourceLine" id="cb9-13" data-line-number="13">                           <span class="dt">strand=</span><span class="kw">character</span>(),</a>
<a class="sourceLine" id="cb9-14" data-line-number="14">                           <span class="dt">source=</span><span class="kw">character</span>(),</a>
<a class="sourceLine" id="cb9-15" data-line-number="15">                           <span class="dt">type=</span><span class="kw">character</span>(),</a>
<a class="sourceLine" id="cb9-16" data-line-number="16">                           <span class="dt">ID=</span><span class="kw">character</span>())</a>
<a class="sourceLine" id="cb9-17" data-line-number="17"></a>
<a class="sourceLine" id="cb9-18" data-line-number="18">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(divergent)) {</a>
<a class="sourceLine" id="cb9-19" data-line-number="19">    start_gene &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(transcripts, ID <span class="op">==</span><span class="st"> </span>divergent<span class="op">$</span>left_gene[i])</a>
<a class="sourceLine" id="cb9-20" data-line-number="20">    end_gene &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(transcripts, ID <span class="op">==</span><span class="st"> </span>divergent<span class="op">$</span>right_gene[i])</a>
<a class="sourceLine" id="cb9-21" data-line-number="21">    new &lt;-<span class="st"> </span>tibble<span class="op">::</span><span class="kw">tibble</span>(<span class="dt">seqnames=</span>start_gene<span class="op">$</span>seqnames,</a>
<a class="sourceLine" id="cb9-22" data-line-number="22">                        <span class="dt">start=</span>start_gene<span class="op">$</span>end,</a>
<a class="sourceLine" id="cb9-23" data-line-number="23">                        <span class="dt">end=</span>end_gene<span class="op">$</span>start,</a>
<a class="sourceLine" id="cb9-24" data-line-number="24">                        <span class="dt">strand=</span><span class="st">&quot;+&quot;</span>,</a>
<a class="sourceLine" id="cb9-25" data-line-number="25">                        <span class="dt">source=</span><span class="st">&quot;PlasmoDB&quot;</span>,</a>
<a class="sourceLine" id="cb9-26" data-line-number="26">                        <span class="dt">type=</span><span class="st">&quot;BiPromoter&quot;</span>,</a>
<a class="sourceLine" id="cb9-27" data-line-number="27">                        <span class="dt">ID=</span><span class="kw">paste0</span>(start_gene<span class="op">$</span>ID,<span class="st">&quot;-&quot;</span>,end_gene<span class="op">$</span>ID))</a>
<a class="sourceLine" id="cb9-28" data-line-number="28">    biprom &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">bind_rows</span>(biprom, new)</a>
<a class="sourceLine" id="cb9-29" data-line-number="29">  }</a>
<a class="sourceLine" id="cb9-30" data-line-number="30">  </a>
<a class="sourceLine" id="cb9-31" data-line-number="31">  biprom &lt;-<span class="st"> </span>GenomicRanges<span class="op">::</span><span class="kw">GRanges</span>(biprom)</a>
<a class="sourceLine" id="cb9-32" data-line-number="32">  rtracklayer<span class="op">::</span><span class="kw">export.gff3</span>(<span class="dt">object=</span>biprom,<span class="dt">con=</span><span class="kw">paste0</span>(<span class="st">&quot;../output/tfbs_analysis/bidirectional_&quot;</span>,strain,<span class="st">&quot;.gff&quot;</span>))</a>
<a class="sourceLine" id="cb9-33" data-line-number="33">  seqs &lt;-<span class="st"> </span>BSgenome<span class="op">::</span><span class="kw">getSeq</span>(BSgenome.Pfalciparum.PlasmoDB.v24, biprom)</a>
<a class="sourceLine" id="cb9-34" data-line-number="34">  <span class="kw">names</span>(seqs) &lt;-<span class="st"> </span>biprom<span class="op">$</span>ID</a>
<a class="sourceLine" id="cb9-35" data-line-number="35">  Biostrings<span class="op">::</span><span class="kw">writeXStringSet</span>(seqs,<span class="kw">paste0</span>(<span class="st">&quot;../output/tfbs_analysis/bidirectional_&quot;</span>,strain,<span class="st">&quot;.fasta&quot;</span>))</a>
<a class="sourceLine" id="cb9-36" data-line-number="36">}</a></code></pre></div>
<p>Now create background files and run fimo:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">for</span> <span class="ex">strain</span> in 3d7 hb3 it<span class="kw">;</span> <span class="kw">do</span> <span class="ex">fasta-get-markov</span> -m 3 ../output/tfbs_analysis/bidirectional_<span class="va">${strain}</span>.fasta ../output/tfbs_analysis/bidirectional_<span class="va">${strain}</span>.background<span class="kw">;</span> <span class="kw">done</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw">for</span> <span class="ex">strain</span> in 3d7 hb3 it<span class="kw">;</span> <span class="kw">do</span> <span class="ex">fimo</span> --bgfile ../output/tfbs_analysis/bidirectional_<span class="va">${strain}</span>.background --no-qvalue -oc ../output/tfbs_analysis/bidirectional_<span class="va">${strain}</span> --thresh 1e-2 ../data/motif_cores/ap2_pbm_cores.meme ../output/tfbs_analysis/bidirectional_<span class="va">${strain}</span>.fasta<span class="kw">;</span> <span class="kw">done</span></a></code></pre></div>
<p>And now we can read those files in and check the motif occurences:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">bimotifs_3d7 &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/bidirectional_3d7/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">biprom_3d7 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import.gff</span>(<span class="st">&quot;../output/tfbs_analysis/bidirectional_3d7.gff&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="st">  </span>tibble<span class="op">::</span><span class="kw">as_tibble</span>()</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"></a>
<a class="sourceLine" id="cb11-6" data-line-number="6">tmp &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(bimotifs_3d7,biprom_3d7, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;sequence_name&quot;</span>=<span class="st">&quot;ID&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(motif_id,sequence_name,start.x,width) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">norm_start=</span>start.x<span class="op">/</span>width)</a>
<a class="sourceLine" id="cb11-9" data-line-number="9"></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="cf">for</span> (m <span class="cf">in</span> <span class="kw">unique</span>(tmp<span class="op">$</span>motif_id)) {</a>
<a class="sourceLine" id="cb11-11" data-line-number="11">  b &lt;-<span class="st"> </span>tmp <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(motif_id<span class="op">==</span>m)</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">  g &lt;-<span class="st"> </span>b <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>norm_start)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="st">    </span><span class="kw">geom_line</span>(<span class="dt">stat=</span><span class="st">&quot;density&quot;</span>,<span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="st">    </span><span class="kw">ggtitle</span>(m) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="st">    </span>fig_theme</a>
<a class="sourceLine" id="cb11-18" data-line-number="18">  <span class="kw">assign</span>(<span class="dt">x=</span>m,<span class="dt">value=</span>g)</a>
<a class="sourceLine" id="cb11-19" data-line-number="19">  <span class="kw">print</span>(g)</a>
<a class="sourceLine" id="cb11-20" data-line-number="20">}</a></code></pre></div>
<p><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-16.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-18.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-20.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-21.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-22.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-23.png" width="672" style="display: block; margin: auto;" /><img src="figure/tfbs_analysis.Rmd/unnamed-chunk-8-24.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">plots &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="st">`</span><span class="dt">PF3D7_1342900-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1408200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1222600</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1143100</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">                            <span class="st">`</span><span class="dt">PF3D7_1466400</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0802100</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0604100-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D3</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                            <span class="st">`</span><span class="dt">PF3D7_0420300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1007700-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1107800</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1350900</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1456000</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">                            <span class="st">`</span><span class="dt">PF3D7_1305200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1317200</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0516800-D2</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0613800-D2ext</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0420300-D2</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">                            <span class="st">`</span><span class="dt">PF3D7_1239200-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1139300-D1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_0730300-DsL</span><span class="st">`</span>,<span class="st">`</span><span class="dt">PF3D7_1222400</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">cowplot<span class="op">::</span><span class="kw">save_plot</span>(<span class="dt">plot=</span>plots,<span class="dt">filename=</span><span class="st">&quot;../output/tfbs_analysis/bidirectional_3d7/all_motifs.svg&quot;</span>,<span class="dt">base_width=</span><span class="dv">26</span>,<span class="dt">base_height=</span><span class="dv">20</span>)</a></code></pre></div>
</div>
<div id="dynamic-motif-usage" class="section level2">
<h2>Dynamic motif usage</h2>
<p>Additionally, we can look at TSSs that we can confidentally say is shifting and analyze the motifs within these regions to find a nice example to display. We did this for KARHP:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="ex">fasta-get-markov</span> -m 3 ../output/tfbs_analysis/kahrp_short.fasta ../output/tfbs_analysis/kahrp_short.background</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="ex">fimo</span> --bgfile ../output/tfbs_analysis/kahrp_short.background -oc ../output/tfbs_analysis/kahrp_short --thresh 1e-3 ../data/motif_cores/ap2_pbm_cores.meme ../output/tfbs_analysis/kahrp_short.fasta</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="ex">fasta-get-markov</span> -m 3 ../output/tfbs_analysis/kahrp_long.fasta ../output/tfbs_analysis/kahrp_long.background</a>
<a class="sourceLine" id="cb13-6" data-line-number="6"></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="ex">fimo</span> --bgfile ../output/tfbs_analysis/kahrp_long.background -oc ../output/tfbs_analysis/kahrp_long --thresh 1e-3 ../data/motif_cores/ap2_pbm_cores.meme ../output/tfbs_analysis/kahrp_long.fasta</a></code></pre></div>
<p>Now we can look which motifs are unique to the short and long isoforms.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">kahrp_short &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/kahrp_short/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"></a>
<a class="sourceLine" id="cb14-4" data-line-number="4">kahrp_long &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../output/tfbs_analysis/kahrp_long/fimo.txt&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">motif_id=</span><span class="st">`</span><span class="dt"># motif_id</span><span class="st">`</span>)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6"></a>
<a class="sourceLine" id="cb14-7" data-line-number="7">short_unique &lt;-<span class="st"> </span>kahrp_short <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>(motif_id <span class="op">%in%</span><span class="st"> </span>kahrp_long<span class="op">$</span>motif_id))</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">long_unique &lt;-<span class="st"> </span>kahrp_long <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span>(motif_id <span class="op">%in%</span><span class="st"> </span>kahrp_short<span class="op">$</span>motif_id))</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"></a>
<a class="sourceLine" id="cb14-10" data-line-number="10">DT<span class="op">::</span><span class="kw">datatable</span>(short_unique, <span class="dt">rownames =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<div id="htmlwidget-e0c2e2742a5382bafd08" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e0c2e2742a5382bafd08">{"x":{"filter":"none","data":[["PF3D7_1317200","PF3D7_0516800-D2"],[null,null],["Pf3D7_02_v3:106234-106630","Pf3D7_02_v3:106234-106630"],[2,213],[7,218],["-","+"],[7.01149,6.69072],[0.00041,0.000696],[0.321,0.514],["TGCTGA","GAAATC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>motif_id<\/th>\n      <th>motif_alt_id<\/th>\n      <th>sequence_name<\/th>\n      <th>start<\/th>\n      <th>stop<\/th>\n      <th>strand<\/th>\n      <th>score<\/th>\n      <th>p-value<\/th>\n      <th>q-value<\/th>\n      <th>matched_sequence<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">DT<span class="op">::</span><span class="kw">datatable</span>(long_unique, <span class="dt">rownames =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<div id="htmlwidget-27bc00647663672df374" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-27bc00647663672df374">{"x":{"filter":"none","data":[["PF3D7_0613800-D2ext","PF3D7_1466400","PF3D7_1466400","PF3D7_1466400","PF3D7_1466400","PF3D7_0604100-DsL","PF3D7_1007700-D3","PF3D7_1007700-D3","PF3D7_1342900-D1","PF3D7_0604100-DsL","PF3D7_1007700-D3","PF3D7_1466400","PF3D7_1466400","PF3D7_0613800-D2ext","PF3D7_0613800-D2ext","PF3D7_1007700-D1","PF3D7_1007700-D1","PF3D7_0802100"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023","Pf3D7_02_v3:106631-107023"],[155,198,198,346,346,212,212,212,310,212,200,259,259,127,127,94,94,212],[160,203,203,351,351,217,217,217,315,217,205,264,264,132,132,99,99,217],["+","+","-","+","-","+","-","+","-","-","-","+","-","+","-","+","-","+"],[13.8667,10.8023,10.8023,10.8023,10.8023,7.45745,6.59302,6.5814,4.56383,6.87234,3.15116,5.2093,5.2093,7.46667,7.46667,3.24038,3.24038,6.92381],[4.37e-05,0.0001,0.0001,0.0001,0.0001,0.000289,0.000298,0.000342,0.000364,0.000425,0.000616,0.00067,0.00067,0.00083,0.00083,0.000849,0.000849,0.000996],[0.0338,0.0192,0.0192,0.0192,0.0192,0.164,0.131,0.131,0.272,0.164,0.157,0.0857,0.0857,0.214,0.214,0.325,0.325,0.731],["CTCTAG","ACATGC","GCATGT","GCATGT","ACATGC","GTACAC","GTGTAC","GTACAC","TCTCCG","GTGTAC","CTGCAT","ACATGT","ACATGT","CTATAG","CTATAG","GTCGTT","AACGAC","GTACAC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>motif_id<\/th>\n      <th>motif_alt_id<\/th>\n      <th>sequence_name<\/th>\n      <th>start<\/th>\n      <th>stop<\/th>\n      <th>strand<\/th>\n      <th>score<\/th>\n      <th>p-value<\/th>\n      <th>q-value<\/th>\n      <th>matched_sequence<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,6,7,8]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>R version 3.5.0 (2018-04-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Gentoo/Linux

Matrix products: default
BLAS: /usr/local/lib64/R/lib/libRblas.so
LAPACK: /usr/local/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gdtools_0.1.7                        
 [2] bindrcpp_0.2.2                       
 [3] BSgenome.Pfalciparum.PlasmoDB.v24_1.0
 [4] BSgenome_1.48.0                      
 [5] rtracklayer_1.40.6                   
 [6] Biostrings_2.48.0                    
 [7] XVector_0.20.0                       
 [8] GenomicRanges_1.32.7                 
 [9] GenomeInfoDb_1.16.0                  
[10] org.Pf.plasmo.db_3.6.0               
[11] AnnotationDbi_1.42.1                 
[12] IRanges_2.14.12                      
[13] S4Vectors_0.18.3                     
[14] Biobase_2.40.0                       
[15] BiocGenerics_0.26.0                  
[16] scales_1.0.0                         
[17] cowplot_0.9.3                        
[18] magrittr_1.5                         
[19] forcats_0.3.0                        
[20] stringr_1.3.1                        
[21] dplyr_0.7.6                          
[22] purrr_0.2.5                          
[23] readr_1.1.1                          
[24] tidyr_0.8.1                          
[25] tibble_1.4.2                         
[26] ggplot2_3.0.0                        
[27] tidyverse_1.2.1                      

loaded via a namespace (and not attached):
 [1] nlme_3.1-137                bitops_1.0-6               
 [3] matrixStats_0.54.0          lubridate_1.7.4            
 [5] bit64_0.9-7                 httr_1.3.1                 
 [7] rprojroot_1.3-2             tools_3.5.0                
 [9] backports_1.1.2             DT_0.4                     
[11] R6_2.3.0                    DBI_1.0.0                  
[13] lazyeval_0.2.1              colorspace_1.3-2           
[15] withr_2.1.2                 tidyselect_0.2.4           
[17] bit_1.1-14                  compiler_3.5.0             
[19] git2r_0.23.0                cli_1.0.1                  
[21] rvest_0.3.2                 xml2_1.2.0                 
[23] DelayedArray_0.6.6          labeling_0.3               
[25] digest_0.6.17               Rsamtools_1.32.3           
[27] svglite_1.2.1               rmarkdown_1.10             
[29] R.utils_2.7.0               pkgconfig_2.0.2            
[31] htmltools_0.3.6             htmlwidgets_1.3            
[33] rlang_0.2.2                 readxl_1.1.0               
[35] rstudioapi_0.8              RSQLite_2.1.1              
[37] shiny_1.1.0                 bindr_0.1.1                
[39] jsonlite_1.5                crosstalk_1.0.0            
[41] BiocParallel_1.14.2         R.oo_1.22.0                
[43] RCurl_1.95-4.11             GenomeInfoDbData_1.1.0     
[45] Matrix_1.2-14               Rcpp_0.12.19               
[47] munsell_0.5.0               R.methodsS3_1.7.1          
[49] stringi_1.2.4               whisker_0.3-2              
[51] yaml_2.2.0                  SummarizedExperiment_1.10.1
[53] zlibbioc_1.26.0             plyr_1.8.4                 
[55] grid_3.5.0                  blob_1.1.1                 
[57] promises_1.0.1              crayon_1.3.4               
[59] lattice_0.20-35             haven_1.1.2                
[61] hms_0.4.2                   knitr_1.20                 
[63] pillar_1.3.0                XML_3.98-1.16              
[65] glue_1.3.0                  evaluate_0.11              
[67] modelr_0.1.2                httpuv_1.4.5               
[69] cellranger_1.1.0            gtable_0.2.0               
[71] assertthat_0.2.0            mime_0.5                   
[73] xtable_1.8-3                broom_0.5.0                
[75] later_0.7.5                 GenomicAlignments_1.16.0   
[77] memoise_1.1.0               workflowr_1.1.1            </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
