---
title: "Make final UTRs"
author: "Philipp Ross"
date: 2018-09-25
output: workflowr::wflow_html
editor_options: 
  chunk_output_type: console
---

```{r,echo=F}
# ---- knitr-opts-chunk ----
knitr::opts_chunk$set(
  comment = NA,
  fig.align = "center",
  tidy = FALSE,
  fig.path = paste0("figure/", knitr::current_input(), "/"),
  warning = FALSE,
  message = FALSE
)

# ---- load-libraries ----
source("../code/utils.R")
load_essentials()
```

## Making final/longest UTRs

There were several methods and data sets used to predict 5' and 3' UTRs.

The first method was using the RNA-seq data sets and continuous read coverage. Any continuous coverage found surrounding known transcripts that met a read threshold cutoff of 5 reads or more was added as a UTR to the transcript from which it extended depending on whether it was on the 5' or 3' end of the transcript. This was done for each strain inidividually.

The second method to predict 5' UTRs was done by using a variant of nanoCAGE in order to tag the extreme ends of mRNA molecules and sequence them along with a synthetic oligo used for the pull down of these extreme ends. No 3' UTRs of transcripts was predicted using this approach although both ends were tagged.

Finally, the 5' mRNA capture data was used to correct 5' UTR predictions in all strains IF there was coverage support from the RNA-seq data in that loci where the capture data predicted it to be. If there was no prediction based on the 5' capture data, then coverage from the 3D7 strain was used to "repair" the 5' UTRs as well.

These predictions were all combined into lists of "longest possible transcripts" for each strain. Priority was given to predictions:

1. 5' mRNA capture prediction
2. 5' mRNA capture prediction used to repair the coverage prediction
3. 3D7 coverage repaired UTRs
4. Original RNA-seq UTR predictions

```{r,eval=F,engine='bash'}
# 3D7
Rscript scripts/create_final_utrs.R \
  ../data/utrs/repaired_cov_utrs/repaired_with_TSO_5UTRs/repaired_with_tso_5UTRs.3d7.filtered.rescued_5utrs.gff \
  ../data/utrs/repaired_cov_utrs/repaired_with_3D7_coverage_5UTRs/repaired_with_3d7_cov_5UTRs.3d7.filtered.rescued_5utrs.gff \
  ../data/utrs/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff \
  ../data/utrs/original_utrs/tso_thr5.gff \
  ../data/utrs/original_utrs/final.3utrs.3d7.3d7_v3_chr.idc.gff \
  TRUE \
  ../output/final_utrs/final_utrs_3d7.gff
  
# HB3
Rscript scripts/create_final_utrs.R \
  ../data/utrs/repaired_cov_utrs/repaired_with_TSO_5UTRs/repaired_with_tso_5UTRs.hb3.filtered.rescued_5utrs.gff \
  ../data/utrs/repaired_cov_utrs/repaired_with_3D7_coverage_5UTRs/repaired_with_3d7_cov_5UTRs.hb3.filtered.rescued_5utrs.gff \
  ../data/utrs/original_utrs/final.5utrs.hb3.3d7_v3_chr.idc.gff \
  ../data/utrs/original_utrs/tso_thr5.gff \
  ../data/utrs/original_utrs/final.3utrs.hb3.3d7_v3_chr.idc.gff \
  TRUE \
  ../output/final_utrs/final_utrs_hb3.gff

#IT
Rscript scripts/create_final_utrs.R \
  ../data/utrs/repaired_cov_utrs/repaired_with_TSO_5UTRs/repaired_with_tso_5UTRs.it.filtered.rescued_5utrs.gff \
  ../data/utrs/repaired_cov_utrs/repaired_with_3D7_coverage_5UTRs/repaired_with_3d7_cov_5UTRs.it.filtered.rescued_5utrs.gff \
  ../data/utrs/original_utrs/final.5utrs.it.3d7_v3_chr.idc.gff \
  ../data/utrs/original_utrs/tso_thr5.gff \
  ../data/utrs/original_utrs/final.3utrs.it.3d7_v3_chr.idc.gff \
  TRUE \
  ../output/final_utrs/final_utrs_it.gff
```

Finally, we should read in the resulting data:

```{r read}
# function to read in the data
read_utrs <- function(file, input_strain) {
  df <- tibble::as_tibble(rtracklayer::import.gff3(file)) %>%
    dplyr::select(Parent,type,width)
  df$strain <- input_strain
  df$Parent <- as.character(df$Parent)
  return(df)
}

utrs3d7 <- read_utrs("../output/final_utrs/longest_utrs_3d7_plasmodb_compatible.gff", "3D7")
utrshb3 <- read_utrs("../output/final_utrs/longest_utrs_hb3_plasmodb_compatible.gff", "HB3")
utrsit <- read_utrs("../output/final_utrs/longest_utrs_it_plasmodb_compatible.gff", "IT")
utrs <- rbind(utrs3d7,utrshb3,utrsit)
```

## 5' UTRs

What does the distribution of 5' UTRs look like?

```{r 5utrs-hist}
utrs %>% 
  dplyr::filter(type == "5UTR") %>% 
  ggplot(aes(x=width)) + 
  geom_histogram(color="grey90",bins=30) + 
  facet_grid(.~strain) +
  theme(axis.text.x = element_text(angle = 55, hjust = 1))
```

```{r 5utrs-boxplots}
utrs %>% 
  dplyr::filter(type == "5UTR") %>% 
  ggplot(aes(x=strain,y=width)) + 
  geom_boxplot(fill="grey70")
```

How do summary statistics vary between strains?

```{r 5utrs-summary}
utrs %>% 
  dplyr::filter(type == "5UTR") %>% 
  dplyr::group_by(strain) %>% 
  dplyr::summarise(mean=mean(width),median=median(width),sd=sd(width))
```

## 3' UTRs

What does the distribution of 3' UTRs look like?

```{r 3utrs-hist}
utrs %>% 
  dplyr::filter(type == "3UTR") %>% 
  ggplot(aes(x=width)) + 
  geom_histogram(color="grey90",bins=30) + 
  facet_grid(.~strain) + 
  theme(axis.text.x = element_text(angle = 55, hjust = 1))
```

```{r 3utrs-boxplots}
utrs %>% 
  dplyr::filter(type == "3UTR") %>% 
  ggplot(aes(x=strain,y=width)) + 
  geom_boxplot(fill="grey70")
```

How do summary statistics vary between strains?

```{r 3utrs-summary}
utrs %>% 
  dplyr::filter(type == "3UTR") %>% 
  dplyr::group_by(strain) %>% 
  dplyr::summarise(mean=mean(width),median=median(width),sd=sd(width))
```

## Homopolymer analysis

One thing we were also interested in looking at was why the coverage UTRs some times suffered from "coverage drops". If we assume that the AT content of the genome isn't as much of an issue anymore now that we are using PCR-free library preparation and better priming of RNA molecules, then one other issue that could be causing these coverage drops are homopolymer tracts. Another reason we are making this assumption is because the CAGE tags support this conclusion. Where the 5' UTR often falls short, there is a lack of supporting CAGE tags. If we look upstream, however, we can see CAGE tags AND coverage if we look at the DAFT-seq data.

In the current analysis, I'm forgetting to take into account introns when looking for the presence of long homopolymer tracts within predicted UTRs. Important questions include:

1. Do long homopolymer tracts exist within prediced coverage UTRs?
2. Are UTRs containing long homopolymer tracts on average shorter than those that don't?

To address these questions, I extended 5' UTRs 100 base pairs beyond their predicted coverage-based TSS, counted the length of the longest homopolymer tract for each nucleotide, and looked whether longer homopolymer tracts tend to occur in UTRs that are shorter.

First let's make 3D7 5UTR sequences for the analysis:

```{r,eval=F,engine='bash'}
cat ../data/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff | 
  awk 'BEGIN {{OFS=FS=\"\t\"}} {{split($9,X,\";\"); split(X[1],Y,\"=\"); print $1,$4-1,$5,Y[2],$5-$4+1,$7}}' | 
  bedtools getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed - -fo output/homopolymer_analysis/3d7_5utrs.fasta -s -name
  
bedtools flank -i ../data/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff -g ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.lengths -s -l 100 -r 0 |
  awk 'BEGIN {{OFS=FS=\"\t\"}} {{split($9,X,\";\"); split(X[1],Y,\"=\"); print $1,$4-1,$5,Y[2],$5-$4+1,$7}}' | 
  bedtools getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed - -fo ../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs.fasta -s -name
  
bedtools slop -i ../data/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff -g ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.lengths -s -l 100 -r 0 |
  awk 'BEGIN {{OFS=FS=\"\t\"}} {{split($9,X,\";\"); split(X[1],Y,\"=\"); print $1,$4-1,$5,Y[2],$5-$4+1,$7}}' | 
  bedtools getfasta -fi ../data/genomes/pf3d7/PlasmoDB-24_Pfalciparum3D7_Genome.fasta -bed - -fo ../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs.fasta -s -name
```

Now let's count the number of homopolymers:

```{r,eval=F,engine='bash'}
Rscript ../code/scripts/final_utrs/count_homopolymers.R -f ../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs.fasta -l 50 -o ../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs_counts.tsv

Rscript ../code/scripts/final_utrs/count_homopolymers.R -f ../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs.fasta -l 50 -o ../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs_counts.tsv

Rscript ../code/scripts/final_utrs/count_homopolymers.R -f ../output/homopolymer_analysis/3d7_5utrs.fasta -l 50 -o ../output/homopolymer_analysis/3d7_5utrs.fasta_counts.tsv
```

Now import that data:

```{r}
utrplus <- readr::read_tsv("../output/homopolymer_analysis/minus_100bp_slop_3d7_5utrs_counts.tsv") # UTR plus 100bp upstream
plus    <- readr::read_tsv("../output/homopolymer_analysis/minus_100bp_flank_3d7_5utrs_counts.tsv") # only 100 bp upstream of UTR
utrs    <- readr::read_tsv("../output/homopolymer_analysis/3d7_5utrs_counts.tsv") # only UTR

originalutrs <- tibble::as_tibble(rtracklayer::import.gff3("../data/utrs/original_utrs/final.5utrs.3d7.3d7_v3_chr.idc.gff"))
originalutrs$Parent <- as.character(originalutrs$Parent)

fixedutrs <- tibble::as_tibble(rtracklayer::import.gff3("../output/final_utrs/final_utrs_3d7.gff")) %>%
  dplyr::filter(length_added > 0)
fixedutrs$Parent <- as.character(fixedutrs$Parent)
```

And now plot the data:

```{r}
tmp <- dplyr::inner_join(originalutrs, utrplus, by = c("Parent" = "gene_id"))

g <- tmp %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  ggplot(aes(y=width/longest,x=longest)) + 
  geom_point(alpha=0.25,color="black") + 
  stat_smooth(method="lm",color="red") +
  ylab("Normalized Length") + 
  xlab("Longest Homopolymer") +
  ggtitle("All UTRs Plus 100 bps")

cowplot::save_plot("../output/final_utrs/utrplus_homopolymers.png",g)
cowplot::save_plot("../output/final_utrs/utrplus_homopolymers.svg",g)

print(g)

tmp %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  dplyr::do(broom::tidy(cor.test(.$width/.$longest,.$longest)))
```

```{r}
tmp <- dplyr::inner_join(originalutrs, plus, by = c("Parent" = "gene_id"))

g <- tmp %>%
  dplyr::filter(width < 8000) %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  ggplot(aes(y=width,x=longest)) + geom_point(alpha=0.5) + stat_smooth(method="lm")

cowplot::save_plot("../output/final_utrs/plus_homopolymers.png",g)
cowplot::save_plot("../output/final_utrs/plus_homopolymers.svg",g)

print(g)

tmp %>%
  #dplyr::filter(width < 8000) %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  dplyr::do(broom::tidy(cor.test(.$width,.$longest)))
```

```{r}
tmp <- dplyr::inner_join(originalutrs, utrs, by = c("Parent" = "gene_id"))

g <- tmp %>%
  dplyr::filter(width < 8000) %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  ggplot(aes(y=width/longest,x=longest)) + geom_point(alpha=0.5) + stat_smooth(method="lm")

cowplot::save_plot("../output/final_utrs/utr_homopolymers.png",g)
cowplot::save_plot("../output/final_utrs/utr_homopolymers.svg",g)

print(g)

tmp %>%
  #dplyr::filter(width < 8000) %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  dplyr::do(broom::tidy(cor.test(.$width/.$longest,.$longest)))
```

```{r}
tmp <- dplyr::inner_join(fixedutrs, plus, by = c("Parent" = "gene_id"))

g <- tmp %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  ggplot(aes(y=width/longest,x=longest)) + 
  geom_point(alpha=0.25,color="black") + 
  stat_smooth(method="lm",color="red") +
  ylab("") + 
  xlab("") +
  scale_y_continuous(limits=c(-10,800)) +
  scale_x_continuous(breaks=c(10,20,30,40,50))
  #ggtitle("Fixed UTRs")

cowplot::save_plot("../output/final_utrs/fixedplus_homopolymers.png",g)
cowplot::save_plot("../output/final_utrs/fixedplus_homopolymers.svg",g)

print(g)

tmp %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  dplyr::do(broom::tidy(cor.test(.$width/.$longest,.$longest)))
```

```{r}
tmp <- dplyr::inner_join(fixedutrs, utrplus, by = c("Parent" = "gene_id"))

g <- tmp %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  ggplot(aes(y=width/longest,x=longest)) + 
  geom_point(alpha=0.25,color="black") + 
  stat_smooth(method="lm",color="red") +
  ylab("Normalized Length") + 
  xlab("Longest Homopolymer") +
  ggtitle("Fixed UTRs")

cowplot::save_plot("../output/final_utrs/fixedutrplus_homopolymers.png",g)
cowplot::save_plot("../output/final_utrs/fixedutrplus_homopolymers.svg",g)

print(g)

tmp %>%
  dplyr::group_by(Parent) %>%
  dplyr::summarise(width=unique(width),longest=max(longest)) %>%
  dplyr::do(broom::tidy(cor.test(.$width/.$longest,.$longest)))
```
