---
title: "Differential transcript expression"
author: "Philipp Ross"
date: 2018-09-25
output: workflowr::wflow_html
editor_options: 
  chunk_output_type: console
---

```{r,echo=F}
# ---- knitr-opts-chunk ----
knitr::opts_chunk$set(
  comment = NA,
  fig.align = "center",
  tidy = FALSE,
  fig.path = paste0("figure/", knitr::current_input(), "/"),
  warning = FALSE,
  message = FALSE
)

# ---- load-libraries ----
source("../code/utils.R")
load_essentials()
```

## Process data

First we need to actually calculate the differentially expressed genes. We'll need to import some data first:

```{r}
core_genes <- readr::read_tsv("../data/gene_lists/core_pf3d7_genes.txt",col_names=F)$X1

x3d7exp <- readRDS("../output/neighboring_genes/gene_reduced_3d7_abund.rds")
xhb3exp <- readRDS("../output/neighboring_genes/gene_reduced_hb3_abund.rds")
xitexp  <- readRDS("../output/neighboring_genes/gene_reduced_it_abund.rds")

abund <- dplyr::bind_rows(x3d7exp,xhb3exp,xitexp)

rm(x3d7exp,xhb3exp,xitexp)

# unique gene IDs
ug <- unique(abund$gene_id)

# Create data frame of genes that are "on" in each strain
detected <- abund %>%
  dplyr::group_by(gene_id, strain) %>%
  dplyr::summarise(n = sum(TPM >= 5)) %>%
  dplyr::mutate(on = ifelse(n > 0, 1, 0)) %>%
  dplyr::ungroup() %>%
  dplyr::select(-n)

# What genes are "on" in all strains
onall <- detected %>%
  dplyr::filter(on == 1) %>%
  dplyr::group_by(gene_id) %>%
  dplyr::summarise(count = length(unique(strain))) %>%
  dplyr::filter(count == 3) %$%
  gene_id

# What genes are "off" in all strains
offall <- detected %>%
  dplyr::filter(on == 0) %>%
  dplyr::group_by(gene_id) %>%
  dplyr::summarise(count = length(unique(strain))) %>%
  dplyr::filter(count == 3) %$%
  gene_id

on3d7 <- detected %>% filter(strain == "3d7" & on == 1 & gene_id %in% core_genes) %$% gene_id
onhb3 <- detected %>% filter(strain == "hb3" & on == 1& gene_id %in% core_genes) %$% gene_id
onit  <- detected %>% filter(strain == "it" & on == 1 & gene_id %in% core_genes) %$% gene_id

on3d7hb3 <- intersect(on3d7,onhb3)
on3d7it  <- intersect(on3d7,onit)
onhb3it  <- intersect(onhb3,onit)

comparable <- intersect(core_genes, onall)
```

Now we can look to calculate the differentially expressed genes. 

```{r}
# First calculate differential expression
diff_exp_3d7hb3 <- abund %>%
  filter(strain %in% c("3d7", "hb3") & gene_id %in% comparable) %>%
  group_by(gene_id) %>%
  do(broom::tidy(t.test(log2(TPM+1)~strain,data=.))) %>%
  dplyr::ungroup() %>%
  dplyr::select(gene_id, p.value)

diff_exp_3d7hb3$strain_pair <- "3d7-hb3"

diff_exp_3d7it <- abund %>%
  filter(strain %in% c("3d7", "it") & gene_id %in% comparable) %>%
  group_by(gene_id) %>%
  do(broom::tidy(t.test(log2(TPM+1)~strain,data=.))) %>%
  dplyr::ungroup() %>%
  dplyr::select(gene_id, p.value)

diff_exp_3d7it$strain_pair <- "3d7-it"

diff_exp_hb3it <- abund %>%
  filter(strain %in% c("hb3", "it") & gene_id %in% comparable) %>%
  group_by(gene_id) %>%
  do(broom::tidy(t.test(log2(TPM+1)~strain,data=.))) %>%
  dplyr::ungroup() %>%
  dplyr::select(gene_id, p.value)

diff_exp_hb3it$strain_pair <- "hb3-it"

diff_exp <- rbind(diff_exp_3d7hb3, diff_exp_3d7it, diff_exp_hb3it)

diff3d7hb3 <- diff_exp %>% filter(strain_pair == "3d7-hb3") %>% separate(strain_pair, into = c("strain1","strain2"), sep = "-")
diff3d7it  <- diff_exp %>% filter(strain_pair == "3d7-it") %>% separate(strain_pair, into = c("strain1","strain2"), sep = "-")
diffhb3it  <- diff_exp %>% filter(strain_pair == "hb3-it") %>% separate(strain_pair, into = c("strain1","strain2"), sep = "-")

diff3d7hb3$qvalue <- qvalue::qvalue(diff3d7hb3$p.value)$qvalues
diff3d7it$qvalue  <- qvalue::qvalue(diff3d7it$p.value)$qvalues
diffhb3it$qvalue  <- qvalue::qvalue(diffhb3it$p.value)$qvalues
```

```{r}
readr::write_tsv(x=diff3d7hb3,path="../output/differential_expression/diff3d7hb3.tsv")
googledrive::drive_upload(media="../output/differential_expression/diff3d7hb3.tsv",
                          path="Shared/Pf RNA-seq manuscript 2017/Supplementary tables/Named tables",
                          name="diff3d7hb3",
                          type="spreadsheet")
readr::write_tsv(x=diff3d7it,path="../output/differential_expression/diff3d7it.tsv")
googledrive::drive_upload(media="../output/differential_expression/diff3d7it.tsv",
                          path="Shared/Pf RNA-seq manuscript 2017/Supplementary tables/Named tables",
                          name="diff3d7it",
                          type="spreadsheet")
readr::write_tsv(x=diffhb3it,path="../output/differential_expression/diffhb3it.tsv")
googledrive::drive_upload(media="../output/differential_expression/diffhb3it.tsv",
                          path="Shared/Pf RNA-seq manuscript 2017/Supplementary tables/Named tables",
                          name="diffhb3it",
                          type="spreadsheet")
```

And finally, we can plot the results and write everything to a file:

```{r}
# Create a scatter plot comparing 3D7 to HB3
plot_3d7_hb3_scatter <- function() {

  overunder <- abund %>%
    filter(gene_id %in% on3d7hb3) %>%
    group_by(gene_id, strain) %>%
    summarise(m = mean(TPM)) %>%
    spread(strain, m) %>%
    mutate(`3d7_over_hb3` = log2(`3d7` / hb3)) %>%
    dplyr::select(-`3d7`, -hb3, -it) %>%
    inner_join(diff3d7hb3, by = "gene_id") %>%
    gather(strain1, strain2,-`3d7_over_hb3`,-gene_id,-p.value,-qvalue) %>%
    dplyr::select(-strain1) %>%
    dplyr::rename(strain=strain2)

  df <- abund %>%
    filter(gene_id %in% on3d7hb3) %>%
    group_by(gene_id, strain) %>%
    summarise(m = mean(TPM)) %>%
    ungroup() %>%
    filter(strain %in% c("3d7","hb3"))

  df <- inner_join(df, overunder, by = c("gene_id", "strain"))

  df %>%
    spread(strain, m) %>%
    ggplot(aes(x=hb3,y=`3d7`,label=gene_id)) +
    geom_point(color="grey80") +
    geom_point(data = subset(df, (`3d7_over_hb3` <= -2) & qvalue < 0.05) %>% ungroup() %>% spread(strain, m), color = "#E41A1C") +
    geom_point(data = subset(df, (`3d7_over_hb3` >= 2) & qvalue < 0.05) %>% ungroup() %>% spread(strain, m), color = "#377EB8") +
    ylab("Log10(TPM) 3D7") +
    xlab("Log10(TPM) HB3") +
    scale_x_continuous(trans="log10",breaks=c(1,10,100,1000,10000),labels=trans_format('log10',math_format(10^.x)),limits=c(1,10000)) +
    scale_y_continuous(trans="log10",breaks=c(1,10,100,1000,10000),labels=trans_format('log10',math_format(10^.x)),limits=c(1,10000)) +
    theme(axis.line.x=element_line(size=1),
          axis.line.y=element_line(size=1),
          axis.text=element_text(size=20),
          axis.title=element_text(size=20),
          axis.ticks.x=element_line(colour="black",size=1),
          axis.ticks.y=element_line(colour="black",size=1))
}

# Create a scatter plot comparing 3D7 to IT
plot_3d7_it_scatter <- function() {

  overunder <- abund %>%
    filter(gene_id %in% on3d7it) %>%
    group_by(gene_id, strain) %>%
    summarise(m = mean(TPM)) %>%
    spread(strain, m) %>%
    mutate(`3d7_over_it` = log2(`3d7` / it)) %>%
    dplyr::select(-`3d7`, -hb3, -it) %>%
    inner_join(diff3d7it, by = "gene_id") %>%
    gather(strain1, strain2,-`3d7_over_it`,-gene_id,-p.value,-qvalue) %>%
    dplyr::select(-strain1) %>%
    dplyr::rename(strain=strain2)

  df <- abund %>%
    filter(gene_id %in% on3d7it) %>%
    group_by(gene_id, strain) %>%
    summarise(m = mean(TPM)) %>%
    ungroup() %>%
    filter(strain %in% c("3d7","it"))

  df <- inner_join(df, overunder, by = c("gene_id", "strain"))

  df %>%
    spread(strain, m) %>%
    ggplot(aes(x=it,y=`3d7`,label=gene_id)) +
    geom_point(color="grey80") +
    #geom_point(data = subset(df, (`3d7_over_it` <= -2) & qvalue < 0.05) %>% ungroup() %>% spread(strain, m), color = "#4DAF4A") +
    geom_point(data = subset(df, (`3d7_over_it` >= 2) & qvalue < 0.05) %>% ungroup() %>% spread(strain, m), color = "#377EB8") +
    ylab("Log10(TPM) 3D7") +
    xlab("Log10(TPM) IT") +
    scale_x_continuous(trans="log10",breaks=c(1,10,100,1000,10000),labels=trans_format('log10',math_format(10^.x)),limits=c(1,10000)) +
    scale_y_continuous(trans="log10",breaks=c(1,10,100,1000,10000),labels=trans_format('log10',math_format(10^.x)),limits=c(1,10000)) +
    theme(axis.line.x=element_line(size=1),
          axis.line.y=element_line(size=1),
          axis.text=element_text(size=20),
          axis.title=element_text(size=20),
          axis.ticks.x=element_line(colour="black",size=1),
          axis.ticks.y=element_line(colour="black",size=1))
}

# Create a scatter plot comparing HB3 to IT
plot_hb3_it_scatter <- function() {

  overunder <- abund %>%
    filter(gene_id %in% onhb3it) %>%
    group_by(gene_id, strain) %>%
    summarise(m = mean(TPM)) %>%
    spread(strain, m) %>%
    mutate(hb3_over_it = log2(hb3 / it)) %>%
    dplyr::select(-`3d7`, -hb3, -it) %>%
    inner_join(diffhb3it, by = "gene_id") %>%
    gather(strain1, strain2,-hb3_over_it,-gene_id,-p.value,-qvalue) %>%
    dplyr::select(-strain1) %>%
    dplyr::rename(strain=strain2)

  df <- abund %>%
    filter(gene_id %in% onhb3it) %>%
    group_by(gene_id, strain) %>%
    summarise(m = mean(TPM)) %>%
    ungroup() %>%
    filter(strain %in% c("hb3","it"))

  df <- inner_join(df, overunder, by = c("gene_id", "strain"))

  df %>%
    spread(strain, m) %>%
    ggplot(aes(x=hb3,y=it,label=gene_id)) +
    geom_point(color="grey80") +
    geom_point(data = subset(df, (hb3_over_it <= -2) & qvalue < 0.05) %>% ungroup() %>% spread(strain, m), color = "#E41A1C") +
    geom_point(data = subset(df, (hb3_over_it >= 2) & qvalue < 0.05) %>% ungroup() %>% spread(strain, m), color = "#4DAF4A") +
    ylab("Log10(TPM) HB3") +
    xlab("Log10(TPM) IT") +
    scale_x_continuous(trans="log10",breaks=c(1,10,100,1000,10000),labels=trans_format('log10',math_format(10^.x)),limits=c(1,10000)) +
    scale_y_continuous(trans="log10",breaks=c(1,10,100,1000,10000),labels=trans_format('log10',math_format(10^.x)),limits=c(1,10000)) +
    theme(axis.line.x=element_line(size=1),
          axis.line.y=element_line(size=1),
          axis.text=element_text(size=20),
          axis.title=element_text(size=20),
          axis.ticks.x=element_line(colour="black",size=1),
          axis.ticks.y=element_line(colour="black",size=1))
}
```

```{r}
g <- plot_3d7_hb3_scatter()
ggsave(plot=g,filename="../output/differential_expression/3d7_hb3_scatter.svg")
ggsave(plot=g,filename="../output/differential_expression/3d7_hb3_scatter.png")
print(g)

g <- plot_3d7_it_scatter()
ggsave(plot=g,filename="../output/differential_expression/3d7_it_scatter.svg")
ggsave(plot=g,filename="../output/differential_expression/3d7_it_scatter.png")
print(g)

g <- plot_hb3_it_scatter()
ggsave(plot=g,filename="../output/differential_expression/hb3_it_scatter.svg")
ggsave(plot=g,filename="../output/differential_expression/hb3_it_scatter.png")
print(g)
```

## Manuscript numbers

### How many differentially expressed genes do we find?

```{r}
overunder <- abund %>%
  filter(gene_id %in% on3d7hb3) %>%
  group_by(gene_id, strain) %>%
  summarise(m = mean(TPM)) %>%
  spread(strain, m) %>%
  mutate(`3d7_over_hb3` = log2(`3d7` / hb3)) %>%
  dplyr::select(-`3d7`, -hb3, -it) %>%
  inner_join(diff3d7hb3, by = "gene_id") %>%
  gather(strain1, strain2,-`3d7_over_hb3`,-gene_id,-p.value,-qvalue) %>%
  dplyr::select(-strain1) %>%
  dplyr::rename(strain=strain2)

df <- abund %>%
  filter(gene_id %in% on3d7hb3) %>%
  group_by(gene_id, strain) %>%
  summarise(m = mean(TPM)) %>%
  ungroup() %>%
  filter(strain %in% c("3d7","hb3"))

df <- inner_join(df, overunder, by = c("gene_id", "strain"))

print("Number of genes overexpressed in HB3:")
print(length(unique(subset(df, (`3d7_over_hb3` <= -2) & qvalue < 0.05)$gene_id)))
readr::write_lines(x=unique(subset(df, (`3d7_over_hb3` <= -2) & qvalue < 0.05)$gene_id),path="../output/differential_expression/hb3_over_3d7_genes.txt")
print("Number of genes overexpressed in 3D7:")
print(length(unique(subset(df, (`3d7_over_hb3` >= 2) & qvalue < 0.05)$gene_id)))
readr::write_lines(x=unique(subset(df, (`3d7_over_hb3` >= 2) & qvalue < 0.05)$gene_id),path="../output/differential_expression/3d7_over_hb3_genes.txt")

###########################################################

overunder <- abund %>%
  filter(gene_id %in% on3d7it) %>%
  group_by(gene_id, strain) %>%
  summarise(m = mean(TPM)) %>%
  spread(strain, m) %>%
  mutate(`3d7_over_it` = log2(`3d7` / it)) %>%
  dplyr::select(-`3d7`, -hb3, -it) %>%
  inner_join(diff3d7it, by = "gene_id") %>%
  gather(strain1, strain2,-`3d7_over_it`,-gene_id,-p.value,-qvalue) %>%
  dplyr::select(-strain1) %>%
  dplyr::rename(strain=strain2)

df <- abund %>%
  filter(gene_id %in% on3d7it) %>%
  group_by(gene_id, strain) %>%
  summarise(m = mean(TPM)) %>%
  ungroup() %>%
  filter(strain %in% c("3d7","it"))

df <- inner_join(df, overunder, by = c("gene_id", "strain"))

print("Number of genes overexpressed in IT:")
print(length(unique(subset(df, (`3d7_over_it` <= -2) & qvalue < 0.05)$gene_id)))
readr::write_lines(x=unique(subset(df, (`3d7_over_it` <= -2) & qvalue < 0.05)$gene_id),path="../output/differential_expression/it_over_3d7_genes.txt")
print("Number of genes overexpressed in 3D7:")
print(length(unique(subset(df, (`3d7_over_it` >= 2) & qvalue < 0.05)$gene_id)))
readr::write_lines(x=unique(subset(df, (`3d7_over_it` >= 2) & qvalue < 0.05)$gene_id),path="../output/differential_expression/3d7_over_it_genes.txt")

###########################################################

overunder <- abund %>%
  filter(gene_id %in% onhb3it) %>%
  group_by(gene_id, strain) %>%
  summarise(m = mean(TPM)) %>%
  spread(strain, m) %>%
  mutate(`hb3_over_it` = log2(`hb3` / it)) %>%
  dplyr::select(-`3d7`, -hb3, -it) %>%
  inner_join(diffhb3it, by = "gene_id") %>%
  gather(strain1, strain2,-`hb3_over_it`,-gene_id,-p.value,-qvalue) %>%
  dplyr::select(-strain1) %>%
  dplyr::rename(strain=strain2)

df <- abund %>%
  filter(gene_id %in% onhb3it) %>%
  group_by(gene_id, strain) %>%
  summarise(m = mean(TPM)) %>%
  ungroup() %>%
  filter(strain %in% c("hb3","it"))

df <- inner_join(df, overunder, by = c("gene_id", "strain"))

print("Number of genes overexpressed in HB3:")
print(length(unique(subset(df, (`hb3_over_it` <= -2) & qvalue < 0.05)$gene_id)))
readr::write_lines(x=unique(subset(df, (`hb3_over_it` <= -2) & qvalue < 0.05)$gene_id),path="../output/differential_expression/it_over_hb3_genes.txt")
print("Number of genes overexpressed in IT:")
print(length(unique(subset(df, (`hb3_over_it` >= 2) & qvalue < 0.05)$gene_id)))
readr::write_lines(x=unique(subset(df, (`hb3_over_it` >= 2) & qvalue < 0.05)$gene_id),path="../output/differential_expression/hb3_over_it_genes.txt")
```

Now run this from the command line:

```{r,engine='bash',eval=F}
for file in $(ls ../output/differential_expression/*.txt); do Rscript code/topgo.R --gene_list ../output/differential_expression/$file --go_terms ../data/gene_ontology/Pf3D7_go_sept2014.txt --anno ../data/gene_ontology/Pf3D7_desc_sept2014.txt --out_prefix ../output/differential_expression/${file%%.*}; done
```

Plot the abundances for the following ApiAP2 transcripts...

```{r}
plot_ap2_strain_abundances <- function(df, gid) {

  gene_names <- list("PF3D7_0604100"="SIP2",
                     "PF3D7_1222400"="",
                     "PF3D7_1143100"="AP2-O",
                     "PF3D7_0516800"="AP2-O2",
                     "PF3D7_0420300"="",
                     "PF3D7_0613800"="",
                     "PF3D7_0802100"="",
                     "PF3D7_1107800"="",
                     "PF3D7_1139300"="",
                     "PF3D7_1239200"="",
                     "PF3D7_1317200"="AP2-G3",
                     "PF3D7_1342900"="",
                     "PF3D7_1222600"="AP2-G",
                     "PF3D7_1466400"="AP2-EXP",
                     "PF3D7_0622900"="AP2-TEL",
                     "PF3D7_1350900"="AP2-O4",
                     "PF3D7_0404100"="AP2-SP2",
                     "PF3D7_0730300"="AP2-L",
                     "PF3D7_1408200"="AP2-G2",
                     "PF3D7_1429200"="AP2-O3",
                     "PF3D7_1449500"="AP2-O5",
                     "PF3D7_1456000"="",
                     "PF3D7_1305200"="",
                     "PF3D7_0611200"="",
                     "PF3D7_0934400"="",
                     "PF3D7_1007700"="AP2-I",
                     "PF3D7_1115500"="",
                     "PF3D7_0932200"="")

  df %>%
    dplyr::filter(gene_id == gid) %>%
    ggplot(aes(x = toupper(strain), y = as.numeric(TPM), group = tp, fill = toupper(strain))) +
    geom_bar(stat="identity", position="dodge", colour="black") +
    ylab("") +
    xlab("") +
    ggtitle(paste0(gene_names[[gid]],"\n",gid)) +
    scale_fill_hue(l=55) +
    theme_classic() +
    theme(axis.text.x = element_blank(),
          axis.text.y = element_text(vjust=0.5, size=24),
          plot.title = element_text(size=28,hjust = 0.5),
          axis.line.x = element_line(size=0.5),
          axis.line.y = element_line(size=0.5),
          axis.ticks.x = element_blank(),
          axis.ticks.y = element_line(size = 0.5),
          axis.ticks.length = unit(0.25, "cm"),
          axis.ticks = element_line(size = 0.5),
          legend.position = "none") +
    fill_colors +
    panel_border(colour="black",size=1)
}
```

```{r}
significant <- c("PF3D7_0604100","PF3D7_1222400","PF3D7_1408200",
          "PF3D7_1143100","PF3D7_0516800","PF3D7_1466400",
          "PF3D7_0420300","PF3D7_0613800","PF3D7_0802100",
          "PF3D7_1107800","PF3D7_1139300","PF3D7_1239200",
          "PF3D7_1317200","PF3D7_1342900","PF3D7_1449500")

for (w in significant) {
  g <- plot_ap2_strain_abundances(abund,w)
  ggsave(plot=g,filename=paste0("../output/differential_expression/",w,".svg"),height=4,width=4)
  ggsave(plot=g,filename=paste0("../output/differential_expression/",w,".png"),height=4,width=4)
  print(g)
}

nonsignificant <- c("PF3D7_1222600","PF3D7_0622900","PF3D7_1350900","PF3D7_0404100",
                    "PF3D7_0730300","PF3D7_1429200",
                    "PF3D7_1456000","PF3D7_1305200","PF3D7_0611200",
                    "PF3D7_0934400","PF3D7_1007700","PF3D7_1115500",
                    "PF3D7_0932200")

for (w in nonsignificant) {
  g <- plot_ap2_strain_abundances(abund,w)
  ggsave(plot=g,filename=paste0("../output/differential_expression/",w,".svg"),height=4,width=4)
  ggsave(plot=g,filename=paste0("../output/differential_expression/",w,".png"),height=4,width=4)
  print(g)
}
```

